<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Bioconductor | R Course</title>
  <meta name="description" content="Chapter 7 Bioconductor | R Course" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Bioconductor | R Course" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Bioconductor | R Course" />
  
  
  

<meta name="author" content="YOUR NAME HERE" />


<meta name="date" content="2024-07-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="ggplot.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">SHORT TITLE HERE</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> R - Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#r---basic-concepts-and-applications"><i class="fa fa-check"></i><b>1.1</b> R - Basic Concepts and applications</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#why-r"><i class="fa fa-check"></i><b>1.2</b> Why R?</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#r-studio"><i class="fa fa-check"></i><b>1.3</b> R studio</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#rstudio---look-and-feel"><i class="fa fa-check"></i><b>1.3.1</b> RStudio - Look and Feel</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#basic-mathematics-like-operators"><i class="fa fa-check"></i><b>1.4</b> Basic Mathematics like operators</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="index.html"><a href="index.html#basics"><i class="fa fa-check"></i><b>1.4.1</b> Basics</a></li>
<li class="chapter" data-level="1.4.2" data-path="index.html"><a href="index.html#bodmas-rule"><i class="fa fa-check"></i><b>1.4.2</b> BODMAS rule</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#r-markdown"><i class="fa fa-check"></i><b>1.5</b> R Markdown</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#installation-of-packages"><i class="fa fa-check"></i><b>1.6</b> Installation of packages</a></li>
<li class="chapter" data-level="1.7" data-path="index.html"><a href="index.html#assignment"><i class="fa fa-check"></i><b>1.7</b> Assignment</a></li>
<li class="chapter" data-level="1.8" data-path="index.html"><a href="index.html#code-challenge"><i class="fa fa-check"></i><b>1.8</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html"><i class="fa fa-check"></i><b>2</b> Data Types and Data Structures</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#data-types"><i class="fa fa-check"></i><b>2.1</b> Data types</a></li>
<li class="chapter" data-level="2.2" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#data-structures"><i class="fa fa-check"></i><b>2.2</b> Data Structures</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#vector"><i class="fa fa-check"></i><b>2.2.1</b> Vector</a></li>
<li class="chapter" data-level="2.2.2" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#matrix"><i class="fa fa-check"></i><b>2.2.2</b> Matrix</a></li>
<li class="chapter" data-level="2.2.3" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#data-frame"><i class="fa fa-check"></i><b>2.2.3</b> Data frame</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#code-challenge-1"><i class="fa fa-check"></i><b>2.3</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="loops.html"><a href="loops.html"><i class="fa fa-check"></i><b>3</b> Loops</a>
<ul>
<li class="chapter" data-level="3.1" data-path="loops.html"><a href="loops.html#for-loops"><i class="fa fa-check"></i><b>3.1</b> For loops</a></li>
<li class="chapter" data-level="3.2" data-path="loops.html"><a href="loops.html#while-loop"><i class="fa fa-check"></i><b>3.2</b> While loop</a></li>
<li class="chapter" data-level="3.3" data-path="loops.html"><a href="loops.html#code-challenge-2"><i class="fa fa-check"></i><b>3.3</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="working-with-files.html"><a href="working-with-files.html"><i class="fa fa-check"></i><b>4</b> Working with files</a>
<ul>
<li class="chapter" data-level="4.1" data-path="working-with-files.html"><a href="working-with-files.html#xlsx-xls-files"><i class="fa fa-check"></i><b>4.1</b> .XLSX, XLS files</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="working-with-files.html"><a href="working-with-files.html#reading-a-file-to-a-dataframe"><i class="fa fa-check"></i><b>4.1.1</b> Reading a file to a dataframe</a></li>
<li class="chapter" data-level="4.1.2" data-path="working-with-files.html"><a href="working-with-files.html#write-a-new-table-to-a-file-locally"><i class="fa fa-check"></i><b>4.1.2</b> Write a new table to a file locally</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="working-with-files.html"><a href="working-with-files.html#csv-files"><i class="fa fa-check"></i><b>4.2</b> .CSV files</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="working-with-files.html"><a href="working-with-files.html#titanic-dataset"><i class="fa fa-check"></i><b>4.2.1</b> Titanic dataset</a></li>
<li class="chapter" data-level="4.2.2" data-path="working-with-files.html"><a href="working-with-files.html#ibm-employee-attrition-data"><i class="fa fa-check"></i><b>4.2.2</b> IBM Employee Attrition data</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="working-with-files.html"><a href="working-with-files.html#code-challenge-3"><i class="fa fa-check"></i><b>4.3</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="dpyr-library.html"><a href="dpyr-library.html"><i class="fa fa-check"></i><b>5</b> Dpyr library</a>
<ul>
<li class="chapter" data-level="5.1" data-path="dpyr-library.html"><a href="dpyr-library.html#dplyr-verbs"><i class="fa fa-check"></i><b>5.1</b> Dplyr Verbs</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="dpyr-library.html"><a href="dpyr-library.html#select"><i class="fa fa-check"></i><b>5.1.1</b> select</a></li>
<li class="chapter" data-level="5.1.2" data-path="dpyr-library.html"><a href="dpyr-library.html#filter"><i class="fa fa-check"></i><b>5.1.2</b> filter</a></li>
<li class="chapter" data-level="5.1.3" data-path="dpyr-library.html"><a href="dpyr-library.html#arrange"><i class="fa fa-check"></i><b>5.1.3</b> arrange</a></li>
<li class="chapter" data-level="5.1.4" data-path="dpyr-library.html"><a href="dpyr-library.html#mutate"><i class="fa fa-check"></i><b>5.1.4</b> mutate</a></li>
<li class="chapter" data-level="5.1.5" data-path="dpyr-library.html"><a href="dpyr-library.html#group_by"><i class="fa fa-check"></i><b>5.1.5</b> group_by</a></li>
<li class="chapter" data-level="5.1.6" data-path="dpyr-library.html"><a href="dpyr-library.html#summarise"><i class="fa fa-check"></i><b>5.1.6</b> Summarise</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="dpyr-library.html"><a href="dpyr-library.html#chaining"><i class="fa fa-check"></i><b>5.2</b> Chaining</a></li>
<li class="chapter" data-level="5.3" data-path="dpyr-library.html"><a href="dpyr-library.html#application-of-dplyr-on-microbiology"><i class="fa fa-check"></i><b>5.3</b> Application of Dplyr on Microbiology</a></li>
<li class="chapter" data-level="5.4" data-path="dpyr-library.html"><a href="dpyr-library.html#code-challenge-4"><i class="fa fa-check"></i><b>5.4</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="ggplot.html"><a href="ggplot.html"><i class="fa fa-check"></i><b>6</b> GGPLOT</a>
<ul>
<li class="chapter" data-level="6.1" data-path="ggplot.html"><a href="ggplot.html#five-key-steps-to-ggplot"><i class="fa fa-check"></i><b>6.1</b> Five key steps to ggplot</a></li>
<li class="chapter" data-level="6.2" data-path="ggplot.html"><a href="ggplot.html#prepare-the-data-for-plotting---data-assessment"><i class="fa fa-check"></i><b>6.2</b> Prepare the data for plotting - Data Assessment</a></li>
<li class="chapter" data-level="6.3" data-path="ggplot.html"><a href="ggplot.html#plotting"><i class="fa fa-check"></i><b>6.3</b> Plotting</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="ggplot.html"><a href="ggplot.html#base-set-up"><i class="fa fa-check"></i><b>6.3.1</b> Base set up</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="ggplot.html"><a href="ggplot.html#working-with-inbuilt-iris-data"><i class="fa fa-check"></i><b>6.4</b> Working with inbuilt iris data</a></li>
<li class="chapter" data-level="6.5" data-path="ggplot.html"><a href="ggplot.html#code-challenge-5"><i class="fa fa-check"></i><b>6.5</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="bioconductor.html"><a href="bioconductor.html"><i class="fa fa-check"></i><b>7</b> Bioconductor</a>
<ul>
<li class="chapter" data-level="7.1" data-path="bioconductor.html"><a href="bioconductor.html#installing-biocmanager-package"><i class="fa fa-check"></i><b>7.1</b> Installing BiocManager Package</a></li>
<li class="chapter" data-level="7.2" data-path="bioconductor.html"><a href="bioconductor.html#searching-and-analyzing-the-strings"><i class="fa fa-check"></i><b>7.2</b> Searching and Analyzing the strings</a></li>
<li class="chapter" data-level="7.3" data-path="bioconductor.html"><a href="bioconductor.html#using-seqinr"><i class="fa fa-check"></i><b>7.3</b> Using seqinr</a></li>
<li class="chapter" data-level="7.4" data-path="bioconductor.html"><a href="bioconductor.html#using-short-read"><i class="fa fa-check"></i><b>7.4</b> Using Short Read</a></li>
<li class="chapter" data-level="7.5" data-path="bioconductor.html"><a href="bioconductor.html#genomic-ranges"><i class="fa fa-check"></i><b>7.5</b> Genomic Ranges</a></li>
<li class="chapter" data-level="7.6" data-path="bioconductor.html"><a href="bioconductor.html#using-deseq-package"><i class="fa fa-check"></i><b>7.6</b> Using `DESeq package</a></li>
<li class="chapter" data-level="7.7" data-path="bioconductor.html"><a href="bioconductor.html#network-analysis-of-genomic-data"><i class="fa fa-check"></i><b>7.7</b> Network Analysis of Genomic Data</a></li>
<li class="chapter" data-level="7.8" data-path="bioconductor.html"><a href="bioconductor.html#code-challenge-6"><i class="fa fa-check"></i><b>7.8</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">R Course</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="bioconductor" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> Bioconductor<a href="bioconductor.html#bioconductor" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Bioconductor is a repository for high-quality, peer-reviewed software packages for the analysis and comprehension of high-throughput genomic data. It contained specific tools that handle, analyze and visualize data from high-throughput assays for instance ChIP-Seq, single-cell RNA-Seq and RNA-Seq. It provides data structures such as <code>ExpressionSet</code>,<code>SummarizedExperiment</code>, and <code>GenomicRanges</code>that store and manipulate large scale genomic data. It also consists over 2000 packages just for data pre-processing, normalization, data analysis and visualization. Here are some of the popular packages in <code>BioConductor</code> has to offer;</p>
<ul>
<li><p><strong><code>Biostrings</code></strong>: For efficient manipulation of large biological sequences.</p></li>
<li><p><strong><code>DESeq2</code></strong>: For differential gene expression analysis based on count data.</p></li>
<li><p><strong><code>GenomicRanges</code></strong>: For representing and manipulating genomic intervals and variables defined along a genome.</p></li>
<li><p><strong><code>AnnotationDbi</code></strong>: For handling and querying biological annotation databases.</p></li>
<li><p><strong><code>edgeR</code></strong>: Another popular package for differential expression analysis, particularly for RNA-Seq data.</p></li>
</ul>
<div id="installing-biocmanager-package" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Installing BiocManager Package<a href="bioconductor.html#installing-biocmanager-package" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To get started with Bioconductor, you need to install the <code>BiocManager</code> package, which facilitates the installation and management of Bioconductor packages. You can find detailed installation instructions in the <a href="https://cran.r-project.org/web/packages/BiocManager/vignettes/BiocManager.html">official documentation</a>.</p>
<p>Load the required libraries</p>
<div class="sourceCode" id="cb490"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb490-1"><a href="bioconductor.html#cb490-1" tabindex="-1"></a><span class="fu">library</span>(Biostrings)</span></code></pre></div>
<p>After loading the necessary libraries, you can read sequence data into R. For instance, if you have sequence data stored in a multifasta file named multifasta_seq.txt, you can use the readDNAStringSet() function.</p>
<div class="sourceCode" id="cb491"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb491-1"><a href="bioconductor.html#cb491-1" tabindex="-1"></a>seq <span class="ot">=</span> <span class="fu">readDNAStringSet</span>(<span class="st">&quot;data/multifasta_seq.txt&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in .Call2(&quot;fasta_index&quot;, filexp_list, nrec, skip, seek.first.rec, :
## reading FASTA file data/multifasta_seq.txt: ignored 2545 invalid one-letter
## sequence codes</code></pre>
<div class="sourceCode" id="cb493"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb493-1"><a href="bioconductor.html#cb493-1" tabindex="-1"></a>seq</span></code></pre></div>
<pre><code>## DNAStringSet object of length 6:
##     width seq                                               names               
## [1]   844 MVVVSSCVNTTRTAYTNSTRGVY...KGCCSCGSCCKDDDSVKGVKHYT Spike_SARS_CoV2_2019
## [2]   842 MTTSGSDDRCTTDDVANYTHTSS...KGACSCGSCCKDDDSVKGVKHYT Spike_SARS_CoV
## [3]   841 MVVVSSCVNTTRTAYTNSSTRGV...KGCCSCGSCCKDDDSVKGVKHYT Spike_Bat_coronav...
## [4]   834 MVVVSSCVNTTRTGGYTNSSTRG...KGCCSCGSCCKDDDSVKGVKHYT Spike_Pangolin_co...
## [5]   846 MHAVNSCVNTGRAASTNSSRGVY...KGCCSCGSCCKDDDSVKGVKHYT Spike_Pangolin_co...
## [6]   835 MKVVATVSSYTKCDDDRTANTSS...KGACSCGSCCKDDDSVKGVKHYT Spike_WIV1_Bat_Co...</code></pre>
<p>Each genomic sequence can be viewed separately by indexing</p>
<div class="sourceCode" id="cb495"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb495-1"><a href="bioconductor.html#cb495-1" tabindex="-1"></a>seq[[<span class="dv">1</span>]] <span class="co">#first genomic sequence</span></span></code></pre></div>
<pre><code>## 844-letter DNAString object
## seq: MVVVSSCVNTTRTAYTNSTRGVYYDKVRSSVHSTDS...MVTMCCMTSCCSCKGCCSCGSCCKDDDSVKGVKHYT</code></pre>
<div class="sourceCode" id="cb497"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb497-1"><a href="bioconductor.html#cb497-1" tabindex="-1"></a>seq[[<span class="dv">2</span>]] <span class="co"># second genomic sequence</span></span></code></pre></div>
<pre><code>## 842-letter DNAString object
## seq: MTTSGSDDRCTTDDVANYTHTSSMRGVYYDRSDTYT...VMVTCCMTSCCSCKGACSCGSCCKDDDSVKGVKHYT</code></pre>
<p>Convert to character to inspect its content.</p>
<div class="sourceCode" id="cb499"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb499-1"><a href="bioconductor.html#cb499-1" tabindex="-1"></a><span class="fu">as.character</span>(seq[[<span class="dv">1</span>]]) </span></code></pre></div>
<pre><code>## [1] &quot;MVVVSSCVNTTRTAYTNSTRGVYYDKVRSSVHSTDSNVTWHAHVSGTNGTKRDNVNDGVYASTKSNRGWGTTDSKTSVNNATNVVKVCCNDGVYYHKNNKSWMSRVYSSANNCTYVSMDGKGNKNRVKNDGYKYSKHTNVRDGSAVDGNTRTAHRSYTGDSSSGWTAGAAAYYVGYRTKYNNGTTDAVDCADSTKCTKSTVKGYTSNRVTSVRNTNCGVNATRASVYAWNRKRSNCVADYSVYNSASSTKCYGVSTKNDCTNVYADSVRGDVRAGTGKADYNYKDDTGCVAWNSNNDSKVGGNYNYYRRKSNKRDSTYAGSTCNGVGNCYSYGTNGVGYYRVVVSHAATVCGKKSTNVKNKCVNNNGTGTGVTSNKKGRDADTTDAVRDTDTCSGGVSVTGTNTSNVAVYDVNCTVVAHADTTWRVYSTGSNVTRAGCGAHVNNSYCDGAGCASYTTNSRRARSVASSAYTMSGANSVAYSNNSATNTSVTTVSMTKTSVDCTMYCGDSTCSNYGSCTNRATGAVDKNTVAVKYKTKDGGNSDSKSKRSDNKVTADAGKYGDCGDAARDCAKNGTVTDMAYTSAAGTTSGWTGAGAAAMMAYRNGGVTNVYNKANNSAGKDSSSTASAGKDVVNNAANTVKSSNGASSVNDSRDKVAVDRTGRSTYVTRAARASANAATKMSCVGSKRVDCGKGYHMSSAHGVVHVTYVAKNTTAACHDGKAHRGVVSNGTHWVTRNYTTDNTVSGNCDVVGVNNTVYDDSKDKYKNHTSDVDGDSGNASVVNKDRNVAKNNSDGKYYKWWYWGAGAVMVTMCCMTSCCSCKGCCSCGSCCKDDDSVKGVKHYT&quot;</code></pre>
<p>The <code>length</code> function from base R is used to count the number of genomic sequences.</p>
<div class="sourceCode" id="cb501"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb501-1"><a href="bioconductor.html#cb501-1" tabindex="-1"></a><span class="fu">length</span>(seq)</span></code></pre></div>
<pre><code>## [1] 6</code></pre>
<ul>
<li>Names of the sequences</li>
</ul>
<div class="sourceCode" id="cb503"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb503-1"><a href="bioconductor.html#cb503-1" tabindex="-1"></a><span class="fu">names</span>(seq)</span></code></pre></div>
<pre><code>## [1] &quot;Spike_SARS_CoV2_2019&quot;          &quot;Spike_SARS_CoV&quot;               
## [3] &quot;Spike_Bat_coronavirus_RaTG13&quot;  &quot;Spike_Pangolin_coronavirus_GX&quot;
## [5] &quot;Spike_Pangolin_coronavirus_GD&quot; &quot;Spike_WIV1_Bat_Coronovirus&quot;</code></pre>
<ul>
<li>view the first 6 sequences</li>
</ul>
<div class="sourceCode" id="cb505"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb505-1"><a href="bioconductor.html#cb505-1" tabindex="-1"></a><span class="fu">head</span>(seq)</span></code></pre></div>
<pre><code>## DNAStringSet object of length 6:
##     width seq                                               names               
## [1]   844 MVVVSSCVNTTRTAYTNSTRGVY...KGCCSCGSCCKDDDSVKGVKHYT Spike_SARS_CoV2_2019
## [2]   842 MTTSGSDDRCTTDDVANYTHTSS...KGACSCGSCCKDDDSVKGVKHYT Spike_SARS_CoV
## [3]   841 MVVVSSCVNTTRTAYTNSSTRGV...KGCCSCGSCCKDDDSVKGVKHYT Spike_Bat_coronav...
## [4]   834 MVVVSSCVNTTRTGGYTNSSTRG...KGCCSCGSCCKDDDSVKGVKHYT Spike_Pangolin_co...
## [5]   846 MHAVNSCVNTGRAASTNSSRGVY...KGCCSCGSCCKDDDSVKGVKHYT Spike_Pangolin_co...
## [6]   835 MKVVATVSSYTKCDDDRTANTSS...KGACSCGSCCKDDDSVKGVKHYT Spike_WIV1_Bat_Co...</code></pre>
<ul>
<li>Checking the reverse compliment of sequence</li>
</ul>
<div class="sourceCode" id="cb507"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb507-1"><a href="bioconductor.html#cb507-1" tabindex="-1"></a>seq_rc <span class="ot">=</span> <span class="fu">reverseComplement</span>(seq[[<span class="dv">1</span>]])</span>
<span id="cb507-2"><a href="bioconductor.html#cb507-2" tabindex="-1"></a><span class="fu">head</span>(seq_rc)</span></code></pre></div>
<pre><code>## 6-letter DNAString object
## seq: ARDMBC</code></pre>
<div class="sourceCode" id="cb509"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb509-1"><a href="bioconductor.html#cb509-1" tabindex="-1"></a><span class="fu">tail</span>(seq_rc)</span></code></pre></div>
<pre><code>## 6-letter DNAString object
## seq: SSBBBK</code></pre>
<div class="sourceCode" id="cb511"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb511-1"><a href="bioconductor.html#cb511-1" tabindex="-1"></a><span class="fu">as.character</span>(seq_rc)</span></code></pre></div>
<pre><code>## [1] &quot;ARDMBCMBSHHHMGGSCGSGGCMGSGGSAKGGKABKBTCTCWRWWMRRMCHSNNMTBNYHMNBBSTNCSHCHBHSADNMRMHMSHHRBANNBCBBHGNCSBANHAARNYABWDACNSBBCYDTMCHDGTTAANMTBRABDBBCDTSSKDRCMCGHBYMSCBGSKMATTNTSTYTTYABRASYCAYHBTBMHYSHNBSSTCNSSMBANTTNNBBHMCTSTASSSHMCTSNNTMNRBNABCCNYRTKKTTTCTCAWCSAACTTSARTKHABACNMTGHYTTHCGHCRMCTHTABMNHSYMSMSHSNCCHMAMRMBTBANMHBTCATYNAGSCRNSGASHCGRKAGHBSAMAKSBAABSANATSNNSRTBSNTCSKARTSSTBSYTYYSNAARSTGCTCHGRSNNBDTCGCTYABNSCASRBYWAAHTDTBBAGNBHRBTBNSANACABSBCCSGAHAHYBTHAAHTHYCMMNSABCACACNNNBGMNMBNASMMCGBATTDSBBBYRRCBCNACRSRGNCBCNGASCTRASHYMNSMYYRRNRNCCBMSHNNSNWTBGCAHHMRNRHTMCACTYBHCYBSHTRBNAGHNMASBCRGMASSTSNRBSRHTBGNSYMYNWTRBSTYATNBCGNANYBSABYNSARCMBASMAGMASHTGHBTHAACNNRMAYRCBRRTTTCTAWCSSSHCARSYDTAYANCHBTSCHYBNADMSRMRCHNMBYNMNCMCHKSBRAGNNTSSRBYSKWSMNNMDRRBCHNGGBMBBNATNNBSAMSHAACWCYNSMASTRBCHNBNHYMACNACSBDTDWABNSHASDBSSYBMHRRBCYASNARTAYAANBGSSBBBK&quot;</code></pre>
<ul>
<li>Sequence composition -default - calculate the DNA composition</li>
</ul>
<div class="sourceCode" id="cb513"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb513-1"><a href="bioconductor.html#cb513-1" tabindex="-1"></a>composition<span class="ot">=</span><span class="fu">alphabetFrequency</span>(seq)</span>
<span id="cb513-2"><a href="bioconductor.html#cb513-2" tabindex="-1"></a><span class="fu">print</span>(composition)</span></code></pre></div>
<pre><code>##       A  C  G   T  M  R  W   S  Y  K  V  H  D B  N - + .
## [1,] 79 40 82  97 14 42 12  99 54 61 97 17 62 0 88 0 0 0
## [2,] 84 39 78 100 20 39 11  96 54 60 91 15 74 0 81 0 0 0
## [3,] 80 40 82  98 14 39 12  97 55 63 94 18 64 0 85 0 0 0
## [4,] 76 40 85  98 15 41 12  94 55 61 97 17 61 0 82 0 0 0
## [5,] 82 40 81  99 15 40 12 100 60 58 95 14 60 0 90 0 0 0
## [6,] 82 39 77  93 17 42 11  98 50 60 93 15 72 0 86 0 0 0</code></pre>
<ul>
<li>Specific composition - lets say we wanna look for A, T, G, C and N</li>
</ul>
<div class="sourceCode" id="cb515"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb515-1"><a href="bioconductor.html#cb515-1" tabindex="-1"></a>composition1<span class="ot">=</span><span class="fu">letterFrequency</span>(seq, <span class="at">letters=</span><span class="fu">c</span>(<span class="st">&quot;A&quot;</span>,<span class="st">&quot;T&quot;</span>, <span class="st">&quot;G&quot;</span>, <span class="st">&quot;C&quot;</span>, <span class="st">&quot;N&quot;</span>))</span>
<span id="cb515-2"><a href="bioconductor.html#cb515-2" tabindex="-1"></a><span class="fu">print</span>(composition1)</span></code></pre></div>
<pre><code>##       A   T  G  C  N
## [1,] 79  97 82 40 88
## [2,] 84 100 78 39 81
## [3,] 80  98 82 40 85
## [4,] 76  98 85 40 82
## [5,] 82  99 81 40 90
## [6,] 82  93 77 39 86</code></pre>
<p>Computing the GC Content</p>
<div class="sourceCode" id="cb517"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb517-1"><a href="bioconductor.html#cb517-1" tabindex="-1"></a>gc_content <span class="ot">&lt;-</span> <span class="fu">letterFrequency</span>(seq, <span class="at">letters=</span><span class="st">&quot;GC&quot;</span>, <span class="at">as.prob=</span><span class="cn">TRUE</span>)</span>
<span id="cb517-2"><a href="bioconductor.html#cb517-2" tabindex="-1"></a>gc_content</span></code></pre></div>
<pre><code>##            G|C
## [1,] 0.1445498
## [2,] 0.1389549
## [3,] 0.1450654
## [4,] 0.1498801
## [5,] 0.1430260
## [6,] 0.1389222</code></pre>
<p>Calculate the mean of Gc content</p>
<div class="sourceCode" id="cb519"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb519-1"><a href="bioconductor.html#cb519-1" tabindex="-1"></a><span class="fu">mean</span>(gc_content)</span></code></pre></div>
<pre><code>## [1] 0.1433997</code></pre>
<ul>
<li>Standard deviation</li>
</ul>
<div class="sourceCode" id="cb521"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb521-1"><a href="bioconductor.html#cb521-1" tabindex="-1"></a><span class="fu">sd</span>(gc_content)</span></code></pre></div>
<pre><code>## [1] 0.004148188</code></pre>
<div class="sourceCode" id="cb523"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb523-1"><a href="bioconductor.html#cb523-1" tabindex="-1"></a><span class="fu">range</span>(gc_content)</span></code></pre></div>
<pre><code>## [1] 0.1389222 0.1498801</code></pre>
<ul>
<li>Which GC has the maximum content</li>
</ul>
<div class="sourceCode" id="cb525"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb525-1"><a href="bioconductor.html#cb525-1" tabindex="-1"></a>seq[<span class="fu">which</span>(gc_content<span class="sc">==</span><span class="fu">max</span>(gc_content))]</span></code></pre></div>
<pre><code>## DNAStringSet object of length 1:
##     width seq                                               names               
## [1]   834 MVVVSSCVNTTRTGGYTNSSTRG...KGCCSCGSCCKDDDSVKGVKHYT Spike_Pangolin_co...</code></pre>
</div>
<div id="searching-and-analyzing-the-strings" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Searching and Analyzing the strings<a href="bioconductor.html#searching-and-analyzing-the-strings" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A biologist can search for specific substrings within the sequences and analyze their positions and characteristics for instance;</p>
<ul>
<li>Search for substrings- in this case we search for <code>"GAATTC"</code> substring.</li>
</ul>
<div class="sourceCode" id="cb527"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb527-1"><a href="bioconductor.html#cb527-1" tabindex="-1"></a>ECoR1<span class="ot">=</span><span class="fu">vcountPattern</span>(<span class="st">&quot;GAATTC&quot;</span>, seq)</span>
<span id="cb527-2"><a href="bioconductor.html#cb527-2" tabindex="-1"></a><span class="fu">print</span>(ECoR1)</span></code></pre></div>
<pre><code>## [1] 0 0 0 0 0 0</code></pre>
<div class="sourceCode" id="cb529"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb529-1"><a href="bioconductor.html#cb529-1" tabindex="-1"></a>HindIII <span class="ot">=</span> <span class="fu">vcountPattern</span>(<span class="st">&quot;AAGTT&quot;</span>, seq)</span>
<span id="cb529-2"><a href="bioconductor.html#cb529-2" tabindex="-1"></a><span class="fu">print</span>(HindIII)</span></code></pre></div>
<pre><code>## [1] 1 0 1 1 1 0</code></pre>
<ul>
<li>Finding the position of substrings</li>
</ul>
<div class="sourceCode" id="cb531"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb531-1"><a href="bioconductor.html#cb531-1" tabindex="-1"></a>HindIII_pos<span class="ot">=</span><span class="fu">vmatchPattern</span>(<span class="st">&quot;AAGCT&quot;</span>, seq)</span>
<span id="cb531-2"><a href="bioconductor.html#cb531-2" tabindex="-1"></a><span class="fu">print</span>(HindIII_pos)</span></code></pre></div>
<pre><code>## MIndex object of length 6
## $Spike_SARS_CoV2_2019
## IRanges object with 0 ranges and 0 metadata columns:
##        start       end     width
##    &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
## 
## $Spike_SARS_CoV
## IRanges object with 0 ranges and 0 metadata columns:
##        start       end     width
##    &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
## 
## $Spike_Bat_coronavirus_RaTG13
## IRanges object with 0 ranges and 0 metadata columns:
##        start       end     width
##    &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
## 
## ...
## &lt;3 more elements&gt;</code></pre>
<ul>
<li>Translate</li>
</ul>
<div class="sourceCode" id="cb533"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb533-1"><a href="bioconductor.html#cb533-1" tabindex="-1"></a><span class="co"># translate(seq[[2]])</span></span>
<span id="cb533-2"><a href="bioconductor.html#cb533-2" tabindex="-1"></a><span class="co">#Check functions available</span></span>
<span id="cb533-3"><a href="bioconductor.html#cb533-3" tabindex="-1"></a><span class="fu">methods</span>(<span class="at">class =</span> <span class="st">&quot;DNAStringSet&quot;</span>)</span></code></pre></div>
<pre><code>##   [1] !                                     
##   [2] !=                                    
##   [3] [                                     
##   [4] [[                                    
##   [5] [[&lt;-                                  
##   [6] [&lt;-                                   
##   [7] %in%                                  
##   [8] &lt;                                     
##   [9] &lt;=                                    
##  [10] ==                                    
##  [11] &gt;                                     
##  [12] &gt;=                                    
##  [13] $                                     
##  [14] $&lt;-                                   
##  [15] aggregate                             
##  [16] alphabetFrequency                     
##  [17] anyDuplicated                         
##  [18] anyNA                                 
##  [19] append                                
##  [20] as.character                          
##  [21] as.complex                            
##  [22] as.data.frame                         
##  [23] as.env                                
##  [24] as.factor                             
##  [25] as.integer                            
##  [26] as.list                               
##  [27] as.logical                            
##  [28] as.matrix                             
##  [29] as.numeric                            
##  [30] as.raw                                
##  [31] as.vector                             
##  [32] bindROWS                              
##  [33] by                                    
##  [34] c                                     
##  [35] cbind                                 
##  [36] chartr                                
##  [37] coerce                                
##  [38] compact                               
##  [39] compareStrings                        
##  [40] complement                            
##  [41] consensusMatrix                       
##  [42] consensusString                       
##  [43] countOverlaps                         
##  [44] countPattern                          
##  [45] countPDict                            
##  [46] dinucleotideFrequencyTest             
##  [47] do.call                               
##  [48] droplevels                            
##  [49] duplicated                            
##  [50] elementMetadata                       
##  [51] elementMetadata&lt;-                     
##  [52] elementNROWS                          
##  [53] elementType                           
##  [54] eval                                  
##  [55] expand                                
##  [56] expand.grid                           
##  [57] extractAt                             
##  [58] extractROWS                           
##  [59] FactorToClass                         
##  [60] Filter                                
##  [61] findOverlaps                          
##  [62] getListElement                        
##  [63] hasOnlyBaseLetters                    
##  [64] head                                  
##  [65] ifelse2                               
##  [66] intersect                             
##  [67] is.na                                 
##  [68] is.unsorted                           
##  [69] isEmpty                               
##  [70] isMatchingEndingAt                    
##  [71] isMatchingStartingAt                  
##  [72] lapply                                
##  [73] length                                
##  [74] lengths                               
##  [75] letterFrequency                       
##  [76] make_XStringSet_from_strings          
##  [77] match                                 
##  [78] matchPattern                          
##  [79] matchPDict                            
##  [80] mcols                                 
##  [81] mcols&lt;-                               
##  [82] merge                                 
##  [83] mergeROWS                             
##  [84] metadata                              
##  [85] metadata&lt;-                            
##  [86] mstack                                
##  [87] names                                 
##  [88] names&lt;-                               
##  [89] nchar                                 
##  [90] neditEndingAt                         
##  [91] neditStartingAt                       
##  [92] normalizeSingleBracketReplacementValue
##  [93] nucleotideFrequencyAt                 
##  [94] oligonucleotideFrequency              
##  [95] order                                 
##  [96] overlapsAny                           
##  [97] PairwiseAlignments                    
##  [98] PairwiseAlignmentsSingleSubject       
##  [99] palindromeArmLength                   
## [100] parallel_slot_names                   
## [101] parallelVectorNames                   
## [102] pcompare                              
## [103] pcompareRecursively                   
## [104] PDict                                 
## [105] PWM                                   
## [106] rank                                  
## [107] rbind                                 
## [108] Reduce                                
## [109] relist                                
## [110] relistToClass                         
## [111] rename                                
## [112] rep                                   
## [113] rep.int                               
## [114] replaceAt                             
## [115] replaceLetterAt                       
## [116] replaceROWS                           
## [117] rev                                   
## [118] revElements                           
## [119] reverse                               
## [120] reverseComplement                     
## [121] sapply                                
## [122] selfmatch                             
## [123] seqinfo                               
## [124] seqinfo&lt;-                             
## [125] seqlevelsInUse                        
## [126] seqtype                               
## [127] seqtype&lt;-                             
## [128] setdiff                               
## [129] setequal                              
## [130] setListElement                        
## [131] shiftApply                            
## [132] show                                  
## [133] showAsCell                            
## [134] sort                                  
## [135] split                                 
## [136] split&lt;-                               
## [137] stack                                 
## [138] stringDist                            
## [139] strsplit                              
## [140] subseq                                
## [141] subseq&lt;-                              
## [142] subset                                
## [143] subsetByOverlaps                      
## [144] summary                               
## [145] table                                 
## [146] tail                                  
## [147] tapply                                
## [148] threebands                            
## [149] toString                              
## [150] transform                             
## [151] translate                             
## [152] trimLRPatterns                        
## [153] twoWayAlphabetFrequency               
## [154] union                                 
## [155] unique                                
## [156] uniqueLetters                         
## [157] unlist                                
## [158] unname                                
## [159] unsplit                               
## [160] unstrsplit                            
## [161] updateObject                          
## [162] values                                
## [163] values&lt;-                              
## [164] vcountPattern                         
## [165] vcountPDict                           
## [166] vmatchPattern                         
## [167] vwhichPDict                           
## [168] which.isMatchingEndingAt              
## [169] which.isMatchingStartingAt            
## [170] whichPDict                            
## [171] width                                 
## [172] window                                
## [173] window&lt;-                              
## [174] windows                               
## [175] with                                  
## [176] within                                
## [177] xtabs                                 
## [178] xtfrm                                 
## [179] xvcopy                                
## [180] zipdown                               
## see &#39;?methods&#39; for accessing help and source code</code></pre>
</div>
<div id="using-seqinr" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Using seqinr<a href="bioconductor.html#using-seqinr" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Another useful library in sequence analysis is <code>seqinr</code>.
After installing the library by invoking <code>install.packages("seqinr")</code> to the console, load it to the workspace</p>
<div class="sourceCode" id="cb535"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb535-1"><a href="bioconductor.html#cb535-1" tabindex="-1"></a><span class="fu">library</span>(seqinr)</span></code></pre></div>
<p>Reading the fasta file containing the protein sequence using the <code>read.fasta</code> function.</p>
<div class="sourceCode" id="cb536"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb536-1"><a href="bioconductor.html#cb536-1" tabindex="-1"></a>proseq <span class="ot">=</span> <span class="fu">read.fasta</span>(<span class="at">file=</span><span class="st">&quot;data/spike_multifasta.txt&quot;</span>, <span class="at">seqtype=</span><span class="st">&quot;AA&quot;</span>)</span>
<span id="cb536-2"><a href="bioconductor.html#cb536-2" tabindex="-1"></a><span class="fu">str</span>(proseq)</span></code></pre></div>
<pre><code>## List of 6
##  $ Spike_SARS_CoV2_2019         : &#39;SeqFastaAA&#39; chr [1:1273] &quot;M&quot; &quot;F&quot; &quot;V&quot; &quot;F&quot; ...
##   ..- attr(*, &quot;name&quot;)= chr &quot;Spike_SARS_CoV2_2019&quot;
##   ..- attr(*, &quot;Annot&quot;)= chr &quot;&gt;Spike_SARS_CoV2_2019&quot;
##  $ Spike_SARS_CoV               : &#39;SeqFastaAA&#39; chr [1:1255] &quot;M&quot; &quot;F&quot; &quot;I&quot; &quot;F&quot; ...
##   ..- attr(*, &quot;name&quot;)= chr &quot;Spike_SARS_CoV&quot;
##   ..- attr(*, &quot;Annot&quot;)= chr &quot;&gt;Spike_SARS_CoV&quot;
##  $ Spike_Bat_coronavirus_RaTG13 : &#39;SeqFastaAA&#39; chr [1:1269] &quot;M&quot; &quot;F&quot; &quot;V&quot; &quot;F&quot; ...
##   ..- attr(*, &quot;name&quot;)= chr &quot;Spike_Bat_coronavirus_RaTG13&quot;
##   ..- attr(*, &quot;Annot&quot;)= chr &quot;&gt;Spike_Bat_coronavirus_RaTG13&quot;
##  $ Spike_Pangolin_coronavirus_GX: &#39;SeqFastaAA&#39; chr [1:1269] &quot;M&quot; &quot;F&quot; &quot;V&quot; &quot;F&quot; ...
##   ..- attr(*, &quot;name&quot;)= chr &quot;Spike_Pangolin_coronavirus_GX&quot;
##   ..- attr(*, &quot;Annot&quot;)= chr &quot;&gt;Spike_Pangolin_coronavirus_GX&quot;
##  $ Spike_Pangolin_coronavirus_GD: &#39;SeqFastaAA&#39; chr [1:1265] &quot;M&quot; &quot;L&quot; &quot;F&quot; &quot;F&quot; ...
##   ..- attr(*, &quot;name&quot;)= chr &quot;Spike_Pangolin_coronavirus_GD&quot;
##   ..- attr(*, &quot;Annot&quot;)= chr &quot;&gt;Spike_Pangolin_coronavirus_GD&quot;
##  $ Spike_WIV1_Bat_Coronovirus   : &#39;SeqFastaAA&#39; chr [1:1256] &quot;M&quot; &quot;K&quot; &quot;L&quot; &quot;L&quot; ...
##   ..- attr(*, &quot;name&quot;)= chr &quot;Spike_WIV1_Bat_Coronovirus&quot;
##   ..- attr(*, &quot;Annot&quot;)= chr &quot;&gt;Spike_WIV1_Bat_Coronovirus&quot;</code></pre>
<p>Count the number of sequences</p>
<div class="sourceCode" id="cb538"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb538-1"><a href="bioconductor.html#cb538-1" tabindex="-1"></a>num_seq <span class="ot">=</span> <span class="fu">length</span>(proseq)</span>
<span id="cb538-2"><a href="bioconductor.html#cb538-2" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&quot;Number of sequences = &quot;</span>, num_seq))</span></code></pre></div>
<pre><code>## [1] &quot;Number of sequences =  6&quot;</code></pre>
<ul>
<li>Names of the sequences</li>
</ul>
<div class="sourceCode" id="cb540"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb540-1"><a href="bioconductor.html#cb540-1" tabindex="-1"></a>names<span class="ot">=</span><span class="fu">getName</span>(proseq)</span>
<span id="cb540-2"><a href="bioconductor.html#cb540-2" tabindex="-1"></a>names </span></code></pre></div>
<pre><code>## [1] &quot;Spike_SARS_CoV2_2019&quot;          &quot;Spike_SARS_CoV&quot;               
## [3] &quot;Spike_Bat_coronavirus_RaTG13&quot;  &quot;Spike_Pangolin_coronavirus_GX&quot;
## [5] &quot;Spike_Pangolin_coronavirus_GD&quot; &quot;Spike_WIV1_Bat_Coronovirus&quot;</code></pre>
<ul>
<li>Annotation of sequences</li>
</ul>
<div class="sourceCode" id="cb542"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb542-1"><a href="bioconductor.html#cb542-1" tabindex="-1"></a>annotation <span class="ot">&lt;-</span> <span class="fu">getAnnot</span>(proseq)</span>
<span id="cb542-2"><a href="bioconductor.html#cb542-2" tabindex="-1"></a>annotation</span></code></pre></div>
<pre><code>## [[1]]
## [1] &quot;&gt;Spike_SARS_CoV2_2019&quot;
## 
## [[2]]
## [1] &quot;&gt;Spike_SARS_CoV&quot;
## 
## [[3]]
## [1] &quot;&gt;Spike_Bat_coronavirus_RaTG13&quot;
## 
## [[4]]
## [1] &quot;&gt;Spike_Pangolin_coronavirus_GX&quot;
## 
## [[5]]
## [1] &quot;&gt;Spike_Pangolin_coronavirus_GD&quot;
## 
## [[6]]
## [1] &quot;&gt;Spike_WIV1_Bat_Coronovirus&quot;</code></pre>
<ul>
<li>Length of individual sequences</li>
</ul>
<div class="sourceCode" id="cb544"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb544-1"><a href="bioconductor.html#cb544-1" tabindex="-1"></a>len_seq <span class="ot">=</span> <span class="fu">getLength</span>(proseq)</span>
<span id="cb544-2"><a href="bioconductor.html#cb544-2" tabindex="-1"></a>len_seq</span></code></pre></div>
<pre><code>## [1] 1273 1255 1269 1269 1265 1256</code></pre>
<ul>
<li>Protein statistics</li>
</ul>
<div class="sourceCode" id="cb546"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb546-1"><a href="bioconductor.html#cb546-1" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">AAstat</span>(proseq[[<span class="dv">1</span>]]))</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-266-1.png" width="672" /></p>
<pre><code>## $Compo
## 
##   *   A   C   D   E   F   G   H   I   K   L   M   N   P   Q   R   S   T   V   W 
##   0  79  40  62  48  77  82  17  76  61 108  14  88  58  62  42  99  97  97  12 
##   Y 
##  54 
## 
## $Prop
## $Prop$Tiny
## [1] 0.3118617
## 
## $Prop$Small
## [1] 0.5514533
## 
## $Prop$Aliphatic
## [1] 0.2207384
## 
## $Prop$Aromatic
## [1] 0.1256874
## 
## $Prop$Non.polar
## [1] 0.5475255
## 
## $Prop$Polar
## [1] 0.4524745
## 
## $Prop$Charged
## [1] 0.1806756
## 
## $Prop$Basic
## [1] 0.09426551
## 
## $Prop$Acidic
## [1] 0.08641005
## 
## 
## $Pi
## [1] 6.236478</code></pre>
<p>It gives the type of proteins</p>
<div class="sourceCode" id="cb548"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb548-1"><a href="bioconductor.html#cb548-1" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>num_seq)</span>
<span id="cb548-2"><a href="bioconductor.html#cb548-2" tabindex="-1"></a>{</span>
<span id="cb548-3"><a href="bioconductor.html#cb548-3" tabindex="-1"></a>  <span class="fu">print</span>(names[i])</span>
<span id="cb548-4"><a href="bioconductor.html#cb548-4" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">AAstat</span>(proseq[[i]]))</span>
<span id="cb548-5"><a href="bioconductor.html#cb548-5" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## [1] &quot;Spike_SARS_CoV2_2019&quot;</code></pre>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-267-1.png" width="672" /></p>
<pre><code>## $Compo
## 
##   *   A   C   D   E   F   G   H   I   K   L   M   N   P   Q   R   S   T   V   W 
##   0  79  40  62  48  77  82  17  76  61 108  14  88  58  62  42  99  97  97  12 
##   Y 
##  54 
## 
## $Prop
## $Prop$Tiny
## [1] 0.3118617
## 
## $Prop$Small
## [1] 0.5514533
## 
## $Prop$Aliphatic
## [1] 0.2207384
## 
## $Prop$Aromatic
## [1] 0.1256874
## 
## $Prop$Non.polar
## [1] 0.5475255
## 
## $Prop$Polar
## [1] 0.4524745
## 
## $Prop$Charged
## [1] 0.1806756
## 
## $Prop$Basic
## [1] 0.09426551
## 
## $Prop$Acidic
## [1] 0.08641005
## 
## 
## $Pi
## [1] 6.236478
## 
## [1] &quot;Spike_SARS_CoV&quot;</code></pre>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-267-2.png" width="672" /></p>
<pre><code>## $Compo
## 
##   *   A   C   D   E   F   G   H   I   K   L   M   N   P   Q   R   S   T   V   W 
##   0  84  39  74  42  83  78  15  77  60  99  20  81  57  55  39  96 100  91  11 
##   Y 
##  54 
## 
## $Prop
## $Prop$Tiny
## [1] 0.3163347
## 
## $Prop$Small
## [1] 0.5577689
## 
## $Prop$Aliphatic
## [1] 0.212749
## 
## $Prop$Aromatic
## [1] 0.1298805
## 
## $Prop$Non.polar
## [1] 0.5521912
## 
## $Prop$Polar
## [1] 0.4478088
## 
## $Prop$Charged
## [1] 0.1832669
## 
## $Prop$Basic
## [1] 0.09083665
## 
## $Prop$Acidic
## [1] 0.09243028
## 
## 
## $Pi
## [1] 5.510607
## 
## [1] &quot;Spike_Bat_coronavirus_RaTG13&quot;</code></pre>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-267-3.png" width="672" /></p>
<pre><code>## $Compo
## 
##   *   A   C   D   E   F   G   H   I   K   L   M   N   P   Q   R   S   T   V   W 
##   0  80  40  64  47  75  82  18  79  63 109  14  85  58  60  39  97  98  94  12 
##   Y 
##  55 
## 
## $Prop
## $Prop$Tiny
## [1] 0.3128448
## 
## $Prop$Small
## [1] 0.5500394
## 
## $Prop$Aliphatic
## [1] 0.2222222
## 
## $Prop$Aromatic
## [1] 0.1260835
## 
## $Prop$Non.polar
## [1] 0.5500394
## 
## $Prop$Polar
## [1] 0.4499606
## 
## $Prop$Charged
## [1] 0.1820331
## 
## $Prop$Basic
## [1] 0.09456265
## 
## $Prop$Acidic
## [1] 0.08747045
## 
## 
## $Pi
## [1] 6.11311
## 
## [1] &quot;Spike_Pangolin_coronavirus_GX&quot;</code></pre>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-267-4.png" width="672" /></p>
<pre><code>## $Compo
## 
##   *   A   C   D   E   F   G   H   I   K   L   M   N   P   Q   R   S   T   V   W 
##   0  76  40  61  48  75  85  17  79  61 113  15  82  56  62  41  94  98  97  12 
##   Y 
##  55 
## 
## $Prop
## $Prop$Tiny
## [1] 0.3096927
## 
## $Prop$Small
## [1] 0.5429472
## 
## $Prop$Aliphatic
## [1] 0.2277384
## 
## $Prop$Aromatic
## [1] 0.1252955
## 
## $Prop$Non.polar
## [1] 0.5539795
## 
## $Prop$Polar
## [1] 0.4444444
## 
## $Prop$Charged
## [1] 0.179669
## 
## $Prop$Basic
## [1] 0.09377463
## 
## $Prop$Acidic
## [1] 0.08589441
## 
## 
## $Pi
## [1] 6.235931
## 
## [1] &quot;Spike_Pangolin_coronavirus_GD&quot;</code></pre>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-267-5.png" width="672" /></p>
<pre><code>## $Compo
## 
##   *   A   C   D   E   F   G   H   I   K   L   M   N   P   Q   R   S   T   V   W 
##   0  82  40  60  48  76  81  14  80  58 103  15  90  54  58  40 100  99  95  12 
##   Y 
##  60 
## 
## $Prop
## $Prop$Tiny
## [1] 0.3177866
## 
## $Prop$Small
## [1] 0.5541502
## 
## $Prop$Aliphatic
## [1] 0.2197628
## 
## $Prop$Aromatic
## [1] 0.1280632
## 
## $Prop$Non.polar
## [1] 0.5517787
## 
## $Prop$Polar
## [1] 0.4482213
## 
## $Prop$Charged
## [1] 0.173913
## 
## $Prop$Basic
## [1] 0.08853755
## 
## $Prop$Acidic
## [1] 0.08537549
## 
## 
## $Pi
## [1] 5.900948
## 
## [1] &quot;Spike_WIV1_Bat_Coronovirus&quot;</code></pre>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-267-6.png" width="672" /></p>
<pre><code>## $Compo
## 
##   *   A   C   D   E   F   G   H   I   K   L   M   N   P   Q   R   S   T   V   W 
##   0  82  39  72  43  83  77  15  77  60 102  17  86  62  54  42  98  93  93  11 
##   Y 
##  50 
## 
## $Prop
## $Prop$Tiny
## [1] 0.3097134
## 
## $Prop$Small
## [1] 0.5589172
## 
## $Prop$Aliphatic
## [1] 0.2165605
## 
## $Prop$Aromatic
## [1] 0.1265924
## 
## $Prop$Non.polar
## [1] 0.5517516
## 
## $Prop$Polar
## [1] 0.4482484
## 
## $Prop$Charged
## [1] 0.1847134
## 
## $Prop$Basic
## [1] 0.09315287
## 
## $Prop$Acidic
## [1] 0.09156051
## 
## 
## $Pi
## [1] 5.739146</code></pre>
<ul>
<li>Show commands in seqinr</li>
</ul>
<div class="sourceCode" id="cb556"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb556-1"><a href="bioconductor.html#cb556-1" tabindex="-1"></a><span class="fu">lseqinr</span>()</span></code></pre></div>
<pre><code>##   [1] &quot;a&quot;                       &quot;aaa&quot;                    
##   [3] &quot;AAstat&quot;                  &quot;acnucclose&quot;             
##   [5] &quot;acnucopen&quot;               &quot;al2bp&quot;                  
##   [7] &quot;alllistranks&quot;            &quot;alr&quot;                    
##   [9] &quot;amb&quot;                     &quot;as.alignment&quot;           
##  [11] &quot;as.matrix.alignment&quot;     &quot;as.SeqAcnucWeb&quot;         
##  [13] &quot;as.SeqFastaAA&quot;           &quot;as.SeqFastadna&quot;         
##  [15] &quot;as.SeqFrag&quot;              &quot;autosocket&quot;             
##  [17] &quot;baselineabif&quot;            &quot;bma&quot;                    
##  [19] &quot;c2s&quot;                     &quot;cai&quot;                    
##  [21] &quot;cfl&quot;                     &quot;choosebank&quot;             
##  [23] &quot;circle&quot;                  &quot;clfcd&quot;                  
##  [25] &quot;clientid&quot;                &quot;closebank&quot;              
##  [27] &quot;col2alpha&quot;               &quot;comp&quot;                   
##  [29] &quot;computePI&quot;               &quot;con&quot;                    
##  [31] &quot;consensus&quot;               &quot;count&quot;                  
##  [33] &quot;countfreelists&quot;          &quot;countsubseqs&quot;           
##  [35] &quot;crelistfromclientdata&quot;   &quot;css&quot;                    
##  [37] &quot;dia.bactgensize&quot;         &quot;dia.db.growth&quot;          
##  [39] &quot;dist.alignment&quot;          &quot;dotchart.uco&quot;           
##  [41] &quot;dotPlot&quot;                 &quot;draw.oriloc&quot;            
##  [43] &quot;draw.rearranged.oriloc&quot;  &quot;draw.recstat&quot;           
##  [45] &quot;exseq&quot;                   &quot;extract.breakpoints&quot;    
##  [47] &quot;extractseqs&quot;             &quot;fastacc&quot;                
##  [49] &quot;gb2fasta&quot;                &quot;gbk2g2&quot;                 
##  [51] &quot;gbk2g2.euk&quot;              &quot;GC&quot;                     
##  [53] &quot;GC1&quot;                     &quot;GC2&quot;                    
##  [55] &quot;GC3&quot;                     &quot;GCpos&quot;                  
##  [57] &quot;get.db.growth&quot;           &quot;getAnnot&quot;               
##  [59] &quot;getAnnot.default&quot;        &quot;getAnnot.list&quot;          
##  [61] &quot;getAnnot.logical&quot;        &quot;getAnnot.qaw&quot;           
##  [63] &quot;getAnnot.SeqAcnucWeb&quot;    &quot;getAnnot.SeqFastaAA&quot;    
##  [65] &quot;getAnnot.SeqFastadna&quot;    &quot;getAttributsocket&quot;      
##  [67] &quot;getFrag&quot;                 &quot;getFrag.character&quot;      
##  [69] &quot;getFrag.default&quot;         &quot;getFrag.list&quot;           
##  [71] &quot;getFrag.logical&quot;         &quot;getFrag.qaw&quot;            
##  [73] &quot;getFrag.SeqAcnucWeb&quot;     &quot;getFrag.SeqFastaAA&quot;     
##  [75] &quot;getFrag.SeqFastadna&quot;     &quot;getFrag.SeqFrag&quot;        
##  [77] &quot;getKeyword&quot;              &quot;getKeyword.default&quot;     
##  [79] &quot;getKeyword.list&quot;         &quot;getKeyword.logical&quot;     
##  [81] &quot;getKeyword.qaw&quot;          &quot;getKeyword.SeqAcnucWeb&quot; 
##  [83] &quot;getLength&quot;               &quot;getLength.character&quot;    
##  [85] &quot;getLength.default&quot;       &quot;getLength.list&quot;         
##  [87] &quot;getLength.logical&quot;       &quot;getLength.qaw&quot;          
##  [89] &quot;getLength.SeqAcnucWeb&quot;   &quot;getLength.SeqFastaAA&quot;   
##  [91] &quot;getLength.SeqFastadna&quot;   &quot;getLength.SeqFrag&quot;      
##  [93] &quot;getlistrank&quot;             &quot;getliststate&quot;           
##  [95] &quot;getLocation&quot;             &quot;getLocation.default&quot;    
##  [97] &quot;getLocation.list&quot;        &quot;getLocation.logical&quot;    
##  [99] &quot;getLocation.qaw&quot;         &quot;getLocation.SeqAcnucWeb&quot;
## [101] &quot;getName&quot;                 &quot;getName.default&quot;        
## [103] &quot;getName.list&quot;            &quot;getName.logical&quot;        
## [105] &quot;getName.qaw&quot;             &quot;getName.SeqAcnucWeb&quot;    
## [107] &quot;getName.SeqFastaAA&quot;      &quot;getName.SeqFastadna&quot;    
## [109] &quot;getName.SeqFrag&quot;         &quot;getNumber.socket&quot;       
## [111] &quot;getSequence&quot;             &quot;getSequence.character&quot;  
## [113] &quot;getSequence.default&quot;     &quot;getSequence.list&quot;       
## [115] &quot;getSequence.logical&quot;     &quot;getSequence.qaw&quot;        
## [117] &quot;getSequence.SeqAcnucWeb&quot; &quot;getSequence.SeqFastaAA&quot; 
## [119] &quot;getSequence.SeqFastadna&quot; &quot;getSequence.SeqFrag&quot;    
## [121] &quot;getTrans&quot;                &quot;getTrans.character&quot;     
## [123] &quot;getTrans.default&quot;        &quot;getTrans.list&quot;          
## [125] &quot;getTrans.logical&quot;        &quot;getTrans.qaw&quot;           
## [127] &quot;getTrans.SeqAcnucWeb&quot;    &quot;getTrans.SeqFastadna&quot;   
## [129] &quot;getTrans.SeqFrag&quot;        &quot;getType&quot;                
## [131] &quot;gfrag&quot;                   &quot;ghelp&quot;                  
## [133] &quot;gln&quot;                     &quot;glr&quot;                    
## [135] &quot;gls&quot;                     &quot;is.SeqAcnucWeb&quot;         
## [137] &quot;is.SeqFastaAA&quot;           &quot;is.SeqFastadna&quot;         
## [139] &quot;is.SeqFrag&quot;              &quot;isenum&quot;                 
## [141] &quot;isn&quot;                     &quot;kaks&quot;                   
## [143] &quot;kdb&quot;                     &quot;knowndbs&quot;               
## [145] &quot;lseqinr&quot;                 &quot;modifylist&quot;             
## [147] &quot;move&quot;                    &quot;mv&quot;                     
## [149] &quot;n2s&quot;                     &quot;oriloc&quot;                 
## [151] &quot;parser.socket&quot;           &quot;peakabif&quot;               
## [153] &quot;permutation&quot;             &quot;pga&quot;                    
## [155] &quot;plot.SeqAcnucWeb&quot;        &quot;plotabif&quot;               
## [157] &quot;plotladder&quot;              &quot;plotPanels&quot;             
## [159] &quot;pmw&quot;                     &quot;prepgetannots&quot;          
## [161] &quot;prettyseq&quot;               &quot;print.qaw&quot;              
## [163] &quot;print.SeqAcnucWeb&quot;       &quot;query&quot;                  
## [165] &quot;quitacnuc&quot;               &quot;read.abif&quot;              
## [167] &quot;read.alignment&quot;          &quot;read.fasta&quot;             
## [169] &quot;readBins&quot;                &quot;readfirstrec&quot;           
## [171] &quot;readPanels&quot;              &quot;readsmj&quot;                
## [173] &quot;rearranged.oriloc&quot;       &quot;recstat&quot;                
## [175] &quot;residuecount&quot;            &quot;reverse.align&quot;          
## [177] &quot;rho&quot;                     &quot;rot13&quot;                  
## [179] &quot;s2c&quot;                     &quot;s2n&quot;                    
## [181] &quot;savelist&quot;                &quot;SEQINR.UTIL&quot;            
## [183] &quot;setlistname&quot;             &quot;splitseq&quot;               
## [185] &quot;stresc&quot;                  &quot;stutterabif&quot;            
## [187] &quot;summary.SeqFastaAA&quot;      &quot;summary.SeqFastadna&quot;    
## [189] &quot;swap&quot;                    &quot;syncodons&quot;              
## [191] &quot;synsequence&quot;             &quot;tablecode&quot;              
## [193] &quot;test.co.recstat&quot;         &quot;test.li.recstat&quot;        
## [195] &quot;translate&quot;               &quot;trimSpace&quot;              
## [197] &quot;uco&quot;                     &quot;ucoweight&quot;              
## [199] &quot;where.is.this.acc&quot;       &quot;words&quot;                  
## [201] &quot;words.pos&quot;               &quot;write.fasta&quot;            
## [203] &quot;zscore&quot;</code></pre>
</div>
<div id="using-short-read" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> Using Short Read<a href="bioconductor.html#using-short-read" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This library can be installed by invoking the <code>BiocManager::install("ShortRead")</code> command to the console.</p>
<p>load the library</p>
<div class="sourceCode" id="cb558"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb558-1"><a href="bioconductor.html#cb558-1" tabindex="-1"></a><span class="fu">library</span>(ShortRead)</span></code></pre></div>
<ul>
<li>Reading the fastq sequences</li>
</ul>
<div class="sourceCode" id="cb559"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb559-1"><a href="bioconductor.html#cb559-1" tabindex="-1"></a>fq <span class="ot">=</span> <span class="fu">readFastq</span>(<span class="st">&quot;data/file2.fastq&quot;</span>)</span>
<span id="cb559-2"><a href="bioconductor.html#cb559-2" tabindex="-1"></a>fq</span></code></pre></div>
<pre><code>## class: ShortReadQ
## length: 5 reads; width: 149..166 cycles</code></pre>
<ul>
<li>Preview of first 3 sequences</li>
</ul>
<div class="sourceCode" id="cb561"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb561-1"><a href="bioconductor.html#cb561-1" tabindex="-1"></a>fq[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>## class: ShortReadQ
## length: 3 reads; width: 149 161 cycles</code></pre>
<ul>
<li>Sequence information</li>
</ul>
<div class="sourceCode" id="cb563"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb563-1"><a href="bioconductor.html#cb563-1" tabindex="-1"></a><span class="fu">sread</span>(fq)</span></code></pre></div>
<pre><code>## DNAStringSet object of length 5:
##     width seq
## [1]   149 CGCAAGCGAGTTAAGCAGTCTGTCCAAAGCATGT...TTATCCACATCTGTCATGACGATTTCGTTTGAT
## [2]   161 CGTTATCAGAGAGCTCTTGTCGCAAGCGAGTTAA...CTACCAAGTTATCCACATCTGTCATGACGATTT
## [3]   161 AGCGAGTTAAGCAGTCTGTCCAAAGCATGTGTAT...GACGATTTCGTTTGATTTTGCCTTAAAAAGAAA
## [4]   166 GACCCCTTCTCATCAGGAAGGGTTAATCTTTAAA...TGGATTCGCCCTTTAGTAGCTGAAGAAGTAGAG
## [5]   166 CCCCTTCTCATCAGGAAGGGTTAATCTTTAAATG...GATTCGCCCTTTAGTAGCTGAAGAAGTAGAGGG</code></pre>
<p>Subset the sequence from second to fifth</p>
<div class="sourceCode" id="cb565"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb565-1"><a href="bioconductor.html#cb565-1" tabindex="-1"></a><span class="fu">sread</span>(fq[<span class="dv">2</span><span class="sc">:</span><span class="dv">5</span>])</span></code></pre></div>
<pre><code>## DNAStringSet object of length 4:
##     width seq
## [1]   161 CGTTATCAGAGAGCTCTTGTCGCAAGCGAGTTAA...CTACCAAGTTATCCACATCTGTCATGACGATTT
## [2]   161 AGCGAGTTAAGCAGTCTGTCCAAAGCATGTGTAT...GACGATTTCGTTTGATTTTGCCTTAAAAAGAAA
## [3]   166 GACCCCTTCTCATCAGGAAGGGTTAATCTTTAAA...TGGATTCGCCCTTTAGTAGCTGAAGAAGTAGAG
## [4]   166 CCCCTTCTCATCAGGAAGGGTTAATCTTTAAATG...GATTCGCCCTTTAGTAGCTGAAGAAGTAGAGGG</code></pre>
<p>Inspect the quality of the sequence</p>
<div class="sourceCode" id="cb567"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb567-1"><a href="bioconductor.html#cb567-1" tabindex="-1"></a><span class="fu">quality</span>(fq)</span></code></pre></div>
<pre><code>## class: FastqQuality
## quality:
## BStringSet object of length 5:
##     width seq
## [1]   149 CCCC&gt;CHDBBC&lt;=8=DCCCDDDCB&gt;BG?GGEDB=...C@B==8=CDDDDDDDDDC==&lt;C@C8&lt;&lt;DD=BCE
## [2]   161 &lt;CE:@C=&lt;&lt;=DDDCBBC&gt;&lt;&lt;&lt;CDD&gt;BCEHGI&gt;C&gt;...999929-39:&gt;BE&gt;BBA@ADEC?&gt;?AAA999C;
## [3]   161 DDEDEEEAE@DD=&gt;=:==BB?BG&gt;FDEDCCBBBB...D&gt;///99*//::4:&lt;BBB6BD-2--555-5:::
## [4]   166 CCCLL;E@DE&gt;&gt;=DDDAE@EEBE@@?CCEE?EE?...BB@DDAD8884;C&lt;CFGGE?@?C?CC=BDEEEA
## [5]   166 666/6:::=BBCCBACADD6@:D@DDEE&gt;DD&gt;DE...&gt;BDAC58;6:D&gt;DDDDGEEBB&lt;CD@DDDEEED&gt;</code></pre>
<ul>
<li>Number of sequence in the file</li>
</ul>
<div class="sourceCode" id="cb569"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb569-1"><a href="bioconductor.html#cb569-1" tabindex="-1"></a><span class="fu">length</span>(fq)</span></code></pre></div>
<pre><code>## [1] 5</code></pre>
<ul>
<li>Length of reads</li>
</ul>
<div class="sourceCode" id="cb571"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb571-1"><a href="bioconductor.html#cb571-1" tabindex="-1"></a>readLength <span class="ot">=</span> <span class="fu">width</span>(fq)</span>
<span id="cb571-2"><a href="bioconductor.html#cb571-2" tabindex="-1"></a>readLength</span></code></pre></div>
<pre><code>## [1] 149 161 161 166 166</code></pre>
<ul>
<li>generating the summary of fasta sequences</li>
</ul>
<div class="sourceCode" id="cb573"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb573-1"><a href="bioconductor.html#cb573-1" tabindex="-1"></a>fq_reads <span class="ot">=</span> <span class="fu">sread</span>(fq)</span>
<span id="cb573-2"><a href="bioconductor.html#cb573-2" tabindex="-1"></a></span>
<span id="cb573-3"><a href="bioconductor.html#cb573-3" tabindex="-1"></a>id_of_reads<span class="ot">=</span><span class="fu">id</span>(fq)</span>
<span id="cb573-4"><a href="bioconductor.html#cb573-4" tabindex="-1"></a>id_of_reads[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>## BStringSet object of length 5:
##     width seq
## [1]    17 OLVOG:01110:02677
## [2]    17 OLVOG:01219:01823
## [3]    17 OLVOG:01240:02825
## [4]    17 OLVOG:02464:02160
## [5]    17 OLVOG:02703:02334</code></pre>
<div class="sourceCode" id="cb575"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb575-1"><a href="bioconductor.html#cb575-1" tabindex="-1"></a>fq_reads</span></code></pre></div>
<pre><code>## DNAStringSet object of length 5:
##     width seq
## [1]   149 CGCAAGCGAGTTAAGCAGTCTGTCCAAAGCATGT...TTATCCACATCTGTCATGACGATTTCGTTTGAT
## [2]   161 CGTTATCAGAGAGCTCTTGTCGCAAGCGAGTTAA...CTACCAAGTTATCCACATCTGTCATGACGATTT
## [3]   161 AGCGAGTTAAGCAGTCTGTCCAAAGCATGTGTAT...GACGATTTCGTTTGATTTTGCCTTAAAAAGAAA
## [4]   166 GACCCCTTCTCATCAGGAAGGGTTAATCTTTAAA...TGGATTCGCCCTTTAGTAGCTGAAGAAGTAGAG
## [5]   166 CCCCTTCTCATCAGGAAGGGTTAATCTTTAAATG...GATTCGCCCTTTAGTAGCTGAAGAAGTAGAGGG</code></pre>
<ul>
<li>Seq length - we can generate the sequence lenght by converting it to a dataframe</li>
</ul>
<div class="sourceCode" id="cb577"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb577-1"><a href="bioconductor.html#cb577-1" tabindex="-1"></a>seq_length <span class="ot">=</span> <span class="fu">as.data.frame</span>(fq_reads<span class="sc">@</span>ranges<span class="sc">@</span>width)</span>
<span id="cb577-2"><a href="bioconductor.html#cb577-2" tabindex="-1"></a>seq_length</span></code></pre></div>
<pre><code>##   fq_reads@ranges@width
## 1                   149
## 2                   161
## 3                   161
## 4                   166
## 5                   166</code></pre>
<ul>
<li>Quality scores</li>
</ul>
<div class="sourceCode" id="cb579"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb579-1"><a href="bioconductor.html#cb579-1" tabindex="-1"></a>quality_scores<span class="ot">=</span><span class="fu">quality</span>(fq)</span>
<span id="cb579-2"><a href="bioconductor.html#cb579-2" tabindex="-1"></a>quality_scores[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>## class: FastqQuality
## quality:
## BStringSet object of length 5:
##     width seq
## [1]   149 CCCC&gt;CHDBBC&lt;=8=DCCCDDDCB&gt;BG?GGEDB=...C@B==8=CDDDDDDDDDC==&lt;C@C8&lt;&lt;DD=BCE
## [2]   161 &lt;CE:@C=&lt;&lt;=DDDCBBC&gt;&lt;&lt;&lt;CDD&gt;BCEHGI&gt;C&gt;...999929-39:&gt;BE&gt;BBA@ADEC?&gt;?AAA999C;
## [3]   161 DDEDEEEAE@DD=&gt;=:==BB?BG&gt;FDEDCCBBBB...D&gt;///99*//::4:&lt;BBB6BD-2--555-5:::
## [4]   166 CCCLL;E@DE&gt;&gt;=DDDAE@EEBE@@?CCEE?EE?...BB@DDAD8884;C&lt;CFGGE?@?C?CC=BDEEEA
## [5]   166 666/6:::=BBCCBACADD6@:D@DDEE&gt;DD&gt;DE...&gt;BDAC58;6:D&gt;DDDDGEEBB&lt;CD@DDDEEED&gt;</code></pre>
<p>Convert the scores to matrix and see as real numbers</p>
<div class="sourceCode" id="cb581"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb581-1"><a href="bioconductor.html#cb581-1" tabindex="-1"></a>scores<span class="ot">=</span><span class="fu">as</span>(quality_scores, <span class="st">&quot;matrix&quot;</span>)</span>
<span id="cb581-2"><a href="bioconductor.html#cb581-2" tabindex="-1"></a>scores</span></code></pre></div>
<pre><code>##      [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10] [,11] [,12] [,13] [,14]
## [1,]   34   34   34   34   29   34   39   35   33    33    34    27    28    23
## [2,]   27   34   36   25   31   34   28   27   27    28    35    35    35    34
## [3,]   35   35   36   35   36   36   36   32   36    31    35    35    28    29
## [4,]   34   34   34   43   43   26   36   31   35    36    29    29    28    35
## [5,]   21   21   21   14   21   25   25   25   28    33    33    34    34    33
##      [,15] [,16] [,17] [,18] [,19] [,20] [,21] [,22] [,23] [,24] [,25] [,26]
## [1,]    28    35    34    34    34    35    35    35    34    33    29    33
## [2,]    33    33    34    29    27    27    27    34    35    35    29    33
## [3,]    28    25    28    28    33    33    30    33    38    29    37    35
## [4,]    35    35    32    36    31    36    36    33    36    31    31    30
## [5,]    32    34    32    35    35    21    31    25    35    31    35    35
##      [,27] [,28] [,29] [,30] [,31] [,32] [,33] [,34] [,35] [,36] [,37] [,38]
## [1,]    38    30    38    38    36    35    33    28    28    28    35    36
## [2,]    34    36    39    38    40    29    34    29    34    37    28    28
## [3,]    36    35    34    34    33    33    33    33    33    34    33    25
## [4,]    34    34    36    36    30    36    36    30    35    35    27    27
## [5,]    36    36    29    35    35    29    35    36    35    35    36    30
##      [,39] [,40] [,41] [,42] [,43] [,44] [,45] [,46] [,47] [,48] [,49] [,50]
## [1,]    38    38    38    26    37    34    34    34    34    34    33    34
## [2,]    29    35    34    34    32    34    23    28    28    25    28    35
## [3,]    33    35    35    35    35    35    35    34    34    28    28    28
## [4,]    21    16    25    30    25    30    33    30    34    40    37    26
## [5,]    36    36    32    36    36    31    36    36    36    26    35    36
##      [,51] [,52] [,53] [,54] [,55] [,56] [,57] [,58] [,59] [,60] [,61] [,62]
## [1,]    34    30    33    33    33    25    33    36    30    37    31    35
## [2,]    35    36    36    34    33    33    33    33    33    33    33    25
## [3,]    28    21    26    34    23    32    26    23    23    18    14    25
## [4,]    33    30    30    36    36    36    36    27    20    27    35    36
## [5,]    36    36    35    35    35    28    28    28    35    35    35    26
##      [,63] [,64] [,65] [,66] [,67] [,68] [,69] [,70] [,71] [,72] [,73] [,74]
## [1,]    35    36    30    35    35    28    27    27    23    27    35    28
## [2,]    33    30    30    27    28    27    25    25    34    25    25    25
## [3,]    33    25    25    25    14    15    28    21    28    27    28    21
## [4,]    36    26    35    33    34    34    35    36    35    35    35    36
## [5,]    35    34    35    35    35    36    35    35    35    37    36    36
##      [,75] [,76] [,77] [,78] [,79] [,80] [,81] [,82] [,83] [,84] [,85] [,86]
## [1,]    35    35    35    26    32    19    25    18    25    23    29    26
## [2,]    25    14    25    25    14    25    26    32    32    36    30    34
## [3,]    27     9    13     9    12    16    20     8    24    25    35    35
## [4,]    35    35    30    36    32    35    35    35    35    35    35    35
## [5,]    29    36    32    35    35    35    34    35    34    36    35    35
##      [,87] [,88] [,89] [,90] [,91] [,92] [,93] [,94] [,95] [,96] [,97] [,98]
## [1,]    34    35    36    37    33    33    33    34    32    37    40    38
## [2,]    34    31    30    29    14    25    25    14    25    34    34    32
## [3,]    35    35    34    35    29    34    34    33    19    31    31    24
## [4,]    35    35    35    30    34    31    35    35    35    35    35    35
## [5,]    35    30    34    31    35    35    35    35    34    34    31    36
##      [,99] [,100] [,101] [,102] [,103] [,104] [,105] [,106] [,107] [,108]
## [1,]    26     34     34     35     35     30     34     35     26     32
## [2,]    34     17     24     24     13     24     17     24     28     34
## [3,]    18     12     18     18     12     17     23     17     17     12
## [4,]    32     36     35     35     35     34     34     30     34     36
## [5,]    35     35     36     35     36     28     33     33     33     33
##      [,109] [,110] [,111] [,112] [,113] [,114] [,115] [,116] [,117] [,118]
## [1,]     32     25     25     25     23     28     33     35     34     31
## [2,]     33     32     33     36     36     35     36     23     26     26
## [3,]     17     16     23     23     25     32     33     33     26     21
## [4,]     35     35     35     35     35     32     33     33     35     35
## [5,]     34     32     32     32     34     34     34     35     35     28
##      [,119] [,120] [,121] [,122] [,123] [,124] [,125] [,126] [,127] [,128]
## [1,]     33     28     28     23     28     34     35     35     35     35
## [2,]     26     12     24     31     30     29     24     30     29     23
## [3,]     21     24     32     35     36     33     33     33     33     33
## [4,]     35     30     34     34     35     35     35     35     29     36
## [5,]     32     33     35     33     33     33     28     33     35     35
##      [,129] [,130] [,131] [,132] [,133] [,134] [,135] [,136] [,137] [,138]
## [1,]     35     35     35     35     35     34     28     28     27     34
## [2,]     24     24     24     24     17     24     12     18     24     25
## [3,]     35     29     14     14     14     24     24      9     14     14
## [4,]     35     35     34     34     33     33     33     31     35     35
## [5,]     35     35     35     33     33     29     33     35     32     34
##      [,139] [,140] [,141] [,142] [,143] [,144] [,145] [,146] [,147] [,148]
## [1,]     31     34     23     27     27     35     35     28     33     34
## [2,]     29     33     36     29     33     33     32     31     32     35
## [3,]     25     25     19     25     27     33     33     33     21     33
## [4,]     32     35     23     23     23     19     26     34     27     34
## [5,]     20     23     26     21     25     35     29     35     35     35
##      [,149] [,150] [,151] [,152] [,153] [,154] [,155] [,156] [,157] [,158]
## [1,]     36     NA     NA     NA     NA     NA     NA     NA     NA     NA
## [2,]     36     34     30     29     30     32     32     32     24     24
## [3,]     35     12     17     12     12     20     20     20     12     20
## [4,]     37     38     38     36     30     31     30     34     30     34
## [5,]     35     38     36     36     33     33     27     34     35     31
##      [,159] [,160] [,161] [,162] [,163] [,164] [,165] [,166]
## [1,]     NA     NA     NA     NA     NA     NA     NA     NA
## [2,]     24     34     26     NA     NA     NA     NA     NA
## [3,]     25     25     25     NA     NA     NA     NA     NA
## [4,]     34     28     33     35     36     36     36     32
## [5,]     35     35     35     36     36     36     35     29</code></pre>
<ul>
<li>Mean score for each sequence</li>
</ul>
<div class="sourceCode" id="cb583"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb583-1"><a href="bioconductor.html#cb583-1" tabindex="-1"></a>mean_score_for_each_sequence <span class="ot">=</span> <span class="fu">rowMeans</span>(scores, <span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb583-2"><a href="bioconductor.html#cb583-2" tabindex="-1"></a>mean_score_for_each_sequence</span></code></pre></div>
<pre><code>## [1] 32.00000 29.18012 26.57764 32.99398 32.71084</code></pre>
<ul>
<li>Mean score for each position</li>
</ul>
<div class="sourceCode" id="cb585"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb585-1"><a href="bioconductor.html#cb585-1" tabindex="-1"></a>mean_score_for_each_position <span class="ot">=</span> <span class="fu">rowMeans</span>(scores, <span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb585-2"><a href="bioconductor.html#cb585-2" tabindex="-1"></a>mean_score_for_each_position</span></code></pre></div>
<pre><code>## [1] 32.00000 29.18012 26.57764 32.99398 32.71084</code></pre>
<p>Plot a boxplot for the scores</p>
<div class="sourceCode" id="cb587"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb587-1"><a href="bioconductor.html#cb587-1" tabindex="-1"></a><span class="fu">boxplot</span>(scores, <span class="at">xlab=</span><span class="st">&quot;Position&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;Phred score&quot;</span>)</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-283-1.png" width="672" /></p>
</div>
<div id="genomic-ranges" class="section level2 hasAnchor" number="7.5">
<h2><span class="header-section-number">7.5</span> Genomic Ranges<a href="bioconductor.html#genomic-ranges" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>GenomicRanges</code> and <code>IRanges</code> packages are used in the calculations. These packages are masked in the biostrings.</p>
<ul>
<li><code>Genomic Ranges</code> are representing and manipulating genomic intervals and variables defined along a genome.</li>
<li><code>IRanges</code> used to provide an efficient handling of integer ranges, a fundamental data structure for representing genomic data.</li>
</ul>
<p>Lets inspect a scenario where the libraries are used. The <code>IRanges</code> object will be randomly created.</p>
<div class="sourceCode" id="cb588"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb588-1"><a href="bioconductor.html#cb588-1" tabindex="-1"></a><span class="fu">library</span>(GenomicRanges)</span>
<span id="cb588-2"><a href="bioconductor.html#cb588-2" tabindex="-1"></a><span class="fu">library</span>(IRanges)</span>
<span id="cb588-3"><a href="bioconductor.html#cb588-3" tabindex="-1"></a></span>
<span id="cb588-4"><a href="bioconductor.html#cb588-4" tabindex="-1"></a><span class="co"># Creating IRanges object</span></span>
<span id="cb588-5"><a href="bioconductor.html#cb588-5" tabindex="-1"></a>ir <span class="ot">&lt;-</span> <span class="fu">IRanges</span>(<span class="at">start=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">6</span>,<span class="dv">10</span>), <span class="at">end=</span><span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">8</span>,<span class="dv">12</span>))</span>
<span id="cb588-6"><a href="bioconductor.html#cb588-6" tabindex="-1"></a><span class="fu">print</span>(ir)</span></code></pre></div>
<pre><code>## IRanges object with 3 ranges and 0 metadata columns:
##           start       end     width
##       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
##   [1]         1         5         5
##   [2]         6         8         3
##   [3]        10        12         3</code></pre>
<div class="sourceCode" id="cb590"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb590-1"><a href="bioconductor.html#cb590-1" tabindex="-1"></a><span class="fu">library</span>(GenomicRanges)</span>
<span id="cb590-2"><a href="bioconductor.html#cb590-2" tabindex="-1"></a><span class="fu">library</span>(IRanges)</span>
<span id="cb590-3"><a href="bioconductor.html#cb590-3" tabindex="-1"></a></span>
<span id="cb590-4"><a href="bioconductor.html#cb590-4" tabindex="-1"></a><span class="co"># Creating GenomicRanges object</span></span>
<span id="cb590-5"><a href="bioconductor.html#cb590-5" tabindex="-1"></a>gr <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="at">seqnames=</span><span class="st">&quot;chr1&quot;</span>, <span class="at">ranges=</span><span class="fu">IRanges</span>(<span class="at">start=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">6</span>,<span class="dv">10</span>), <span class="at">end=</span><span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">8</span>,<span class="dv">12</span>)))</span>
<span id="cb590-6"><a href="bioconductor.html#cb590-6" tabindex="-1"></a><span class="fu">print</span>(gr)</span></code></pre></div>
<pre><code>## GRanges object with 3 ranges and 0 metadata columns:
##       seqnames    ranges strand
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]     chr1       1-5      *
##   [2]     chr1       6-8      *
##   [3]     chr1     10-12      *
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<p>Genomic Ranges objects can be created from the IRanges. Lets now find the overlap in the genomic ranges created.</p>
<div class="sourceCode" id="cb592"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb592-1"><a href="bioconductor.html#cb592-1" tabindex="-1"></a><span class="fu">library</span>(GenomicRanges)</span>
<span id="cb592-2"><a href="bioconductor.html#cb592-2" tabindex="-1"></a><span class="fu">library</span>(IRanges)</span>
<span id="cb592-3"><a href="bioconductor.html#cb592-3" tabindex="-1"></a></span>
<span id="cb592-4"><a href="bioconductor.html#cb592-4" tabindex="-1"></a><span class="co"># Finding overlaps</span></span>
<span id="cb592-5"><a href="bioconductor.html#cb592-5" tabindex="-1"></a>query <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="at">seqnames=</span><span class="st">&quot;chr1&quot;</span>, <span class="at">ranges=</span><span class="fu">IRanges</span>(<span class="at">start=</span><span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">7</span>), <span class="at">end=</span><span class="fu">c</span>(<span class="dv">8</span>,<span class="dv">10</span>)))</span>
<span id="cb592-6"><a href="bioconductor.html#cb592-6" tabindex="-1"></a>overlap <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(query, gr)</span>
<span id="cb592-7"><a href="bioconductor.html#cb592-7" tabindex="-1"></a><span class="fu">print</span>(overlap)</span></code></pre></div>
<pre><code>## Hits object with 4 hits and 0 metadata columns:
##       queryHits subjectHits
##       &lt;integer&gt;   &lt;integer&gt;
##   [1]         1           1
##   [2]         1           2
##   [3]         2           2
##   [4]         2           3
##   -------
##   queryLength: 2 / subjectLength: 3</code></pre>
</div>
<div id="using-deseq-package" class="section level2 hasAnchor" number="7.6">
<h2><span class="header-section-number">7.6</span> Using `DESeq package<a href="bioconductor.html#using-deseq-package" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <code>DESeq</code> package is used to analyze the RNA-Seq data from the genome to find the gene expression. <code>EdgeR</code> can also be used especially where differential analysis of RNA data is required however in this section the <code>DESeq2</code> package will be used.</p>
<p>Lets create a random data to work on the task</p>
<pre><code>library(DESeq2)

# Load example data
data &lt;- matrix(rnorm(1000), ncol=5)
colData &lt;- data.frame(condition=rep(c(&quot;control&quot;, &quot;treated&quot;), each=2))
colData</code></pre>
<p>Lets now use the randomly generated data to create DESeqDataSet</p>
<pre><code>library(DESeq2)

# Load example data
data &lt;- matrix(rnorm(1000), ncol=5)
colData &lt;- data.frame(condition=rep(c(&quot;control&quot;, &quot;treated&quot;), each=2))

# Create DESeqDataSet
dds &lt;- DESeqDataSetFromMatrix(countData=data, colData=colData, design=~condition)

# Run DESeq pipeline
dds &lt;- DESeq(dds)
results &lt;- results(dds)

# Summarize results
summary(results)</code></pre>
<pre><code>if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)
BiocManager::install(&quot;DESeq2&quot;)

# Check dimensions of data and colData
print(dim(data))
print(dim(colData))

# Check a few rows of colData
print(head(colData))

# Check if row names are set correctly in data
print(rownames(data))

# Try setting row names if they are missing
if (is.null(rownames(data))) {
  rownames(data) &lt;- paste0(&quot;Gene&quot;, seq_len(nrow(data)))
}

# Load DESeq2 library
library(DESeq2)

# Create DESeqDataSet
dds &lt;- DESeqDataSetFromMatrix(countData = data,
                              colData = colData,
                              design = ~ condition)</code></pre>
</div>
<div id="network-analysis-of-genomic-data" class="section level2 hasAnchor" number="7.7">
<h2><span class="header-section-number">7.7</span> Network Analysis of Genomic Data<a href="bioconductor.html#network-analysis-of-genomic-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>MarkovChain is installed by <code>install.packages("markovchain")</code></p>
<pre><code># Example of creating a Markov chain object with markovchain
library(markovchain)

# Define states and transition matrix (replace with actual data)
states &lt;- c(&quot;Up-regulated&quot;, &quot;Down-regulated&quot;, &quot;No change&quot;)
transition_matrix &lt;- matrix(c(0.7, 0.1, 0.2,
                              0.2, 0.6, 0.2,
                              0.1, 0.3, 0.6), byrow = TRUE, nrow = 3, dimnames = list(states, states))

# Create a Markov chain object
mc &lt;- new(&quot;markovchain&quot;, states = states, transitionMatrix = transition_matrix)

# Example of constructing and analyzing a gene regulatory network with igraph
library(igraph)

# Generate example data (replace with actual data)
genes &lt;- c(&quot;Gene1&quot;, &quot;Gene2&quot;, &quot;Gene3&quot;, &quot;Gene4&quot;)
edges &lt;- data.frame(from = c(&quot;Gene1&quot;, &quot;Gene1&quot;, &quot;Gene2&quot;, &quot;Gene3&quot;),
                    to = c(&quot;Gene2&quot;, &quot;Gene3&quot;, &quot;Gene4&quot;, &quot;Gene1&quot;))

# Create a graph object
g &lt;- graph_from_data_frame(edges, directed = TRUE, vertices = genes)

# Analyze the graph
print(summary(g))
plot(g, layout = layout.circle)</code></pre>
<p>The generate data was fed into DESeq pipeline.</p>
</div>
<div id="code-challenge-6" class="section level2 hasAnchor" number="7.8">
<h2><span class="header-section-number">7.8</span> CODE CHALLENGE<a href="bioconductor.html#code-challenge-6" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this code challenge you will be required to installed the Bioconductor locally as guided above. Thereafter, you will install <code>MultiAssayExperiment</code> package that is part of the Bioconductor project. Below is the code to install the <code>MultiAssayExperiment</code> locally;</p>
<pre><code>if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)
BiocManager::install(&quot;MultiAssayExperiment&quot;)</code></pre>
<p>You will load the package along with the ggplot2 library and work on the <code>genCov</code> data. <code>genCov</code> data is an inbuilt data set from <code>MultiAssayExperiment</code> that contains simulated data showcasing the integration and analysis of multiple types of data from different assays. Challenge yourself with the following activities to get a clear understanding of genomic coverage data;</p>
<ol style="list-style-type: decimal">
<li>Load the <code>genCov</code> using the code, <code>data("genCov", package = "MultiAssayExperiment")</code>.<br />
</li>
<li>Check the structure <code>str()</code> of the data set and calculate the descriptive statistics.</li>
<li>Extract the coverage data with code <code>experiments(genCov)[["coverage"]]</code> and store in a variable <code>coverage_data</code>. Display the first 6 records of the the data.</li>
<li>Extract and inspect the genomic ranges from <code>genCov</code> with <code>rowRanges()</code> command. Display the first 6 genomic ranges.</li>
<li>Plot a line chart of the <code>"Coverage"</code> against the <code>"Genomic Position"</code>.</li>
<li>Calculate the average coverage for each sample in the <code>coverageData</code> using <code>colMeans()</code> command. Display the result on a bar chart.</li>
</ol>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="ggplot.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/balsaedi/RCourse/edit/main/07-bioconductor_library_Advance-R.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/balsaedi/RCourse/blob/main/07-bioconductor_library_Advance-R.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
