<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Bioconductor Package | R Course</title>
  <meta name="description" content="Chapter 2 Bioconductor Package | R Course" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Bioconductor Package | R Course" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Bioconductor Package | R Course" />
  
  
  

<meta name="author" content="YOUR NAME HERE" />


<meta name="date" content="2024-08-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="chemical-properties-prediction-and-visualized..html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">SHORT TITLE HERE</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Analysis of Drug like Molecule using ChemmineR</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#objectives"><i class="fa fa-check"></i><b>1.1</b> Objectives</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#outcome"><i class="fa fa-check"></i><b>1.2</b> Outcome</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#informatics"><i class="fa fa-check"></i><b>1.3</b> Informatics</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#bioinformatics"><i class="fa fa-check"></i><b>1.3.1</b> Bioinformatics</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#chemoinformatics"><i class="fa fa-check"></i><b>1.3.2</b> Chemoinformatics</a></li>
<li class="chapter" data-level="1.3.3" data-path="index.html"><a href="index.html#application-of-chemoinformatics-in-the-drug-industry"><i class="fa fa-check"></i><b>1.3.3</b> Application of Chemoinformatics in the Drug Industry</a></li>
<li class="chapter" data-level="1.3.4" data-path="index.html"><a href="index.html#chemical-databases"><i class="fa fa-check"></i><b>1.3.4</b> Chemical databases</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#the-chemminer-tool"><i class="fa fa-check"></i><b>1.4</b> The ChemmineR tool</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="index.html"><a href="index.html#overview-of-chemmine-web-tool"><i class="fa fa-check"></i><b>1.4.1</b> Overview of Chemmine web tool</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="bioconductor-package.html"><a href="bioconductor-package.html"><i class="fa fa-check"></i><b>2</b> Bioconductor Package</a>
<ul>
<li class="chapter" data-level="2.1" data-path="bioconductor-package.html"><a href="bioconductor-package.html#overview-of-the-bioconductor-package"><i class="fa fa-check"></i><b>2.1</b> Overview of the BioConductor package</a></li>
<li class="chapter" data-level="2.2" data-path="bioconductor-package.html"><a href="bioconductor-package.html#installation-of-bioconductor-package"><i class="fa fa-check"></i><b>2.2</b> Installation of BioConductor package</a></li>
<li class="chapter" data-level="2.3" data-path="bioconductor-package.html"><a href="bioconductor-package.html#importing-data-in-r"><i class="fa fa-check"></i><b>2.3</b> Importing Data in R</a></li>
<li class="chapter" data-level="2.4" data-path="bioconductor-package.html"><a href="bioconductor-package.html#writing-script-of-code-in-r"><i class="fa fa-check"></i><b>2.4</b> Writing Script of Code in R</a></li>
<li class="chapter" data-level="2.5" data-path="bioconductor-package.html"><a href="bioconductor-package.html#data-interface"><i class="fa fa-check"></i><b>2.5</b> Data Interface</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="bioconductor-package.html"><a href="bioconductor-package.html#reading-the-.csv-file"><i class="fa fa-check"></i><b>2.5.1</b> Reading the *.csv file</a></li>
<li class="chapter" data-level="2.5.2" data-path="bioconductor-package.html"><a href="bioconductor-package.html#brief-data-analysis"><i class="fa fa-check"></i><b>2.5.2</b> Brief data analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="bioconductor-package.html"><a href="bioconductor-package.html#introduction-to-bioconductor"><i class="fa fa-check"></i><b>2.6</b> Introduction to Bioconductor</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="bioconductor-package.html"><a href="bioconductor-package.html#exploring-genomic-data"><i class="fa fa-check"></i><b>2.6.1</b> Exploring Genomic data</a></li>
<li class="chapter" data-level="2.6.2" data-path="bioconductor-package.html"><a href="bioconductor-package.html#visualizing-gene-expression-data"><i class="fa fa-check"></i><b>2.6.2</b> Visualizing Gene Expression Data</a></li>
<li class="chapter" data-level="2.6.3" data-path="bioconductor-package.html"><a href="bioconductor-package.html#annotating-genomic-data"><i class="fa fa-check"></i><b>2.6.3</b> Annotating Genomic Data</a></li>
<li class="chapter" data-level="2.6.4" data-path="bioconductor-package.html"><a href="bioconductor-package.html#working-with-sequence-data"><i class="fa fa-check"></i><b>2.6.4</b> Working with Sequence Data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="chemical-properties-prediction-and-visualized..html"><a href="chemical-properties-prediction-and-visualized..html"><i class="fa fa-check"></i><b>3</b> Chemical Properties prediction and visualized.</a>
<ul>
<li class="chapter" data-level="3.1" data-path="chemical-properties-prediction-and-visualized..html"><a href="chemical-properties-prediction-and-visualized..html#whats-is-chemminer"><i class="fa fa-check"></i><b>3.1</b> Whats is ChemmineR?</a></li>
<li class="chapter" data-level="3.2" data-path="chemical-properties-prediction-and-visualized..html"><a href="chemical-properties-prediction-and-visualized..html#installation"><i class="fa fa-check"></i><b>3.2</b> Installation</a></li>
<li class="chapter" data-level="3.3" data-path="chemical-properties-prediction-and-visualized..html"><a href="chemical-properties-prediction-and-visualized..html#loading-the-package-and-documentation"><i class="fa fa-check"></i><b>3.3</b> Loading the package and Documentation</a></li>
<li class="chapter" data-level="3.4" data-path="chemical-properties-prediction-and-visualized..html"><a href="chemical-properties-prediction-and-visualized..html#create-instances-of-sdfset-class"><i class="fa fa-check"></i><b>3.4</b> Create Instances of SDFset class</a></li>
<li class="chapter" data-level="3.5" data-path="chemical-properties-prediction-and-visualized..html"><a href="chemical-properties-prediction-and-visualized..html#retrieving-atom-bond-and-data-block"><i class="fa fa-check"></i><b>3.5</b> Retrieving atom, bond and data block</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="chemical-properties-prediction-and-visualized..html"><a href="chemical-properties-prediction-and-visualized..html#assigning-compound-ids-and-keeping-them-unique"><i class="fa fa-check"></i><b>3.5.1</b> Assigning compound IDS and keeping them unique</a></li>
<li class="chapter" data-level="3.5.2" data-path="chemical-properties-prediction-and-visualized..html"><a href="chemical-properties-prediction-and-visualized..html#converting-the-data-blocks-in-an-sdfset-to-a-matrix"><i class="fa fa-check"></i><b>3.5.2</b> Converting the data blocks in an SDFset to a matrix</a></li>
<li class="chapter" data-level="3.5.3" data-path="chemical-properties-prediction-and-visualized..html"><a href="chemical-properties-prediction-and-visualized..html#string-searching-in-sdfset"><i class="fa fa-check"></i><b>3.5.3</b> String searching in SDFset</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="chemical-properties-prediction-and-visualized..html"><a href="chemical-properties-prediction-and-visualized..html#molecular-visualization-in-r"><i class="fa fa-check"></i><b>3.6</b> Molecular visualization in R</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="chemical-properties-prediction-and-visualized..html"><a href="chemical-properties-prediction-and-visualized..html#molecular-property-functions"><i class="fa fa-check"></i><b>3.6.1</b> Molecular Property Functions</a></li>
<li class="chapter" data-level="3.6.2" data-path="chemical-properties-prediction-and-visualized..html"><a href="chemical-properties-prediction-and-visualized..html#similarity-comparisons-and-searching"><i class="fa fa-check"></i><b>3.6.2</b> Similarity Comparisons and Searching</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="chemical-properties-prediction-and-visualized..html"><a href="chemical-properties-prediction-and-visualized..html#resources"><i class="fa fa-check"></i><b>3.7</b> Resources</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">R Course</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="bioconductor-package" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Bioconductor Package<a href="bioconductor-package.html#bioconductor-package" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="overview-of-the-bioconductor-package" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Overview of the BioConductor package<a href="bioconductor-package.html#overview-of-the-bioconductor-package" class="anchor-section" aria-label="Anchor link to header"></a></h2>
</div>
<div id="installation-of-bioconductor-package" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Installation of BioConductor package<a href="bioconductor-package.html#installation-of-bioconductor-package" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To install BioConductor run the below command</p>
<pre><code>if (!requireNamespace(&quot;BiocManager&quot;, quitely=TRUE))
     install.packages(&quot;BiocManager&quot;)
 BiocManager::install(&quot;ChemmineR&quot;)</code></pre>
</div>
<div id="importing-data-in-r" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Importing Data in R<a href="bioconductor-package.html#importing-data-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Depending on the type of data file, R has a varying methods to load the data. The below list shows how to load different types of files to the R program.</p>
<ul>
<li><code>data &lt;- read_csv("path/to/your/file.csv")</code> for csv file</li>
<li><code>data &lt;- read_excel("path/to/your/file.xlsx", sheet = "Sheet1")</code> for excel file</li>
<li><code>data &lt;- fread("path/to/your/file.csv")</code> for large csv files.</li>
<li><code>data &lt;- read_sav("path/to/your/file.sav")</code> for SPSS files.</li>
<li><code>data &lt;- read_dta("path/to/your/file.dta")</code> for stata files.</li>
<li><code>data &lt;- read_sas("path/to/your/file.sas7bdat")</code> for SAS files.</li>
</ul>
</div>
<div id="writing-script-of-code-in-r" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Writing Script of Code in R<a href="bioconductor-package.html#writing-script-of-code-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In R, the use can interact directly with the console or prepare a script inform of <code>*.R</code> or <code>*.Rmd</code> (R Markdown) files.
Script are useful as they allow one to;</p>
<ul>
<li>easily pick up where they left off on a project and progressively build and refine code and analysis.</li>
<li>easily reproduce analysis run earlier and documentation. This also saves time on repetetive tasks.</li>
<li>do a batch processing such that tasks are automotated instead of running them one by one.</li>
<li>debug the code and find errors line by line.</li>
<li>version control such as git that tracks the changes and enables collaboration with other developers.</li>
</ul>
<p>A script is a set of commands that usually includes commenting on what each piece of code intended to do</p>
</div>
<div id="data-interface" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> Data Interface<a href="bioconductor-package.html#data-interface" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Follow the following steps when working with data from external interfaces;</p>
<ul>
<li>Refer CSV working file(filepath) e.g <code>test.csv</code></li>
<li>Get the current working directory, <code>getwd()</code></li>
<li>Set the current working directory <code>setwd()</code>.</li>
</ul>
<div id="reading-the-.csv-file" class="section level3 hasAnchor" number="2.5.1">
<h3><span class="header-section-number">2.5.1</span> Reading the *.csv file<a href="bioconductor-package.html#reading-the-.csv-file" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In this section, we will work with two csv files, <code>test.csv</code> and <code>data.csv</code>. Both files are stored in a directory named <code>data</code>. Lets load the <code>data.csv</code> to the program.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="bioconductor-package.html#cb2-1" tabindex="-1"></a>person_info <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/data.csv&quot;</span>)</span></code></pre></div>
<p>The <code>read.csv()</code> function automatically converts the loaded to a data frame. <code>is.data.frame()</code> is used to find if a variable is a data frame</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="bioconductor-package.html#cb3-1" tabindex="-1"></a><span class="co"># just to confirm if the said data is a data frame </span></span>
<span id="cb3-2"><a href="bioconductor-package.html#cb3-2" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">is.data.frame</span>(person_info))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p>The <code>person_info</code> data set is confirmed to be a variable.
The <code>ncol()</code> function counts the number of columns/variables/features in a data frame</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="bioconductor-package.html#cb5-1" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">ncol</span>(person_info)) <span class="co"># count the number of columns </span></span></code></pre></div>
<pre><code>## [1] 6</code></pre>
<p>This data frame has 6 columns. Alternatively, the <code>nrow()</code> function counts the number of rows/observations/records.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="bioconductor-package.html#cb7-1" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">nrow</span>(person_info)) <span class="co"># count the number of records/rows/observations</span></span></code></pre></div>
<pre><code>## [1] 10</code></pre>
<p>This data set has 10 rows and 6 columns.</p>
</div>
<div id="brief-data-analysis" class="section level3 hasAnchor" number="2.5.2">
<h3><span class="header-section-number">2.5.2</span> Brief data analysis<a href="bioconductor-package.html#brief-data-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>When the function, <code>attach()</code> is called to a data frame, the R program can refer to specific column without associating it with the data frame name. In this section the <code>person_info</code> will be attached to R and dive shallow into data analysis.</p>
<p>Calculate the mean of the salary</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="bioconductor-package.html#cb9-1" tabindex="-1"></a><span class="co"># attach the file inorder to avoid the $ sign</span></span>
<span id="cb9-2"><a href="bioconductor-package.html#cb9-2" tabindex="-1"></a><span class="fu">attach</span>(person_info)</span>
<span id="cb9-3"><a href="bioconductor-package.html#cb9-3" tabindex="-1"></a></span>
<span id="cb9-4"><a href="bioconductor-package.html#cb9-4" tabindex="-1"></a><span class="fu">mean</span>(Salary)</span></code></pre></div>
<pre><code>## [1] 65200</code></pre>
<p>The average salary in the data set is 65200. The <code>names()</code> function is called to find the column names of the data set.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="bioconductor-package.html#cb11-1" tabindex="-1"></a><span class="co"># find the column names </span></span>
<span id="cb11-2"><a href="bioconductor-package.html#cb11-2" tabindex="-1"></a><span class="fu">names</span>(person_info)</span></code></pre></div>
<pre><code>## [1] &quot;Name&quot;   &quot;Emp.Id&quot; &quot;Gender&quot; &quot;Age&quot;    &quot;Smoke&quot;  &quot;Salary&quot;</code></pre>
<p>Now, lets jump to the protein data which is stored in the <code>test.csv</code> file. The <code>head()</code> function is used to show the first 6 records of the dataset</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="bioconductor-package.html#cb13-1" tabindex="-1"></a><span class="co"># Load the data set </span></span>
<span id="cb13-2"><a href="bioconductor-package.html#cb13-2" tabindex="-1"></a>proteindata <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/test.csv&quot;</span>)</span>
<span id="cb13-3"><a href="bioconductor-package.html#cb13-3" tabindex="-1"></a><span class="fu">head</span>(proteindata)</span></code></pre></div>
<pre><code>##   SNO  DBNAME PROTEIN_NAME NO_DIMER NO_RESIDUES DEPOSITED_DATE
## 1   1     PDB      TUBULIN        3         349     05-05-2014
## 2   2     PDB      KERATIN        5         450     05-06-2015
## 3   3 UNIPROT   HEMOGLOBIN        7         345     05-07-2017
## 4   4     PIR  CYTOKINEASE       10         950     05-05-2014
## 5   5  EXPASY     PROTEASE        5         493     05-06-2015
## 6   6  ENTREX     TUBULIN1        4         760     05-07-2017</code></pre>
<p>Count the number of rows and columns in the protein dataset</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="bioconductor-package.html#cb15-1" tabindex="-1"></a><span class="fu">ncol</span>(proteindata)</span></code></pre></div>
<pre><code>## [1] 6</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="bioconductor-package.html#cb17-1" tabindex="-1"></a><span class="fu">nrow</span>(proteindata)</span></code></pre></div>
<pre><code>## [1] 10</code></pre>
<p>The data frame has 10 rows and 6 columns. Alternatively, the <code>dim()</code> can be used, it returns the number of rows and columns respectively.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="bioconductor-package.html#cb19-1" tabindex="-1"></a><span class="fu">dim</span>(proteindata)</span></code></pre></div>
<pre><code>## [1] 10  6</code></pre>
<p>Get the protein detail having the most residues and the max residues value.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="bioconductor-package.html#cb21-1" tabindex="-1"></a>objname1 <span class="ot">&lt;-</span> <span class="fu">max</span>(proteindata<span class="sc">$</span>NO_RESIDUES)</span>
<span id="cb21-2"><a href="bioconductor-package.html#cb21-2" tabindex="-1"></a>objname1</span></code></pre></div>
<pre><code>## [1] 950</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="bioconductor-package.html#cb23-1" tabindex="-1"></a>objname2 <span class="ot">&lt;-</span> <span class="fu">subset</span>(proteindata, NO_RESIDUES <span class="sc">==</span> <span class="fu">max</span>(NO_RESIDUES))</span>
<span id="cb23-2"><a href="bioconductor-package.html#cb23-2" tabindex="-1"></a>objname2</span></code></pre></div>
<pre><code>##   SNO DBNAME PROTEIN_NAME NO_DIMER NO_RESIDUES DEPOSITED_DATE
## 4   4    PIR  CYTOKINEASE       10         950     05-05-2014</code></pre>
<p>The <code>max()</code> function is used to find the maximum value in a vector or column. In this code, <code>max(proteindata$NO_RESIDUES)</code> returns the highest number of residues (<code>NO_RESIDUES</code>) found in the <code>proteindata</code> dataset, and this value is stored in <code>objname1</code>.</p>
<p>The <code>subset()</code> function is used to filter rows in a data frame that meet certain conditions. Here, <code>subset(proteindata, NO_RESIDUES == max(NO_RESIDUES))</code> returns all rows from the <code>proteindata</code> dataset where the <code>NO_RESIDUES</code> column matches the maximum value, and the result is stored in <code>objname2</code>.</p>
<ul>
<li>Get the protein detail having entry in PDB database</li>
</ul>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="bioconductor-package.html#cb25-1" tabindex="-1"></a>objname3 <span class="ot">&lt;-</span> <span class="fu">subset</span>(proteindata, DBNAME <span class="sc">==</span> <span class="st">&quot;PDB&quot;</span>)</span>
<span id="cb25-2"><a href="bioconductor-package.html#cb25-2" tabindex="-1"></a><span class="fu">print</span>(objname3)</span></code></pre></div>
<pre><code>##   SNO DBNAME PROTEIN_NAME NO_DIMER NO_RESIDUES DEPOSITED_DATE
## 1   1    PDB      TUBULIN        3         349     05-05-2014
## 2   2    PDB      KERATIN        5         450     05-06-2015
## 7   7    PDB     TUBULIN2        3         387     05-05-2014</code></pre>
<p>The <code>proteindata</code> dataset is filtered to select rows where the <code>DBNAME</code> column is equal to <code>"PDB"</code>. This means it extracts all proteins that are listed in the PDB database. The resulting subset is stored in <code>objname3</code>.</p>
<p>Get the protein detail having more than 3 dimer from PDB database</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="bioconductor-package.html#cb27-1" tabindex="-1"></a>objname5 <span class="ot">&lt;-</span> <span class="fu">subset</span>(proteindata, <span class="fu">as.Date</span>(DEPOSITED_DATE) <span class="sc">&gt;</span> <span class="fu">as.Date</span>(<span class="st">&quot;2014-05-05&quot;</span>))</span>
<span id="cb27-2"><a href="bioconductor-package.html#cb27-2" tabindex="-1"></a><span class="fu">print</span>(objname5)</span></code></pre></div>
<pre><code>## [1] SNO            DBNAME         PROTEIN_NAME   NO_DIMER       NO_RESIDUES   
## [6] DEPOSITED_DATE
## &lt;0 rows&gt; (or 0-length row.names)</code></pre>
<p>The subset function is used to filter the <code>proteindata</code> dataset to include only rows where <code>DBNAME</code> is <code>"PDB"</code> and <code>DIMER_COUNT</code> is greater than 3.</p>
<p>The filtered data is stored in <code>objname5</code></p>
<p>Now that we have the basic concept of how R works, lets introduce the Bioconductor package.</p>
</div>
</div>
<div id="introduction-to-bioconductor" class="section level2 hasAnchor" number="2.6">
<h2><span class="header-section-number">2.6</span> Introduction to Bioconductor<a href="bioconductor-package.html#introduction-to-bioconductor" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This open-source project provides tools for analyzing and understanding high-throughput genomic data. It includes a broad selection of packages for both bioinformatics and chemoinformatics. Before we explore the ChemmineR package in depth, let’s start with a few basic concepts.</p>
<div id="exploring-genomic-data" class="section level3 hasAnchor" number="2.6.1">
<h3><span class="header-section-number">2.6.1</span> Exploring Genomic data<a href="bioconductor-package.html#exploring-genomic-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <code>GenomicRanges</code> package is used to work with genomic intervals. It installed by:</p>
<pre><code>BiocManager::install(&quot;GenomicRanges&quot;)</code></pre>
<p>Lets now import the genomic ranges package to the program</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="bioconductor-package.html#cb30-1" tabindex="-1"></a><span class="fu">library</span>(GenomicRanges)</span></code></pre></div>
<p><code>GRanges</code> objects represent genomic ranges, allowing you to manipulate genomic intervals and align them with genomic data like gene annotations.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="bioconductor-package.html#cb31-1" tabindex="-1"></a><span class="co"># Create a simple genomic range object</span></span>
<span id="cb31-2"><a href="bioconductor-package.html#cb31-2" tabindex="-1"></a>gr <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="at">seqnames =</span> <span class="st">&quot;chr1&quot;</span>, <span class="at">ranges =</span> <span class="fu">IRanges</span>(<span class="at">start =</span> <span class="dv">100</span>, <span class="at">end =</span> <span class="dv">200</span>))</span>
<span id="cb31-3"><a href="bioconductor-package.html#cb31-3" tabindex="-1"></a>gr</span></code></pre></div>
<pre><code>## GRanges object with 1 range and 0 metadata columns:
##       seqnames    ranges strand
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]     chr1   100-200      *
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
</div>
<div id="visualizing-gene-expression-data" class="section level3 hasAnchor" number="2.6.2">
<h3><span class="header-section-number">2.6.2</span> Visualizing Gene Expression Data<a href="bioconductor-package.html#visualizing-gene-expression-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <code>DESeq2</code> package is used to find the differential gene expression analysis. It is widely used to identify genes that are differentially expressed under different conditions. Its is installed by;</p>
<pre><code>BiocManager::install(&quot;DESeq2&quot;)</code></pre>
<p>Load the library</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="bioconductor-package.html#cb34-1" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span></code></pre></div>
<p><code>DESeq2</code> handles RNA-Seq data to model and estimate the expression levels of genes, providing statistical tools for identifying differentially expressed genes.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="bioconductor-package.html#cb35-1" tabindex="-1"></a><span class="co"># Example dataset</span></span>
<span id="cb35-2"><a href="bioconductor-package.html#cb35-2" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">makeExampleDESeqDataSet</span>()</span>
<span id="cb35-3"><a href="bioconductor-package.html#cb35-3" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span></code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="bioconductor-package.html#cb42-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds)</span>
<span id="cb42-2"><a href="bioconductor-package.html#cb42-2" tabindex="-1"></a><span class="fu">plotMA</span>(res)</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
</div>
<div id="annotating-genomic-data" class="section level3 hasAnchor" number="2.6.3">
<h3><span class="header-section-number">2.6.3</span> Annotating Genomic Data<a href="bioconductor-package.html#annotating-genomic-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <code>AnnotationDbi</code> and <code>org.Hs.eg.db</code> packages are used to annotate gene identifiers with gene names, descriptions and other relevant metadata. The packages can be installed by;</p>
<pre><code>BiocManager::install(&quot;org.Hs.eg.db&quot;)</code></pre>
<p>Load the <code>org.Hs.eg.db</code> library.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="bioconductor-package.html#cb44-1" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span></code></pre></div>
<p>The packages provide a framework for mapping and retrieving biological data from databases. It’s crucial for adding biological context to your data, such as converting gene IDs to gene names</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="bioconductor-package.html#cb45-1" tabindex="-1"></a><span class="co"># Example: Map Entrez IDs to gene symbols</span></span>
<span id="cb45-2"><a href="bioconductor-package.html#cb45-2" tabindex="-1"></a>gene_ids <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;1&quot;</span>, <span class="st">&quot;2&quot;</span>, <span class="st">&quot;3&quot;</span>)</span>
<span id="cb45-3"><a href="bioconductor-package.html#cb45-3" tabindex="-1"></a>symbols <span class="ot">&lt;-</span> <span class="fu">mapIds</span>(org.Hs.eg.db, <span class="at">keys =</span> gene_ids, <span class="at">column =</span> <span class="st">&quot;SYMBOL&quot;</span>, <span class="at">keytype =</span> <span class="st">&quot;ENTREZID&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;select()&#39; returned 1:1 mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="bioconductor-package.html#cb47-1" tabindex="-1"></a>symbols</span></code></pre></div>
<pre><code>##       1       2       3 
##  &quot;A1BG&quot;   &quot;A2M&quot; &quot;A2MP1&quot;</code></pre>
</div>
<div id="working-with-sequence-data" class="section level3 hasAnchor" number="2.6.4">
<h3><span class="header-section-number">2.6.4</span> Working with Sequence Data<a href="bioconductor-package.html#working-with-sequence-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The popular `BioStrings package is used to analyze the protein, RNA and DNA sequences. The library is installed as the ones before.</p>
<pre><code>BiocManager::install(&quot;Biostrings&quot;)</code></pre>
<p>Load the library</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="bioconductor-package.html#cb50-1" tabindex="-1"></a><span class="fu">library</span>(Biostrings)</span></code></pre></div>
<p>Lets manipulate the DNA String <code>"ATCGGCTA"</code> by reverse complimenting.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="bioconductor-package.html#cb51-1" tabindex="-1"></a><span class="co"># Create a DNAString object</span></span>
<span id="cb51-2"><a href="bioconductor-package.html#cb51-2" tabindex="-1"></a>dna_seq <span class="ot">&lt;-</span> <span class="fu">DNAString</span>(<span class="st">&quot;ATCGGCTA&quot;</span>)</span>
<span id="cb51-3"><a href="bioconductor-package.html#cb51-3" tabindex="-1"></a><span class="fu">reverseComplement</span>(dna_seq)</span></code></pre></div>
<pre><code>## 8-letter DNAString object
## seq: TAGCCGAT</code></pre>
<p>This Bioconductor is an essential toolkit for anyone working with genomic data. Its packages are designed to facilitate a wide range of bioinformatics tasks, from basic sequence manipulation to complex statistical analyses.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="chemical-properties-prediction-and-visualized..html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/balsaedi/RCourse/edit/main/02-bioconductor-package.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/balsaedi/RCourse/blob/main/02-bioconductor-package.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
