<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Using seqinr | R Course</title>
  <meta name="description" content="Chapter 9 Using seqinr | R Course" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Using seqinr | R Course" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Using seqinr | R Course" />
  
  
  

<meta name="author" content="YOUR NAME HERE" />


<meta name="date" content="2024-04-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="bioconductor.html"/>
<link rel="next" href="using-short-read.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">SHORT TITLE HERE</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> R Course</a></li>
<li class="chapter" data-level="2" data-path="r---introduction.html"><a href="r---introduction.html"><i class="fa fa-check"></i><b>2</b> R - Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="r---introduction.html"><a href="r---introduction.html#r---basic-concepts-and-applications"><i class="fa fa-check"></i><b>2.1</b> R - Basic Concepts and applications</a></li>
<li class="chapter" data-level="2.2" data-path="r---introduction.html"><a href="r---introduction.html#why-r"><i class="fa fa-check"></i><b>2.2</b> Why R?</a></li>
<li class="chapter" data-level="2.3" data-path="r---introduction.html"><a href="r---introduction.html#r-studio"><i class="fa fa-check"></i><b>2.3</b> R studio</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="r---introduction.html"><a href="r---introduction.html#r---look-and-feel"><i class="fa fa-check"></i><b>2.3.1</b> R - Look and Feel</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="r---introduction.html"><a href="r---introduction.html#basic-mathematics-like-operators"><i class="fa fa-check"></i><b>2.4</b> Basic Mathematics like operators</a></li>
<li class="chapter" data-level="2.5" data-path="r---introduction.html"><a href="r---introduction.html#r-markdown"><i class="fa fa-check"></i><b>2.5</b> R Markdown</a></li>
<li class="chapter" data-level="2.6" data-path="r---introduction.html"><a href="r---introduction.html#installation-of-packages"><i class="fa fa-check"></i><b>2.6</b> Installation of packages</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html"><i class="fa fa-check"></i><b>3</b> Data Types and Data Structures</a>
<ul>
<li class="chapter" data-level="3.1" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#data-types"><i class="fa fa-check"></i><b>3.1</b> Data types</a></li>
<li class="chapter" data-level="3.2" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#data-structures"><i class="fa fa-check"></i><b>3.2</b> Data Structures</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#vector"><i class="fa fa-check"></i><b>3.2.1</b> Vector</a></li>
<li class="chapter" data-level="3.2.2" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#matrix"><i class="fa fa-check"></i><b>3.2.2</b> Matrix</a></li>
<li class="chapter" data-level="3.2.3" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#data-frame"><i class="fa fa-check"></i><b>3.2.3</b> Data frame</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="loops.html"><a href="loops.html"><i class="fa fa-check"></i><b>4</b> Loops</a>
<ul>
<li class="chapter" data-level="4.1" data-path="loops.html"><a href="loops.html#for-loops"><i class="fa fa-check"></i><b>4.1</b> For loops</a></li>
<li class="chapter" data-level="4.2" data-path="loops.html"><a href="loops.html#while-loop"><i class="fa fa-check"></i><b>4.2</b> While loop</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="working-with-files.html"><a href="working-with-files.html"><i class="fa fa-check"></i><b>5</b> Working with files</a>
<ul>
<li class="chapter" data-level="5.1" data-path="working-with-files.html"><a href="working-with-files.html#reading-a-file-from-a-dataframe"><i class="fa fa-check"></i><b>5.1</b> Reading a file from a dataframe</a></li>
<li class="chapter" data-level="5.2" data-path="working-with-files.html"><a href="working-with-files.html#write-a-new-table-to-a-file-locally"><i class="fa fa-check"></i><b>5.2</b> Write a new table to a file locally</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="dpyr-library.html"><a href="dpyr-library.html"><i class="fa fa-check"></i><b>6</b> Dpyr library</a>
<ul>
<li class="chapter" data-level="6.1" data-path="dpyr-library.html"><a href="dpyr-library.html#application-of-dplyr-on-microbiology"><i class="fa fa-check"></i><b>6.1</b> Application of Dplyr on Microbiology</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="ggplot.html"><a href="ggplot.html"><i class="fa fa-check"></i><b>7</b> GGPLOT</a>
<ul>
<li class="chapter" data-level="7.1" data-path="ggplot.html"><a href="ggplot.html#five-key-steps-to-ggplot"><i class="fa fa-check"></i><b>7.1</b> Five key steps to ggplot</a></li>
<li class="chapter" data-level="7.2" data-path="ggplot.html"><a href="ggplot.html#biology-behind-the-data-we-are-using"><i class="fa fa-check"></i><b>7.2</b> Biology behind the data we are using</a></li>
<li class="chapter" data-level="7.3" data-path="ggplot.html"><a href="ggplot.html#prepare-the-data-for-plotting---data-assessment"><i class="fa fa-check"></i><b>7.3</b> Prepare the data for plotting - Data Assessment</a></li>
<li class="chapter" data-level="7.4" data-path="ggplot.html"><a href="ggplot.html#plotting"><i class="fa fa-check"></i><b>7.4</b> Plotting</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="ggplot.html"><a href="ggplot.html#base-set-up"><i class="fa fa-check"></i><b>7.4.1</b> Base set up</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="ggplot.html"><a href="ggplot.html#working-with-inbuilt-iris-data"><i class="fa fa-check"></i><b>7.5</b> Working with inbuilt iris data</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="bioconductor.html"><a href="bioconductor.html"><i class="fa fa-check"></i><b>8</b> Bioconductor</a>
<ul>
<li class="chapter" data-level="8.1" data-path="bioconductor.html"><a href="bioconductor.html#installing-biocmanager-package"><i class="fa fa-check"></i><b>8.1</b> Installing BiocManager Package</a></li>
<li class="chapter" data-level="8.2" data-path="bioconductor.html"><a href="bioconductor.html#searching-and-analyzing-the-strings"><i class="fa fa-check"></i><b>8.2</b> Searching and Analyzing the strings</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="using-seqinr.html"><a href="using-seqinr.html"><i class="fa fa-check"></i><b>9</b> Using seqinr</a></li>
<li class="chapter" data-level="10" data-path="using-short-read.html"><a href="using-short-read.html"><i class="fa fa-check"></i><b>10</b> Using Short Read</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">R Course</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="using-seqinr" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> Using seqinr<a href="using-seqinr.html#using-seqinr" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Another useful library in sequence analysis is <code>seqinr</code>.
After installing the library by invoking <code>install.packages("seqinr")</code> to the console, load it to the workspace</p>
<div class="sourceCode" id="cb334"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb334-1"><a href="using-seqinr.html#cb334-1" tabindex="-1"></a><span class="fu">library</span>(seqinr)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;seqinr&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:Biostrings&#39;:
## 
##     translate</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     count</code></pre>
<p>Reading the fasta file containing the protein sequence using the <code>read.fasta</code> function.</p>
<div class="sourceCode" id="cb338"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb338-1"><a href="using-seqinr.html#cb338-1" tabindex="-1"></a>proseq <span class="ot">=</span> <span class="fu">read.fasta</span>(<span class="at">file=</span><span class="st">&quot;data/spike_multifasta.txt&quot;</span>, <span class="at">seqtype=</span><span class="st">&quot;AA&quot;</span>)</span>
<span id="cb338-2"><a href="using-seqinr.html#cb338-2" tabindex="-1"></a><span class="fu">str</span>(proseq)</span></code></pre></div>
<pre><code>## List of 6
##  $ Spike_SARS_CoV2_2019         : &#39;SeqFastaAA&#39; chr [1:1273] &quot;M&quot; &quot;F&quot; &quot;V&quot; &quot;F&quot; ...
##   ..- attr(*, &quot;name&quot;)= chr &quot;Spike_SARS_CoV2_2019&quot;
##   ..- attr(*, &quot;Annot&quot;)= chr &quot;&gt;Spike_SARS_CoV2_2019&quot;
##  $ Spike_SARS_CoV               : &#39;SeqFastaAA&#39; chr [1:1255] &quot;M&quot; &quot;F&quot; &quot;I&quot; &quot;F&quot; ...
##   ..- attr(*, &quot;name&quot;)= chr &quot;Spike_SARS_CoV&quot;
##   ..- attr(*, &quot;Annot&quot;)= chr &quot;&gt;Spike_SARS_CoV&quot;
##  $ Spike_Bat_coronavirus_RaTG13 : &#39;SeqFastaAA&#39; chr [1:1269] &quot;M&quot; &quot;F&quot; &quot;V&quot; &quot;F&quot; ...
##   ..- attr(*, &quot;name&quot;)= chr &quot;Spike_Bat_coronavirus_RaTG13&quot;
##   ..- attr(*, &quot;Annot&quot;)= chr &quot;&gt;Spike_Bat_coronavirus_RaTG13&quot;
##  $ Spike_Pangolin_coronavirus_GX: &#39;SeqFastaAA&#39; chr [1:1269] &quot;M&quot; &quot;F&quot; &quot;V&quot; &quot;F&quot; ...
##   ..- attr(*, &quot;name&quot;)= chr &quot;Spike_Pangolin_coronavirus_GX&quot;
##   ..- attr(*, &quot;Annot&quot;)= chr &quot;&gt;Spike_Pangolin_coronavirus_GX&quot;
##  $ Spike_Pangolin_coronavirus_GD: &#39;SeqFastaAA&#39; chr [1:1265] &quot;M&quot; &quot;L&quot; &quot;F&quot; &quot;F&quot; ...
##   ..- attr(*, &quot;name&quot;)= chr &quot;Spike_Pangolin_coronavirus_GD&quot;
##   ..- attr(*, &quot;Annot&quot;)= chr &quot;&gt;Spike_Pangolin_coronavirus_GD&quot;
##  $ Spike_WIV1_Bat_Coronovirus   : &#39;SeqFastaAA&#39; chr [1:1256] &quot;M&quot; &quot;K&quot; &quot;L&quot; &quot;L&quot; ...
##   ..- attr(*, &quot;name&quot;)= chr &quot;Spike_WIV1_Bat_Coronovirus&quot;
##   ..- attr(*, &quot;Annot&quot;)= chr &quot;&gt;Spike_WIV1_Bat_Coronovirus&quot;</code></pre>
<p>Count the number of sequences</p>
<div class="sourceCode" id="cb340"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb340-1"><a href="using-seqinr.html#cb340-1" tabindex="-1"></a>num_seq <span class="ot">=</span> <span class="fu">length</span>(proseq)</span>
<span id="cb340-2"><a href="using-seqinr.html#cb340-2" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&quot;Number of sequences = &quot;</span>, num_seq))</span></code></pre></div>
<pre><code>## [1] &quot;Number of sequences =  6&quot;</code></pre>
<ul>
<li>Names of the sequences</li>
</ul>
<div class="sourceCode" id="cb342"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb342-1"><a href="using-seqinr.html#cb342-1" tabindex="-1"></a>names<span class="ot">=</span><span class="fu">getName</span>(proseq)</span>
<span id="cb342-2"><a href="using-seqinr.html#cb342-2" tabindex="-1"></a>names </span></code></pre></div>
<pre><code>## [1] &quot;Spike_SARS_CoV2_2019&quot;          &quot;Spike_SARS_CoV&quot;               
## [3] &quot;Spike_Bat_coronavirus_RaTG13&quot;  &quot;Spike_Pangolin_coronavirus_GX&quot;
## [5] &quot;Spike_Pangolin_coronavirus_GD&quot; &quot;Spike_WIV1_Bat_Coronovirus&quot;</code></pre>
<ul>
<li>Annotation of sequences</li>
</ul>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb344-1"><a href="using-seqinr.html#cb344-1" tabindex="-1"></a>annotation <span class="ot">&lt;-</span> <span class="fu">getAnnot</span>(proseq)</span>
<span id="cb344-2"><a href="using-seqinr.html#cb344-2" tabindex="-1"></a>annotation</span></code></pre></div>
<pre><code>## [[1]]
## [1] &quot;&gt;Spike_SARS_CoV2_2019&quot;
## 
## [[2]]
## [1] &quot;&gt;Spike_SARS_CoV&quot;
## 
## [[3]]
## [1] &quot;&gt;Spike_Bat_coronavirus_RaTG13&quot;
## 
## [[4]]
## [1] &quot;&gt;Spike_Pangolin_coronavirus_GX&quot;
## 
## [[5]]
## [1] &quot;&gt;Spike_Pangolin_coronavirus_GD&quot;
## 
## [[6]]
## [1] &quot;&gt;Spike_WIV1_Bat_Coronovirus&quot;</code></pre>
<ul>
<li>Length of individual sequences</li>
</ul>
<div class="sourceCode" id="cb346"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb346-1"><a href="using-seqinr.html#cb346-1" tabindex="-1"></a>len_seq <span class="ot">=</span> <span class="fu">getLength</span>(proseq)</span>
<span id="cb346-2"><a href="using-seqinr.html#cb346-2" tabindex="-1"></a>len_seq</span></code></pre></div>
<pre><code>## [1] 1273 1255 1269 1269 1265 1256</code></pre>
<ul>
<li>Protein statistics</li>
</ul>
<div class="sourceCode" id="cb348"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb348-1"><a href="using-seqinr.html#cb348-1" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">AAstat</span>(proseq[[<span class="dv">1</span>]]))</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-149-1.png" width="672" /></p>
<pre><code>## $Compo
## 
##   *   A   C   D   E   F   G   H   I   K   L   M   N   P   Q   R   S   T   V   W 
##   0  79  40  62  48  77  82  17  76  61 108  14  88  58  62  42  99  97  97  12 
##   Y 
##  54 
## 
## $Prop
## $Prop$Tiny
## [1] 0.3118617
## 
## $Prop$Small
## [1] 0.5514533
## 
## $Prop$Aliphatic
## [1] 0.2207384
## 
## $Prop$Aromatic
## [1] 0.1256874
## 
## $Prop$Non.polar
## [1] 0.5475255
## 
## $Prop$Polar
## [1] 0.4524745
## 
## $Prop$Charged
## [1] 0.1806756
## 
## $Prop$Basic
## [1] 0.09426551
## 
## $Prop$Acidic
## [1] 0.08641005
## 
## 
## $Pi
## [1] 6.236478</code></pre>
<p>It gives the type of proteins</p>
<div class="sourceCode" id="cb350"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb350-1"><a href="using-seqinr.html#cb350-1" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>num_seq)</span>
<span id="cb350-2"><a href="using-seqinr.html#cb350-2" tabindex="-1"></a>{</span>
<span id="cb350-3"><a href="using-seqinr.html#cb350-3" tabindex="-1"></a>  <span class="fu">print</span>(names[i])</span>
<span id="cb350-4"><a href="using-seqinr.html#cb350-4" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">AAstat</span>(proseq[[i]]))</span>
<span id="cb350-5"><a href="using-seqinr.html#cb350-5" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## [1] &quot;Spike_SARS_CoV2_2019&quot;</code></pre>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-150-1.png" width="672" /></p>
<pre><code>## $Compo
## 
##   *   A   C   D   E   F   G   H   I   K   L   M   N   P   Q   R   S   T   V   W 
##   0  79  40  62  48  77  82  17  76  61 108  14  88  58  62  42  99  97  97  12 
##   Y 
##  54 
## 
## $Prop
## $Prop$Tiny
## [1] 0.3118617
## 
## $Prop$Small
## [1] 0.5514533
## 
## $Prop$Aliphatic
## [1] 0.2207384
## 
## $Prop$Aromatic
## [1] 0.1256874
## 
## $Prop$Non.polar
## [1] 0.5475255
## 
## $Prop$Polar
## [1] 0.4524745
## 
## $Prop$Charged
## [1] 0.1806756
## 
## $Prop$Basic
## [1] 0.09426551
## 
## $Prop$Acidic
## [1] 0.08641005
## 
## 
## $Pi
## [1] 6.236478
## 
## [1] &quot;Spike_SARS_CoV&quot;</code></pre>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-150-2.png" width="672" /></p>
<pre><code>## $Compo
## 
##   *   A   C   D   E   F   G   H   I   K   L   M   N   P   Q   R   S   T   V   W 
##   0  84  39  74  42  83  78  15  77  60  99  20  81  57  55  39  96 100  91  11 
##   Y 
##  54 
## 
## $Prop
## $Prop$Tiny
## [1] 0.3163347
## 
## $Prop$Small
## [1] 0.5577689
## 
## $Prop$Aliphatic
## [1] 0.212749
## 
## $Prop$Aromatic
## [1] 0.1298805
## 
## $Prop$Non.polar
## [1] 0.5521912
## 
## $Prop$Polar
## [1] 0.4478088
## 
## $Prop$Charged
## [1] 0.1832669
## 
## $Prop$Basic
## [1] 0.09083665
## 
## $Prop$Acidic
## [1] 0.09243028
## 
## 
## $Pi
## [1] 5.510607
## 
## [1] &quot;Spike_Bat_coronavirus_RaTG13&quot;</code></pre>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-150-3.png" width="672" /></p>
<pre><code>## $Compo
## 
##   *   A   C   D   E   F   G   H   I   K   L   M   N   P   Q   R   S   T   V   W 
##   0  80  40  64  47  75  82  18  79  63 109  14  85  58  60  39  97  98  94  12 
##   Y 
##  55 
## 
## $Prop
## $Prop$Tiny
## [1] 0.3128448
## 
## $Prop$Small
## [1] 0.5500394
## 
## $Prop$Aliphatic
## [1] 0.2222222
## 
## $Prop$Aromatic
## [1] 0.1260835
## 
## $Prop$Non.polar
## [1] 0.5500394
## 
## $Prop$Polar
## [1] 0.4499606
## 
## $Prop$Charged
## [1] 0.1820331
## 
## $Prop$Basic
## [1] 0.09456265
## 
## $Prop$Acidic
## [1] 0.08747045
## 
## 
## $Pi
## [1] 6.11311
## 
## [1] &quot;Spike_Pangolin_coronavirus_GX&quot;</code></pre>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-150-4.png" width="672" /></p>
<pre><code>## $Compo
## 
##   *   A   C   D   E   F   G   H   I   K   L   M   N   P   Q   R   S   T   V   W 
##   0  76  40  61  48  75  85  17  79  61 113  15  82  56  62  41  94  98  97  12 
##   Y 
##  55 
## 
## $Prop
## $Prop$Tiny
## [1] 0.3096927
## 
## $Prop$Small
## [1] 0.5429472
## 
## $Prop$Aliphatic
## [1] 0.2277384
## 
## $Prop$Aromatic
## [1] 0.1252955
## 
## $Prop$Non.polar
## [1] 0.5539795
## 
## $Prop$Polar
## [1] 0.4444444
## 
## $Prop$Charged
## [1] 0.179669
## 
## $Prop$Basic
## [1] 0.09377463
## 
## $Prop$Acidic
## [1] 0.08589441
## 
## 
## $Pi
## [1] 6.235931
## 
## [1] &quot;Spike_Pangolin_coronavirus_GD&quot;</code></pre>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-150-5.png" width="672" /></p>
<pre><code>## $Compo
## 
##   *   A   C   D   E   F   G   H   I   K   L   M   N   P   Q   R   S   T   V   W 
##   0  82  40  60  48  76  81  14  80  58 103  15  90  54  58  40 100  99  95  12 
##   Y 
##  60 
## 
## $Prop
## $Prop$Tiny
## [1] 0.3177866
## 
## $Prop$Small
## [1] 0.5541502
## 
## $Prop$Aliphatic
## [1] 0.2197628
## 
## $Prop$Aromatic
## [1] 0.1280632
## 
## $Prop$Non.polar
## [1] 0.5517787
## 
## $Prop$Polar
## [1] 0.4482213
## 
## $Prop$Charged
## [1] 0.173913
## 
## $Prop$Basic
## [1] 0.08853755
## 
## $Prop$Acidic
## [1] 0.08537549
## 
## 
## $Pi
## [1] 5.900948
## 
## [1] &quot;Spike_WIV1_Bat_Coronovirus&quot;</code></pre>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-150-6.png" width="672" /></p>
<pre><code>## $Compo
## 
##   *   A   C   D   E   F   G   H   I   K   L   M   N   P   Q   R   S   T   V   W 
##   0  82  39  72  43  83  77  15  77  60 102  17  86  62  54  42  98  93  93  11 
##   Y 
##  50 
## 
## $Prop
## $Prop$Tiny
## [1] 0.3097134
## 
## $Prop$Small
## [1] 0.5589172
## 
## $Prop$Aliphatic
## [1] 0.2165605
## 
## $Prop$Aromatic
## [1] 0.1265924
## 
## $Prop$Non.polar
## [1] 0.5517516
## 
## $Prop$Polar
## [1] 0.4482484
## 
## $Prop$Charged
## [1] 0.1847134
## 
## $Prop$Basic
## [1] 0.09315287
## 
## $Prop$Acidic
## [1] 0.09156051
## 
## 
## $Pi
## [1] 5.739146</code></pre>
<ul>
<li>Show commands in seqinr</li>
</ul>
<div class="sourceCode" id="cb358"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb358-1"><a href="using-seqinr.html#cb358-1" tabindex="-1"></a><span class="fu">lseqinr</span>()</span></code></pre></div>
<pre><code>##   [1] &quot;a&quot;                       &quot;aaa&quot;                    
##   [3] &quot;AAstat&quot;                  &quot;acnucclose&quot;             
##   [5] &quot;acnucopen&quot;               &quot;al2bp&quot;                  
##   [7] &quot;alllistranks&quot;            &quot;alr&quot;                    
##   [9] &quot;amb&quot;                     &quot;as.alignment&quot;           
##  [11] &quot;as.matrix.alignment&quot;     &quot;as.SeqAcnucWeb&quot;         
##  [13] &quot;as.SeqFastaAA&quot;           &quot;as.SeqFastadna&quot;         
##  [15] &quot;as.SeqFrag&quot;              &quot;autosocket&quot;             
##  [17] &quot;baselineabif&quot;            &quot;bma&quot;                    
##  [19] &quot;c2s&quot;                     &quot;cai&quot;                    
##  [21] &quot;cfl&quot;                     &quot;choosebank&quot;             
##  [23] &quot;circle&quot;                  &quot;clfcd&quot;                  
##  [25] &quot;clientid&quot;                &quot;closebank&quot;              
##  [27] &quot;col2alpha&quot;               &quot;comp&quot;                   
##  [29] &quot;computePI&quot;               &quot;con&quot;                    
##  [31] &quot;consensus&quot;               &quot;count&quot;                  
##  [33] &quot;countfreelists&quot;          &quot;countsubseqs&quot;           
##  [35] &quot;crelistfromclientdata&quot;   &quot;css&quot;                    
##  [37] &quot;dia.bactgensize&quot;         &quot;dia.db.growth&quot;          
##  [39] &quot;dist.alignment&quot;          &quot;dotchart.uco&quot;           
##  [41] &quot;dotPlot&quot;                 &quot;draw.oriloc&quot;            
##  [43] &quot;draw.rearranged.oriloc&quot;  &quot;draw.recstat&quot;           
##  [45] &quot;exseq&quot;                   &quot;extract.breakpoints&quot;    
##  [47] &quot;extractseqs&quot;             &quot;fastacc&quot;                
##  [49] &quot;gb2fasta&quot;                &quot;gbk2g2&quot;                 
##  [51] &quot;gbk2g2.euk&quot;              &quot;GC&quot;                     
##  [53] &quot;GC1&quot;                     &quot;GC2&quot;                    
##  [55] &quot;GC3&quot;                     &quot;GCpos&quot;                  
##  [57] &quot;get.db.growth&quot;           &quot;getAnnot&quot;               
##  [59] &quot;getAnnot.default&quot;        &quot;getAnnot.list&quot;          
##  [61] &quot;getAnnot.logical&quot;        &quot;getAnnot.qaw&quot;           
##  [63] &quot;getAnnot.SeqAcnucWeb&quot;    &quot;getAnnot.SeqFastaAA&quot;    
##  [65] &quot;getAnnot.SeqFastadna&quot;    &quot;getAttributsocket&quot;      
##  [67] &quot;getFrag&quot;                 &quot;getFrag.character&quot;      
##  [69] &quot;getFrag.default&quot;         &quot;getFrag.list&quot;           
##  [71] &quot;getFrag.logical&quot;         &quot;getFrag.qaw&quot;            
##  [73] &quot;getFrag.SeqAcnucWeb&quot;     &quot;getFrag.SeqFastaAA&quot;     
##  [75] &quot;getFrag.SeqFastadna&quot;     &quot;getFrag.SeqFrag&quot;        
##  [77] &quot;getKeyword&quot;              &quot;getKeyword.default&quot;     
##  [79] &quot;getKeyword.list&quot;         &quot;getKeyword.logical&quot;     
##  [81] &quot;getKeyword.qaw&quot;          &quot;getKeyword.SeqAcnucWeb&quot; 
##  [83] &quot;getLength&quot;               &quot;getLength.character&quot;    
##  [85] &quot;getLength.default&quot;       &quot;getLength.list&quot;         
##  [87] &quot;getLength.logical&quot;       &quot;getLength.qaw&quot;          
##  [89] &quot;getLength.SeqAcnucWeb&quot;   &quot;getLength.SeqFastaAA&quot;   
##  [91] &quot;getLength.SeqFastadna&quot;   &quot;getLength.SeqFrag&quot;      
##  [93] &quot;getlistrank&quot;             &quot;getliststate&quot;           
##  [95] &quot;getLocation&quot;             &quot;getLocation.default&quot;    
##  [97] &quot;getLocation.list&quot;        &quot;getLocation.logical&quot;    
##  [99] &quot;getLocation.qaw&quot;         &quot;getLocation.SeqAcnucWeb&quot;
## [101] &quot;getName&quot;                 &quot;getName.default&quot;        
## [103] &quot;getName.list&quot;            &quot;getName.logical&quot;        
## [105] &quot;getName.qaw&quot;             &quot;getName.SeqAcnucWeb&quot;    
## [107] &quot;getName.SeqFastaAA&quot;      &quot;getName.SeqFastadna&quot;    
## [109] &quot;getName.SeqFrag&quot;         &quot;getNumber.socket&quot;       
## [111] &quot;getSequence&quot;             &quot;getSequence.character&quot;  
## [113] &quot;getSequence.default&quot;     &quot;getSequence.list&quot;       
## [115] &quot;getSequence.logical&quot;     &quot;getSequence.qaw&quot;        
## [117] &quot;getSequence.SeqAcnucWeb&quot; &quot;getSequence.SeqFastaAA&quot; 
## [119] &quot;getSequence.SeqFastadna&quot; &quot;getSequence.SeqFrag&quot;    
## [121] &quot;getTrans&quot;                &quot;getTrans.character&quot;     
## [123] &quot;getTrans.default&quot;        &quot;getTrans.list&quot;          
## [125] &quot;getTrans.logical&quot;        &quot;getTrans.qaw&quot;           
## [127] &quot;getTrans.SeqAcnucWeb&quot;    &quot;getTrans.SeqFastadna&quot;   
## [129] &quot;getTrans.SeqFrag&quot;        &quot;getType&quot;                
## [131] &quot;gfrag&quot;                   &quot;ghelp&quot;                  
## [133] &quot;gln&quot;                     &quot;glr&quot;                    
## [135] &quot;gls&quot;                     &quot;is.SeqAcnucWeb&quot;         
## [137] &quot;is.SeqFastaAA&quot;           &quot;is.SeqFastadna&quot;         
## [139] &quot;is.SeqFrag&quot;              &quot;isenum&quot;                 
## [141] &quot;isn&quot;                     &quot;kaks&quot;                   
## [143] &quot;kdb&quot;                     &quot;knowndbs&quot;               
## [145] &quot;lseqinr&quot;                 &quot;modifylist&quot;             
## [147] &quot;move&quot;                    &quot;mv&quot;                     
## [149] &quot;n2s&quot;                     &quot;oriloc&quot;                 
## [151] &quot;parser.socket&quot;           &quot;peakabif&quot;               
## [153] &quot;permutation&quot;             &quot;pga&quot;                    
## [155] &quot;plot.SeqAcnucWeb&quot;        &quot;plotabif&quot;               
## [157] &quot;plotladder&quot;              &quot;plotPanels&quot;             
## [159] &quot;pmw&quot;                     &quot;prepgetannots&quot;          
## [161] &quot;prettyseq&quot;               &quot;print.qaw&quot;              
## [163] &quot;print.SeqAcnucWeb&quot;       &quot;query&quot;                  
## [165] &quot;quitacnuc&quot;               &quot;read.abif&quot;              
## [167] &quot;read.alignment&quot;          &quot;read.fasta&quot;             
## [169] &quot;readBins&quot;                &quot;readfirstrec&quot;           
## [171] &quot;readPanels&quot;              &quot;readsmj&quot;                
## [173] &quot;rearranged.oriloc&quot;       &quot;recstat&quot;                
## [175] &quot;residuecount&quot;            &quot;reverse.align&quot;          
## [177] &quot;rho&quot;                     &quot;rot13&quot;                  
## [179] &quot;s2c&quot;                     &quot;s2n&quot;                    
## [181] &quot;savelist&quot;                &quot;SEQINR.UTIL&quot;            
## [183] &quot;setlistname&quot;             &quot;splitseq&quot;               
## [185] &quot;stresc&quot;                  &quot;stutterabif&quot;            
## [187] &quot;summary.SeqFastaAA&quot;      &quot;summary.SeqFastadna&quot;    
## [189] &quot;swap&quot;                    &quot;syncodons&quot;              
## [191] &quot;synsequence&quot;             &quot;tablecode&quot;              
## [193] &quot;test.co.recstat&quot;         &quot;test.li.recstat&quot;        
## [195] &quot;translate&quot;               &quot;trimSpace&quot;              
## [197] &quot;uco&quot;                     &quot;ucoweight&quot;              
## [199] &quot;where.is.this.acc&quot;       &quot;words&quot;                  
## [201] &quot;words.pos&quot;               &quot;write.fasta&quot;            
## [203] &quot;zscore&quot;</code></pre>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="bioconductor.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="using-short-read.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/balsaedi/RCourse/edit/main/bioconductor_library_Advance-R.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/balsaedi/RCourse/blob/main/bioconductor_library_Advance-R.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
