<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Working with files | R Course</title>
  <meta name="description" content="Chapter 4 Working with files | R Course" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Working with files | R Course" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Working with files | R Course" />
  
  
  

<meta name="author" content="YOUR NAME HERE" />


<meta name="date" content="2024-07-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="loops.html"/>
<link rel="next" href="dpyr-library.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">SHORT TITLE HERE</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> R - Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#r---basic-concepts-and-applications"><i class="fa fa-check"></i><b>1.1</b> R - Basic Concepts and applications</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#why-r"><i class="fa fa-check"></i><b>1.2</b> Why R?</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#r-studio"><i class="fa fa-check"></i><b>1.3</b> R studio</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#rstudio---look-and-feel"><i class="fa fa-check"></i><b>1.3.1</b> RStudio - Look and Feel</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#basic-mathematics-like-operators"><i class="fa fa-check"></i><b>1.4</b> Basic Mathematics like operators</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="index.html"><a href="index.html#basics"><i class="fa fa-check"></i><b>1.4.1</b> Basics</a></li>
<li class="chapter" data-level="1.4.2" data-path="index.html"><a href="index.html#bodmas-rule"><i class="fa fa-check"></i><b>1.4.2</b> BODMAS rule</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#r-markdown"><i class="fa fa-check"></i><b>1.5</b> R Markdown</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#installation-of-packages"><i class="fa fa-check"></i><b>1.6</b> Installation of packages</a></li>
<li class="chapter" data-level="1.7" data-path="index.html"><a href="index.html#assignment"><i class="fa fa-check"></i><b>1.7</b> Assignment</a></li>
<li class="chapter" data-level="1.8" data-path="index.html"><a href="index.html#code-challenge"><i class="fa fa-check"></i><b>1.8</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html"><i class="fa fa-check"></i><b>2</b> Data Types and Data Structures</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#data-types"><i class="fa fa-check"></i><b>2.1</b> Data types</a></li>
<li class="chapter" data-level="2.2" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#data-structures"><i class="fa fa-check"></i><b>2.2</b> Data Structures</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#vector"><i class="fa fa-check"></i><b>2.2.1</b> Vector</a></li>
<li class="chapter" data-level="2.2.2" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#matrix"><i class="fa fa-check"></i><b>2.2.2</b> Matrix</a></li>
<li class="chapter" data-level="2.2.3" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#data-frame"><i class="fa fa-check"></i><b>2.2.3</b> Data frame</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="data-types-and-data-structures.html"><a href="data-types-and-data-structures.html#code-challenge-1"><i class="fa fa-check"></i><b>2.3</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="loops.html"><a href="loops.html"><i class="fa fa-check"></i><b>3</b> Loops</a>
<ul>
<li class="chapter" data-level="3.1" data-path="loops.html"><a href="loops.html#for-loops"><i class="fa fa-check"></i><b>3.1</b> For loops</a></li>
<li class="chapter" data-level="3.2" data-path="loops.html"><a href="loops.html#while-loop"><i class="fa fa-check"></i><b>3.2</b> While loop</a></li>
<li class="chapter" data-level="3.3" data-path="loops.html"><a href="loops.html#code-challenge-2"><i class="fa fa-check"></i><b>3.3</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="working-with-files.html"><a href="working-with-files.html"><i class="fa fa-check"></i><b>4</b> Working with files</a>
<ul>
<li class="chapter" data-level="4.1" data-path="working-with-files.html"><a href="working-with-files.html#xlsx-xls-files"><i class="fa fa-check"></i><b>4.1</b> .XLSX, XLS files</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="working-with-files.html"><a href="working-with-files.html#reading-a-file-to-a-dataframe"><i class="fa fa-check"></i><b>4.1.1</b> Reading a file to a dataframe</a></li>
<li class="chapter" data-level="4.1.2" data-path="working-with-files.html"><a href="working-with-files.html#write-a-new-table-to-a-file-locally"><i class="fa fa-check"></i><b>4.1.2</b> Write a new table to a file locally</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="working-with-files.html"><a href="working-with-files.html#csv-files"><i class="fa fa-check"></i><b>4.2</b> .CSV files</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="working-with-files.html"><a href="working-with-files.html#titanic-dataset"><i class="fa fa-check"></i><b>4.2.1</b> Titanic dataset</a></li>
<li class="chapter" data-level="4.2.2" data-path="working-with-files.html"><a href="working-with-files.html#ibm-employee-attrition-data"><i class="fa fa-check"></i><b>4.2.2</b> IBM Employee Attrition data</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="working-with-files.html"><a href="working-with-files.html#code-challenge-3"><i class="fa fa-check"></i><b>4.3</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="dpyr-library.html"><a href="dpyr-library.html"><i class="fa fa-check"></i><b>5</b> Dpyr library</a>
<ul>
<li class="chapter" data-level="5.1" data-path="dpyr-library.html"><a href="dpyr-library.html#dplyr-verbs"><i class="fa fa-check"></i><b>5.1</b> Dplyr Verbs</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="dpyr-library.html"><a href="dpyr-library.html#select"><i class="fa fa-check"></i><b>5.1.1</b> select</a></li>
<li class="chapter" data-level="5.1.2" data-path="dpyr-library.html"><a href="dpyr-library.html#filter"><i class="fa fa-check"></i><b>5.1.2</b> filter</a></li>
<li class="chapter" data-level="5.1.3" data-path="dpyr-library.html"><a href="dpyr-library.html#arrange"><i class="fa fa-check"></i><b>5.1.3</b> arrange</a></li>
<li class="chapter" data-level="5.1.4" data-path="dpyr-library.html"><a href="dpyr-library.html#mutate"><i class="fa fa-check"></i><b>5.1.4</b> mutate</a></li>
<li class="chapter" data-level="5.1.5" data-path="dpyr-library.html"><a href="dpyr-library.html#group_by"><i class="fa fa-check"></i><b>5.1.5</b> group_by</a></li>
<li class="chapter" data-level="5.1.6" data-path="dpyr-library.html"><a href="dpyr-library.html#summarise"><i class="fa fa-check"></i><b>5.1.6</b> Summarise</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="dpyr-library.html"><a href="dpyr-library.html#chaining"><i class="fa fa-check"></i><b>5.2</b> Chaining</a></li>
<li class="chapter" data-level="5.3" data-path="dpyr-library.html"><a href="dpyr-library.html#application-of-dplyr-on-microbiology"><i class="fa fa-check"></i><b>5.3</b> Application of Dplyr on Microbiology</a></li>
<li class="chapter" data-level="5.4" data-path="dpyr-library.html"><a href="dpyr-library.html#code-challenge-4"><i class="fa fa-check"></i><b>5.4</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="ggplot.html"><a href="ggplot.html"><i class="fa fa-check"></i><b>6</b> GGPLOT</a>
<ul>
<li class="chapter" data-level="6.1" data-path="ggplot.html"><a href="ggplot.html#five-key-steps-to-ggplot"><i class="fa fa-check"></i><b>6.1</b> Five key steps to ggplot</a></li>
<li class="chapter" data-level="6.2" data-path="ggplot.html"><a href="ggplot.html#prepare-the-data-for-plotting---data-assessment"><i class="fa fa-check"></i><b>6.2</b> Prepare the data for plotting - Data Assessment</a></li>
<li class="chapter" data-level="6.3" data-path="ggplot.html"><a href="ggplot.html#plotting"><i class="fa fa-check"></i><b>6.3</b> Plotting</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="ggplot.html"><a href="ggplot.html#base-set-up"><i class="fa fa-check"></i><b>6.3.1</b> Base set up</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="ggplot.html"><a href="ggplot.html#working-with-inbuilt-iris-data"><i class="fa fa-check"></i><b>6.4</b> Working with inbuilt iris data</a></li>
<li class="chapter" data-level="6.5" data-path="ggplot.html"><a href="ggplot.html#code-challenge-5"><i class="fa fa-check"></i><b>6.5</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="bioconductor.html"><a href="bioconductor.html"><i class="fa fa-check"></i><b>7</b> Bioconductor</a>
<ul>
<li class="chapter" data-level="7.1" data-path="bioconductor.html"><a href="bioconductor.html#installing-biocmanager-package"><i class="fa fa-check"></i><b>7.1</b> Installing BiocManager Package</a></li>
<li class="chapter" data-level="7.2" data-path="bioconductor.html"><a href="bioconductor.html#searching-and-analyzing-the-strings"><i class="fa fa-check"></i><b>7.2</b> Searching and Analyzing the strings</a></li>
<li class="chapter" data-level="7.3" data-path="bioconductor.html"><a href="bioconductor.html#using-seqinr"><i class="fa fa-check"></i><b>7.3</b> Using seqinr</a></li>
<li class="chapter" data-level="7.4" data-path="bioconductor.html"><a href="bioconductor.html#using-short-read"><i class="fa fa-check"></i><b>7.4</b> Using Short Read</a></li>
<li class="chapter" data-level="7.5" data-path="bioconductor.html"><a href="bioconductor.html#genomic-ranges"><i class="fa fa-check"></i><b>7.5</b> Genomic Ranges</a></li>
<li class="chapter" data-level="7.6" data-path="bioconductor.html"><a href="bioconductor.html#using-deseq-package"><i class="fa fa-check"></i><b>7.6</b> Using `DESeq package</a></li>
<li class="chapter" data-level="7.7" data-path="bioconductor.html"><a href="bioconductor.html#network-analysis-of-genomic-data"><i class="fa fa-check"></i><b>7.7</b> Network Analysis of Genomic Data</a></li>
<li class="chapter" data-level="7.8" data-path="bioconductor.html"><a href="bioconductor.html#code-challenge-6"><i class="fa fa-check"></i><b>7.8</b> CODE CHALLENGE</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">R Course</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="working-with-files" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> Working with files<a href="working-with-files.html#working-with-files" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="xlsx-xls-files" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> .XLSX, XLS files<a href="working-with-files.html#xlsx-xls-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="reading-a-file-to-a-dataframe" class="section level3 hasAnchor" number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> Reading a file to a dataframe<a href="working-with-files.html#reading-a-file-to-a-dataframe" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This chapter explains how to read a file to a data frame using the <code>readxl</code> library. The <code>readxl</code> package can be installed from the console by <code>install.packages("readxl")</code> command.</p>
<p>Lets begin by importing the library</p>
<div class="sourceCode" id="cb299"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb299-1"><a href="working-with-files.html#cb299-1" tabindex="-1"></a><span class="co"># Import the library</span></span>
<span id="cb299-2"><a href="working-with-files.html#cb299-2" tabindex="-1"></a><span class="fu">library</span>(readxl)</span></code></pre></div>
<p>The <code>readxl</code> library can read .xlsx, xls files to a data frame. For this case, Human genome excel file is read.</p>
<div class="sourceCode" id="cb300"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb300-1"><a href="working-with-files.html#cb300-1" tabindex="-1"></a><span class="co"># Import the library</span></span>
<span id="cb300-2"><a href="working-with-files.html#cb300-2" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb300-3"><a href="working-with-files.html#cb300-3" tabindex="-1"></a></span>
<span id="cb300-4"><a href="working-with-files.html#cb300-4" tabindex="-1"></a><span class="co"># Read the excel file</span></span>
<span id="cb300-5"><a href="working-with-files.html#cb300-5" tabindex="-1"></a>hg <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="st">&quot;data/Human_genome.xlsx&quot;</span>)</span></code></pre></div>
<p>View the data set</p>
<div class="sourceCode" id="cb301"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb301-1"><a href="working-with-files.html#cb301-1" tabindex="-1"></a><span class="co"># Import the library</span></span>
<span id="cb301-2"><a href="working-with-files.html#cb301-2" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb301-3"><a href="working-with-files.html#cb301-3" tabindex="-1"></a></span>
<span id="cb301-4"><a href="working-with-files.html#cb301-4" tabindex="-1"></a><span class="co"># Read the excel file</span></span>
<span id="cb301-5"><a href="working-with-files.html#cb301-5" tabindex="-1"></a>hg <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="st">&quot;data/Human_genome.xlsx&quot;</span>)</span>
<span id="cb301-6"><a href="working-with-files.html#cb301-6" tabindex="-1"></a></span>
<span id="cb301-7"><a href="working-with-files.html#cb301-7" tabindex="-1"></a><span class="co"># View the dataset</span></span>
<span id="cb301-8"><a href="working-with-files.html#cb301-8" tabindex="-1"></a><span class="fu">str</span>(hg)</span></code></pre></div>
<pre><code>## tibble [25 × 8] (S3: tbl_df/tbl/data.frame)
##  $ Chromosome               : chr [1:25] &quot;1&quot; &quot;2&quot; &quot;3&quot; &quot;4&quot; ...
##  $ Base_Pairs               : num [1:25] 2.49e+08 2.42e+08 1.98e+08 1.90e+08 1.82e+08 ...
##  $ Protein_Coding_genes     : num [1:25] 2058 1309 1078 752 876 ...
##  $ Pseudogenes              : num [1:25] 1220 1023 763 727 721 ...
##  $ Total_long_non_coding_RNA: num [1:25] 1200 1037 711 657 844 ...
##  $ Total_small_nc_RA        : num [1:25] 496 375 298 228 235 234 208 214 190 204 ...
##  $ miRNA                    : num [1:25] 134 115 99 92 83 81 90 80 69 64 ...
##  $ rRNA                     : num [1:25] 66 40 29 24 25 26 24 28 19 32 ...</code></pre>
<p>In R, <code>head()</code> and <code>tail()</code> are functions used to view the first few rows or the last few rows of a data frame respectively. These functions are particularly useful for quickly inspecting the structure and content of a data set.</p>
<p>Here’s how you can use them for this case;</p>
<div class="sourceCode" id="cb303"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb303-1"><a href="working-with-files.html#cb303-1" tabindex="-1"></a><span class="co"># Import the library</span></span>
<span id="cb303-2"><a href="working-with-files.html#cb303-2" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb303-3"><a href="working-with-files.html#cb303-3" tabindex="-1"></a></span>
<span id="cb303-4"><a href="working-with-files.html#cb303-4" tabindex="-1"></a><span class="co"># Read the excel file</span></span>
<span id="cb303-5"><a href="working-with-files.html#cb303-5" tabindex="-1"></a>hg <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="st">&quot;data/Human_genome.xlsx&quot;</span>)</span>
<span id="cb303-6"><a href="working-with-files.html#cb303-6" tabindex="-1"></a></span>
<span id="cb303-7"><a href="working-with-files.html#cb303-7" tabindex="-1"></a><span class="co"># Display the first 6 records of the data set</span></span>
<span id="cb303-8"><a href="working-with-files.html#cb303-8" tabindex="-1"></a><span class="fu">head</span>(hg)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 8
##   Chromosome Base_Pairs Protein_Coding_genes Pseudogenes Total_long_non_coding…¹
##   &lt;chr&gt;           &lt;dbl&gt;                &lt;dbl&gt;       &lt;dbl&gt;                   &lt;dbl&gt;
## 1 1           248956422                 2058        1220                    1200
## 2 2           242193529                 1309        1023                    1037
## 3 3           198295559                 1078         763                     711
## 4 4           190214555                  752         727                     657
## 5 5           181538259                  876         721                     844
## 6 6           170805979                 1048         801                     639
## # ℹ abbreviated name: ¹​Total_long_non_coding_RNA
## # ℹ 3 more variables: Total_small_nc_RA &lt;dbl&gt;, miRNA &lt;dbl&gt;, rRNA &lt;dbl&gt;</code></pre>
<p>The <code>head()</code> function displays the first few rows of a data frame or a vector. By default, it shows the first 6 rows, but you can specify the number of rows to display using the <code>n</code> argument.</p>
<div class="sourceCode" id="cb305"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb305-1"><a href="working-with-files.html#cb305-1" tabindex="-1"></a><span class="co"># Import the library</span></span>
<span id="cb305-2"><a href="working-with-files.html#cb305-2" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb305-3"><a href="working-with-files.html#cb305-3" tabindex="-1"></a></span>
<span id="cb305-4"><a href="working-with-files.html#cb305-4" tabindex="-1"></a><span class="co"># Read the excel file</span></span>
<span id="cb305-5"><a href="working-with-files.html#cb305-5" tabindex="-1"></a>hg <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="st">&quot;data/Human_genome.xlsx&quot;</span>)</span>
<span id="cb305-6"><a href="working-with-files.html#cb305-6" tabindex="-1"></a></span>
<span id="cb305-7"><a href="working-with-files.html#cb305-7" tabindex="-1"></a><span class="co"># Specify the number of the first rows you want to view</span></span>
<span id="cb305-8"><a href="working-with-files.html#cb305-8" tabindex="-1"></a><span class="fu">head</span>(hg, <span class="at">n=</span><span class="dv">10</span>)</span></code></pre></div>
<pre><code>## # A tibble: 10 × 8
##    Chromosome Base_Pairs Protein_Coding_genes Pseudogenes Total_long_non_codin…¹
##    &lt;chr&gt;           &lt;dbl&gt;                &lt;dbl&gt;       &lt;dbl&gt;                  &lt;dbl&gt;
##  1 1           248956422                 2058        1220                   1200
##  2 2           242193529                 1309        1023                   1037
##  3 3           198295559                 1078         763                    711
##  4 4           190214555                  752         727                    657
##  5 5           181538259                  876         721                    844
##  6 6           170805979                 1048         801                    639
##  7 7           159345973                  989         885                    605
##  8 8           145138636                  677         613                    735
##  9 9           138394717                  786         661                    491
## 10 10          133797422                  733         568                    579
## # ℹ abbreviated name: ¹​Total_long_non_coding_RNA
## # ℹ 3 more variables: Total_small_nc_RA &lt;dbl&gt;, miRNA &lt;dbl&gt;, rRNA &lt;dbl&gt;</code></pre>
<p>The tail argument can be used as;</p>
<div class="sourceCode" id="cb307"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb307-1"><a href="working-with-files.html#cb307-1" tabindex="-1"></a><span class="co"># Import the library</span></span>
<span id="cb307-2"><a href="working-with-files.html#cb307-2" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb307-3"><a href="working-with-files.html#cb307-3" tabindex="-1"></a></span>
<span id="cb307-4"><a href="working-with-files.html#cb307-4" tabindex="-1"></a><span class="co"># Read the excel file</span></span>
<span id="cb307-5"><a href="working-with-files.html#cb307-5" tabindex="-1"></a>hg <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="st">&quot;data/Human_genome.xlsx&quot;</span>)</span>
<span id="cb307-6"><a href="working-with-files.html#cb307-6" tabindex="-1"></a></span>
<span id="cb307-7"><a href="working-with-files.html#cb307-7" tabindex="-1"></a><span class="co"># View the last 6 rows of the dataset. </span></span>
<span id="cb307-8"><a href="working-with-files.html#cb307-8" tabindex="-1"></a><span class="fu">tail</span>(hg)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 8
##   Chromosome Base_Pairs Protein_Coding_genes Pseudogenes Total_long_non_coding…¹
##   &lt;chr&gt;           &lt;dbl&gt;                &lt;dbl&gt;       &lt;dbl&gt;                   &lt;dbl&gt;
## 1 20           64444167                  544         249                     384
## 2 21           46709983                  234         185                     305
## 3 22           50818468                  488         324                     357
## 4 X           156040895                  842         874                     271
## 5 Y            57227415                   71         388                      71
## 6 mtDNA           16569                   13           0                       0
## # ℹ abbreviated name: ¹​Total_long_non_coding_RNA
## # ℹ 3 more variables: Total_small_nc_RA &lt;dbl&gt;, miRNA &lt;dbl&gt;, rRNA &lt;dbl&gt;</code></pre>
<p>to show the last 6 records while specifying <code>n</code> value will show the specified number of rows.</p>
<p>R identifies this type of data as table or data frame.</p>
<div class="sourceCode" id="cb309"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb309-1"><a href="working-with-files.html#cb309-1" tabindex="-1"></a><span class="fu">class</span>(hg)</span></code></pre></div>
<pre><code>## [1] &quot;tbl_df&quot;     &quot;tbl&quot;        &quot;data.frame&quot;</code></pre>
<p><code>str()</code> is used to find the nature of data such as column data types, number of rows and columns, column names , and first records in the dataframe</p>
<div class="sourceCode" id="cb311"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb311-1"><a href="working-with-files.html#cb311-1" tabindex="-1"></a><span class="co"># Import the library</span></span>
<span id="cb311-2"><a href="working-with-files.html#cb311-2" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb311-3"><a href="working-with-files.html#cb311-3" tabindex="-1"></a></span>
<span id="cb311-4"><a href="working-with-files.html#cb311-4" tabindex="-1"></a><span class="co"># Read the excel file</span></span>
<span id="cb311-5"><a href="working-with-files.html#cb311-5" tabindex="-1"></a>hg <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="st">&quot;data/Human_genome.xlsx&quot;</span>)</span>
<span id="cb311-6"><a href="working-with-files.html#cb311-6" tabindex="-1"></a></span>
<span id="cb311-7"><a href="working-with-files.html#cb311-7" tabindex="-1"></a><span class="fu">str</span>(hg)</span></code></pre></div>
<pre><code>## tibble [25 × 8] (S3: tbl_df/tbl/data.frame)
##  $ Chromosome               : chr [1:25] &quot;1&quot; &quot;2&quot; &quot;3&quot; &quot;4&quot; ...
##  $ Base_Pairs               : num [1:25] 2.49e+08 2.42e+08 1.98e+08 1.90e+08 1.82e+08 ...
##  $ Protein_Coding_genes     : num [1:25] 2058 1309 1078 752 876 ...
##  $ Pseudogenes              : num [1:25] 1220 1023 763 727 721 ...
##  $ Total_long_non_coding_RNA: num [1:25] 1200 1037 711 657 844 ...
##  $ Total_small_nc_RA        : num [1:25] 496 375 298 228 235 234 208 214 190 204 ...
##  $ miRNA                    : num [1:25] 134 115 99 92 83 81 90 80 69 64 ...
##  $ rRNA                     : num [1:25] 66 40 29 24 25 26 24 28 19 32 ...</code></pre>
<p><code>is.na()</code> is used to check if there is null/blank values in the dataset. If there is it returns TRUE in place of the null point while it return FALSE in place of the value.</p>
<div class="sourceCode" id="cb313"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb313-1"><a href="working-with-files.html#cb313-1" tabindex="-1"></a><span class="co"># Import the library</span></span>
<span id="cb313-2"><a href="working-with-files.html#cb313-2" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb313-3"><a href="working-with-files.html#cb313-3" tabindex="-1"></a></span>
<span id="cb313-4"><a href="working-with-files.html#cb313-4" tabindex="-1"></a><span class="co"># Read the excel file</span></span>
<span id="cb313-5"><a href="working-with-files.html#cb313-5" tabindex="-1"></a>hg <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="st">&quot;data/Human_genome.xlsx&quot;</span>)</span>
<span id="cb313-6"><a href="working-with-files.html#cb313-6" tabindex="-1"></a></span>
<span id="cb313-7"><a href="working-with-files.html#cb313-7" tabindex="-1"></a><span class="co">#Count the null values in the data frame </span></span>
<span id="cb313-8"><a href="working-with-files.html#cb313-8" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">is.na</span>(hg))</span></code></pre></div>
<pre><code>## [1] 0</code></pre>
<p>The “hg” data frame has no null values.</p>
<p>Lets dive into biology to solve some of the questions about genetics</p>
<p><strong>Q1. Calculate the gene density for each chromosome</strong></p>
<p>The gene density is calculated in terms of genes per megabase</p>
<div class="sourceCode" id="cb315"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb315-1"><a href="working-with-files.html#cb315-1" tabindex="-1"></a><span class="co"># Import the library</span></span>
<span id="cb315-2"><a href="working-with-files.html#cb315-2" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb315-3"><a href="working-with-files.html#cb315-3" tabindex="-1"></a></span>
<span id="cb315-4"><a href="working-with-files.html#cb315-4" tabindex="-1"></a><span class="co"># Read the excel file</span></span>
<span id="cb315-5"><a href="working-with-files.html#cb315-5" tabindex="-1"></a>hg <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="st">&quot;data/Human_genome.xlsx&quot;</span>)</span>
<span id="cb315-6"><a href="working-with-files.html#cb315-6" tabindex="-1"></a></span>
<span id="cb315-7"><a href="working-with-files.html#cb315-7" tabindex="-1"></a><span class="co"># Calculating the gene density for each chromosomes </span></span>
<span id="cb315-8"><a href="working-with-files.html#cb315-8" tabindex="-1"></a>base_Mb <span class="ot">=</span> hg<span class="sc">$</span>Base_Pairs<span class="sc">/</span><span class="dv">1000000</span></span>
<span id="cb315-9"><a href="working-with-files.html#cb315-9" tabindex="-1"></a>base_Mb</span></code></pre></div>
<pre><code>##  [1] 248.956422 242.193529 198.295559 190.214555 181.538259 170.805979
##  [7] 159.345973 145.138636 138.394717 133.797422 135.086622 133.275309
## [13] 114.364328 107.043718 101.991189  90.338345  83.257441  80.373285
## [19]  58.617616  64.444167  46.709983  50.818468 156.040895  57.227415
## [25]   0.016569</code></pre>
<p>To make the values more clear, the results can be rounded off to two decimal place.</p>
<div class="sourceCode" id="cb317"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb317-1"><a href="working-with-files.html#cb317-1" tabindex="-1"></a><span class="co"># Import the library</span></span>
<span id="cb317-2"><a href="working-with-files.html#cb317-2" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb317-3"><a href="working-with-files.html#cb317-3" tabindex="-1"></a></span>
<span id="cb317-4"><a href="working-with-files.html#cb317-4" tabindex="-1"></a><span class="co"># Read the excel file</span></span>
<span id="cb317-5"><a href="working-with-files.html#cb317-5" tabindex="-1"></a>hg <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="st">&quot;data/Human_genome.xlsx&quot;</span>)</span>
<span id="cb317-6"><a href="working-with-files.html#cb317-6" tabindex="-1"></a></span>
<span id="cb317-7"><a href="working-with-files.html#cb317-7" tabindex="-1"></a><span class="co"># Calculating the gene density for each chromosomes </span></span>
<span id="cb317-8"><a href="working-with-files.html#cb317-8" tabindex="-1"></a>base_Mb <span class="ot">=</span> hg<span class="sc">$</span>Base_Pairs<span class="sc">/</span><span class="dv">1000000</span></span>
<span id="cb317-9"><a href="working-with-files.html#cb317-9" tabindex="-1"></a>base_Mb</span></code></pre></div>
<pre><code>##  [1] 248.956422 242.193529 198.295559 190.214555 181.538259 170.805979
##  [7] 159.345973 145.138636 138.394717 133.797422 135.086622 133.275309
## [13] 114.364328 107.043718 101.991189  90.338345  83.257441  80.373285
## [19]  58.617616  64.444167  46.709983  50.818468 156.040895  57.227415
## [25]   0.016569</code></pre>
<div class="sourceCode" id="cb319"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb319-1"><a href="working-with-files.html#cb319-1" tabindex="-1"></a>rbase_Mb<span class="ot">=</span><span class="fu">round</span>(base_Mb, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb319-2"><a href="working-with-files.html#cb319-2" tabindex="-1"></a>rbase_Mb</span></code></pre></div>
<pre><code>##  [1] 248.96 242.19 198.30 190.21 181.54 170.81 159.35 145.14 138.39 133.80
## [11] 135.09 133.28 114.36 107.04 101.99  90.34  83.26  80.37  58.62  64.44
## [21]  46.71  50.82 156.04  57.23   0.02</code></pre>
<p><strong>Q2. Calculate number of genes per megabase per chromosomes</strong></p>
<p>The protein coding genes are divided to the gene density.</p>
<div class="sourceCode" id="cb321"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb321-1"><a href="working-with-files.html#cb321-1" tabindex="-1"></a><span class="co"># Import the library</span></span>
<span id="cb321-2"><a href="working-with-files.html#cb321-2" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb321-3"><a href="working-with-files.html#cb321-3" tabindex="-1"></a></span>
<span id="cb321-4"><a href="working-with-files.html#cb321-4" tabindex="-1"></a><span class="co"># Read the excel file</span></span>
<span id="cb321-5"><a href="working-with-files.html#cb321-5" tabindex="-1"></a>hg <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="st">&quot;data/Human_genome.xlsx&quot;</span>)</span>
<span id="cb321-6"><a href="working-with-files.html#cb321-6" tabindex="-1"></a></span>
<span id="cb321-7"><a href="working-with-files.html#cb321-7" tabindex="-1"></a><span class="co"># Calculating the number of genes per megabase per chromosomes </span></span>
<span id="cb321-8"><a href="working-with-files.html#cb321-8" tabindex="-1"></a>genes_per_megabase <span class="ot">=</span> <span class="fu">round</span>(hg<span class="sc">$</span>Protein_Coding_genes<span class="sc">/</span>base_Mb, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb321-9"><a href="working-with-files.html#cb321-9" tabindex="-1"></a>genes_per_megabase</span></code></pre></div>
<pre><code>##  [1]   8.27   5.40   5.44   3.95   4.83   6.14   6.21   4.66   5.68   5.48
## [11]   9.61   7.76   2.86   7.75   6.01   9.66  14.38   3.36  25.11   8.44
## [21]   5.01   9.60   5.40   1.24 784.60</code></pre>
<p>Lets create a new column to the dataframe to the dataset</p>
<div class="sourceCode" id="cb323"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb323-1"><a href="working-with-files.html#cb323-1" tabindex="-1"></a><span class="co"># Import the library</span></span>
<span id="cb323-2"><a href="working-with-files.html#cb323-2" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb323-3"><a href="working-with-files.html#cb323-3" tabindex="-1"></a></span>
<span id="cb323-4"><a href="working-with-files.html#cb323-4" tabindex="-1"></a><span class="co"># Read the excel file</span></span>
<span id="cb323-5"><a href="working-with-files.html#cb323-5" tabindex="-1"></a>hg <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="st">&quot;data/Human_genome.xlsx&quot;</span>)</span>
<span id="cb323-6"><a href="working-with-files.html#cb323-6" tabindex="-1"></a></span>
<span id="cb323-7"><a href="working-with-files.html#cb323-7" tabindex="-1"></a><span class="co"># Calculating the number of genes per megabase per chromosomes </span></span>
<span id="cb323-8"><a href="working-with-files.html#cb323-8" tabindex="-1"></a>genes_per_megabase <span class="ot">=</span> <span class="fu">round</span>(hg<span class="sc">$</span>Protein_Coding_genes<span class="sc">/</span>base_Mb, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb323-9"><a href="working-with-files.html#cb323-9" tabindex="-1"></a>genes_per_megabase</span></code></pre></div>
<pre><code>##  [1]   8.27   5.40   5.44   3.95   4.83   6.14   6.21   4.66   5.68   5.48
## [11]   9.61   7.76   2.86   7.75   6.01   9.66  14.38   3.36  25.11   8.44
## [21]   5.01   9.60   5.40   1.24 784.60</code></pre>
<div class="sourceCode" id="cb325"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb325-1"><a href="working-with-files.html#cb325-1" tabindex="-1"></a><span class="co"># Store the result as a column to the data frame</span></span>
<span id="cb325-2"><a href="working-with-files.html#cb325-2" tabindex="-1"></a>hg[<span class="st">&quot;genes_per_Mb&quot;</span>]<span class="ot">=</span>genes_per_megabase</span>
<span id="cb325-3"><a href="working-with-files.html#cb325-3" tabindex="-1"></a><span class="fu">str</span>(hg)</span></code></pre></div>
<pre><code>## tibble [25 × 9] (S3: tbl_df/tbl/data.frame)
##  $ Chromosome               : chr [1:25] &quot;1&quot; &quot;2&quot; &quot;3&quot; &quot;4&quot; ...
##  $ Base_Pairs               : num [1:25] 2.49e+08 2.42e+08 1.98e+08 1.90e+08 1.82e+08 ...
##  $ Protein_Coding_genes     : num [1:25] 2058 1309 1078 752 876 ...
##  $ Pseudogenes              : num [1:25] 1220 1023 763 727 721 ...
##  $ Total_long_non_coding_RNA: num [1:25] 1200 1037 711 657 844 ...
##  $ Total_small_nc_RA        : num [1:25] 496 375 298 228 235 234 208 214 190 204 ...
##  $ miRNA                    : num [1:25] 134 115 99 92 83 81 90 80 69 64 ...
##  $ rRNA                     : num [1:25] 66 40 29 24 25 26 24 28 19 32 ...
##  $ genes_per_Mb             : num [1:25] 8.27 5.4 5.44 3.95 4.83 6.14 6.21 4.66 5.68 5.48 ...</code></pre>
</div>
<div id="write-a-new-table-to-a-file-locally" class="section level3 hasAnchor" number="4.1.2">
<h3><span class="header-section-number">4.1.2</span> Write a new table to a file locally<a href="working-with-files.html#write-a-new-table-to-a-file-locally" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <code>write.table()</code> function is used to write table to file on a local machine. The file is saved by deault to the working directory unless the path is specified.</p>
<div class="sourceCode" id="cb327"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb327-1"><a href="working-with-files.html#cb327-1" tabindex="-1"></a><span class="co"># Import the library</span></span>
<span id="cb327-2"><a href="working-with-files.html#cb327-2" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb327-3"><a href="working-with-files.html#cb327-3" tabindex="-1"></a></span>
<span id="cb327-4"><a href="working-with-files.html#cb327-4" tabindex="-1"></a><span class="co"># Read the excel file</span></span>
<span id="cb327-5"><a href="working-with-files.html#cb327-5" tabindex="-1"></a>hg <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="st">&quot;data/Human_genome.xlsx&quot;</span>)</span>
<span id="cb327-6"><a href="working-with-files.html#cb327-6" tabindex="-1"></a></span>
<span id="cb327-7"><a href="working-with-files.html#cb327-7" tabindex="-1"></a><span class="co"># Calculating the number of genes per megabase per chromosomes </span></span>
<span id="cb327-8"><a href="working-with-files.html#cb327-8" tabindex="-1"></a>genes_per_megabase <span class="ot">=</span> <span class="fu">round</span>(hg<span class="sc">$</span>Protein_Coding_genes<span class="sc">/</span>base_Mb, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb327-9"><a href="working-with-files.html#cb327-9" tabindex="-1"></a></span>
<span id="cb327-10"><a href="working-with-files.html#cb327-10" tabindex="-1"></a><span class="co"># Store the result as a column to the data frame</span></span>
<span id="cb327-11"><a href="working-with-files.html#cb327-11" tabindex="-1"></a>hg[<span class="st">&quot;genes_per_Mb&quot;</span>]<span class="ot">=</span>genes_per_megabase</span>
<span id="cb327-12"><a href="working-with-files.html#cb327-12" tabindex="-1"></a></span>
<span id="cb327-13"><a href="working-with-files.html#cb327-13" tabindex="-1"></a><span class="co"># Write thge files locally</span></span>
<span id="cb327-14"><a href="working-with-files.html#cb327-14" tabindex="-1"></a><span class="fu">write.table</span>(hg, </span>
<span id="cb327-15"><a href="working-with-files.html#cb327-15" tabindex="-1"></a>            <span class="at">file=</span><span class="st">&quot;data/hg_modified&quot;</span>, </span>
<span id="cb327-16"><a href="working-with-files.html#cb327-16" tabindex="-1"></a>            <span class="at">quote=</span><span class="cn">FALSE</span>, </span>
<span id="cb327-17"><a href="working-with-files.html#cb327-17" tabindex="-1"></a>            <span class="at">sep=</span><span class="st">&#39;</span><span class="sc">\t</span><span class="st">&#39;</span>, </span>
<span id="cb327-18"><a href="working-with-files.html#cb327-18" tabindex="-1"></a>            <span class="at">row.names=</span><span class="cn">FALSE</span>, </span>
<span id="cb327-19"><a href="working-with-files.html#cb327-19" tabindex="-1"></a>            <span class="at">col.names=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<p>This file will be saved by default on the working directory unless the file path is changed.</p>
<p>Checking if the new file is saved.</p>
<pre><code>file.show(&quot;data/hg_modified&quot;)</code></pre>
</div>
</div>
<div id="csv-files" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> .CSV files<a href="working-with-files.html#csv-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>CSV stands for “Comma Separated Values”. This is a plain text file that contains data in a tabular format for example the data below that represents athletes’ statistics.</p>
<pre><code>Name, Age, Weight
Salman, 30, 75
Pragya, 35, 72
Vidyat, 25, 68</code></pre>
<p>Each line in a CSV file represents a row while each field in a row is separated by a comma “,”. These files are often used in data manipulations since they are lightweight, simple and widely supported. In this course we shall briefly go through two csv files and do brief analyses.</p>
<div id="titanic-dataset" class="section level3 hasAnchor" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> Titanic dataset<a href="working-with-files.html#titanic-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The titanic data is a famous data set in data analysis. It contains information about passengers that boarded the ship Titan(not actual true) including their age, ticket class, fare and others.</p>
<div id="reading-the-csv-file" class="section level4 hasAnchor" number="4.2.1.1">
<h4><span class="header-section-number">4.2.1.1</span> Reading the csv file<a href="working-with-files.html#reading-the-csv-file" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The <code>read.csv()</code> function is used to read a csv file to a data frame.</p>
<div class="sourceCode" id="cb330"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb330-1"><a href="working-with-files.html#cb330-1" tabindex="-1"></a><span class="co"># Read data from a CSV file into a data frame</span></span>
<span id="cb330-2"><a href="working-with-files.html#cb330-2" tabindex="-1"></a>titanic_df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/titanic_dataset.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="data-assessment" class="section level4 hasAnchor" number="4.2.1.2">
<h4><span class="header-section-number">4.2.1.2</span> Data Assessment<a href="working-with-files.html#data-assessment" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Structure of the data set</li>
</ul>
<div class="sourceCode" id="cb331"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb331-1"><a href="working-with-files.html#cb331-1" tabindex="-1"></a><span class="co"># Read data from a CSV file into a data frame</span></span>
<span id="cb331-2"><a href="working-with-files.html#cb331-2" tabindex="-1"></a>titanic_df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/titanic_dataset.csv&quot;</span>)</span>
<span id="cb331-3"><a href="working-with-files.html#cb331-3" tabindex="-1"></a></span>
<span id="cb331-4"><a href="working-with-files.html#cb331-4" tabindex="-1"></a><span class="co"># Checkout the nature of the data set</span></span>
<span id="cb331-5"><a href="working-with-files.html#cb331-5" tabindex="-1"></a><span class="fu">str</span>(titanic_df)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    418 obs. of  12 variables:
##  $ PassengerId: int  892 893 894 895 896 897 898 899 900 901 ...
##  $ Survived   : int  0 1 0 0 1 0 1 0 1 0 ...
##  $ Pclass     : int  3 3 2 3 3 3 3 2 3 3 ...
##  $ Name       : chr  &quot;Kelly, Mr. James&quot; &quot;Wilkes, Mrs. James (Ellen Needs)&quot; &quot;Myles, Mr. Thomas Francis&quot; &quot;Wirz, Mr. Albert&quot; ...
##  $ Sex        : chr  &quot;male&quot; &quot;female&quot; &quot;male&quot; &quot;male&quot; ...
##  $ Age        : num  34.5 47 62 27 22 14 30 26 18 21 ...
##  $ SibSp      : int  0 1 0 0 1 0 0 1 0 2 ...
##  $ Parch      : int  0 0 0 0 1 0 0 1 0 0 ...
##  $ Ticket     : chr  &quot;330911&quot; &quot;363272&quot; &quot;240276&quot; &quot;315154&quot; ...
##  $ Fare       : num  7.83 7 9.69 8.66 12.29 ...
##  $ Cabin      : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ Embarked   : chr  &quot;Q&quot; &quot;S&quot; &quot;Q&quot; &quot;S&quot; ...</code></pre>
<ul>
<li>Show the first 6 records of the data set.</li>
</ul>
<div class="sourceCode" id="cb333"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb333-1"><a href="working-with-files.html#cb333-1" tabindex="-1"></a><span class="co"># Read data from a CSV file into a data frame</span></span>
<span id="cb333-2"><a href="working-with-files.html#cb333-2" tabindex="-1"></a>titanic_df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/titanic_dataset.csv&quot;</span>)</span>
<span id="cb333-3"><a href="working-with-files.html#cb333-3" tabindex="-1"></a></span>
<span id="cb333-4"><a href="working-with-files.html#cb333-4" tabindex="-1"></a><span class="co"># Show the first five rows of the data set</span></span>
<span id="cb333-5"><a href="working-with-files.html#cb333-5" tabindex="-1"></a><span class="fu">head</span>(titanic_df)</span></code></pre></div>
<pre><code>##   PassengerId Survived Pclass                                         Name
## 1         892        0      3                             Kelly, Mr. James
## 2         893        1      3             Wilkes, Mrs. James (Ellen Needs)
## 3         894        0      2                    Myles, Mr. Thomas Francis
## 4         895        0      3                             Wirz, Mr. Albert
## 5         896        1      3 Hirvonen, Mrs. Alexander (Helga E Lindqvist)
## 6         897        0      3                   Svensson, Mr. Johan Cervin
##      Sex  Age SibSp Parch  Ticket    Fare Cabin Embarked
## 1   male 34.5     0     0  330911  7.8292              Q
## 2 female 47.0     1     0  363272  7.0000              S
## 3   male 62.0     0     0  240276  9.6875              Q
## 4   male 27.0     0     0  315154  8.6625              S
## 5 female 22.0     1     1 3101298 12.2875              S
## 6   male 14.0     0     0    7538  9.2250              S</code></pre>
<ul>
<li>Count the null values in the data</li>
</ul>
<div class="sourceCode" id="cb335"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb335-1"><a href="working-with-files.html#cb335-1" tabindex="-1"></a><span class="co"># Read data from a CSV file into a data frame</span></span>
<span id="cb335-2"><a href="working-with-files.html#cb335-2" tabindex="-1"></a>titanic_df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/titanic_dataset.csv&quot;</span>)</span>
<span id="cb335-3"><a href="working-with-files.html#cb335-3" tabindex="-1"></a></span>
<span id="cb335-4"><a href="working-with-files.html#cb335-4" tabindex="-1"></a><span class="co">#Find out the null values in the data set</span></span>
<span id="cb335-5"><a href="working-with-files.html#cb335-5" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">is.na</span>(titanic_df))</span></code></pre></div>
<pre><code>## [1] 87</code></pre>
<p>The are 87 fields in the data that are null(empty)</p>
<ul>
<li>Calculate the descriptive statistics</li>
</ul>
<div class="sourceCode" id="cb337"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb337-1"><a href="working-with-files.html#cb337-1" tabindex="-1"></a><span class="co"># Read data from a CSV file into a data frame</span></span>
<span id="cb337-2"><a href="working-with-files.html#cb337-2" tabindex="-1"></a>titanic_df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/titanic_dataset.csv&quot;</span>)</span>
<span id="cb337-3"><a href="working-with-files.html#cb337-3" tabindex="-1"></a></span>
<span id="cb337-4"><a href="working-with-files.html#cb337-4" tabindex="-1"></a><span class="co"># Calculate the descriptive statistics in the data set </span></span>
<span id="cb337-5"><a href="working-with-files.html#cb337-5" tabindex="-1"></a><span class="fu">summary</span>(titanic_df)</span></code></pre></div>
<pre><code>##   PassengerId        Survived          Pclass          Name          
##  Min.   : 892.0   Min.   :0.0000   Min.   :1.000   Length:418        
##  1st Qu.: 996.2   1st Qu.:0.0000   1st Qu.:1.000   Class :character  
##  Median :1100.5   Median :0.0000   Median :3.000   Mode  :character  
##  Mean   :1100.5   Mean   :0.3636   Mean   :2.266                     
##  3rd Qu.:1204.8   3rd Qu.:1.0000   3rd Qu.:3.000                     
##  Max.   :1309.0   Max.   :1.0000   Max.   :3.000                     
##                                                                      
##      Sex                 Age            SibSp            Parch       
##  Length:418         Min.   : 0.17   Min.   :0.0000   Min.   :0.0000  
##  Class :character   1st Qu.:21.00   1st Qu.:0.0000   1st Qu.:0.0000  
##  Mode  :character   Median :27.00   Median :0.0000   Median :0.0000  
##                     Mean   :30.27   Mean   :0.4474   Mean   :0.3923  
##                     3rd Qu.:39.00   3rd Qu.:1.0000   3rd Qu.:0.0000  
##                     Max.   :76.00   Max.   :8.0000   Max.   :9.0000  
##                     NA&#39;s   :86                                       
##     Ticket               Fare            Cabin             Embarked        
##  Length:418         Min.   :  0.000   Length:418         Length:418        
##  Class :character   1st Qu.:  7.896   Class :character   Class :character  
##  Mode  :character   Median : 14.454   Mode  :character   Mode  :character  
##                     Mean   : 35.627                                        
##                     3rd Qu.: 31.500                                        
##                     Max.   :512.329                                        
##                     NA&#39;s   :1</code></pre>
</div>
<div id="data-analysis" class="section level4 hasAnchor" number="4.2.1.3">
<h4><span class="header-section-number">4.2.1.3</span> Data Analysis<a href="working-with-files.html#data-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The answers to the two questions will give a brief understanding of titanic passengers.</p>
<ul>
<li><strong>What is the survival rate by gender?</strong></li>
</ul>
<div class="sourceCode" id="cb339"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb339-1"><a href="working-with-files.html#cb339-1" tabindex="-1"></a><span class="co"># Read data from a CSV file into a data frame</span></span>
<span id="cb339-2"><a href="working-with-files.html#cb339-2" tabindex="-1"></a>titanic_df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/titanic_dataset.csv&quot;</span>)</span>
<span id="cb339-3"><a href="working-with-files.html#cb339-3" tabindex="-1"></a></span>
<span id="cb339-4"><a href="working-with-files.html#cb339-4" tabindex="-1"></a><span class="co"># Survival rate by gender</span></span>
<span id="cb339-5"><a href="working-with-files.html#cb339-5" tabindex="-1"></a><span class="fu">table</span>(titanic_df<span class="sc">$</span>Sex, titanic_df<span class="sc">$</span>Survived)</span></code></pre></div>
<pre><code>##         
##            0   1
##   female   0 152
##   male   266   0</code></pre>
<p>From the table above, all males drowned but all females survived.</p>
<ul>
<li><strong>What is the survival rate by ticket class?</strong></li>
</ul>
<div class="sourceCode" id="cb341"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb341-1"><a href="working-with-files.html#cb341-1" tabindex="-1"></a><span class="co"># Read data from a CSV file into a data frame</span></span>
<span id="cb341-2"><a href="working-with-files.html#cb341-2" tabindex="-1"></a>titanic_df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/titanic_dataset.csv&quot;</span>)</span>
<span id="cb341-3"><a href="working-with-files.html#cb341-3" tabindex="-1"></a></span>
<span id="cb341-4"><a href="working-with-files.html#cb341-4" tabindex="-1"></a><span class="co"># Survival rate by ticket class</span></span>
<span id="cb341-5"><a href="working-with-files.html#cb341-5" tabindex="-1"></a><span class="fu">table</span>(titanic_df<span class="sc">$</span>Pclass, titanic_df<span class="sc">$</span>Survived)</span></code></pre></div>
<pre><code>##    
##       0   1
##   1  57  50
##   2  63  30
##   3 146  72</code></pre>
<p>Most passengers from ticket class 0 survived.</p>
</div>
</div>
<div id="ibm-employee-attrition-data" class="section level3 hasAnchor" number="4.2.2">
<h3><span class="header-section-number">4.2.2</span> IBM Employee Attrition data<a href="working-with-files.html#ibm-employee-attrition-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This is a common data set used for HR analytics. It contains information about all IBM employees, who are still in the company and the ones who left. Analyzing this attrition data set will provide meaningful insights on employees’ behavior and their engagement. HR can also find employee retention techniques with organizations. In this course section, we will briefly go through basic analysis after assessing the data.</p>
<div id="reading-the-csv-file-1" class="section level4 hasAnchor" number="4.2.2.1">
<h4><span class="header-section-number">4.2.2.1</span> Reading the csv file<a href="working-with-files.html#reading-the-csv-file-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb343"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb343-1"><a href="working-with-files.html#cb343-1" tabindex="-1"></a>attrition_df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/HR-Employee-Attrition.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="data-assessment-1" class="section level4 hasAnchor" number="4.2.2.2">
<h4><span class="header-section-number">4.2.2.2</span> Data Assessment<a href="working-with-files.html#data-assessment-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Structure of the data</li>
</ul>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb344-1"><a href="working-with-files.html#cb344-1" tabindex="-1"></a><span class="co"># Read the csv file </span></span>
<span id="cb344-2"><a href="working-with-files.html#cb344-2" tabindex="-1"></a>attrition_df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/HR-Employee-Attrition.csv&quot;</span>)</span>
<span id="cb344-3"><a href="working-with-files.html#cb344-3" tabindex="-1"></a></span>
<span id="cb344-4"><a href="working-with-files.html#cb344-4" tabindex="-1"></a><span class="co">#Nature of the data set </span></span>
<span id="cb344-5"><a href="working-with-files.html#cb344-5" tabindex="-1"></a><span class="fu">str</span>(attrition_df)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    1470 obs. of  35 variables:
##  $ Age                     : int  41 49 37 33 27 32 59 30 38 36 ...
##  $ Attrition               : chr  &quot;Yes&quot; &quot;No&quot; &quot;Yes&quot; &quot;No&quot; ...
##  $ BusinessTravel          : chr  &quot;Travel_Rarely&quot; &quot;Travel_Frequently&quot; &quot;Travel_Rarely&quot; &quot;Travel_Frequently&quot; ...
##  $ DailyRate               : int  1102 279 1373 1392 591 1005 1324 1358 216 1299 ...
##  $ Department              : chr  &quot;Sales&quot; &quot;Research &amp; Development&quot; &quot;Research &amp; Development&quot; &quot;Research &amp; Development&quot; ...
##  $ DistanceFromHome        : int  1 8 2 3 2 2 3 24 23 27 ...
##  $ Education               : int  2 1 2 4 1 2 3 1 3 3 ...
##  $ EducationField          : chr  &quot;Life Sciences&quot; &quot;Life Sciences&quot; &quot;Other&quot; &quot;Life Sciences&quot; ...
##  $ EmployeeCount           : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ EmployeeNumber          : int  1 2 4 5 7 8 10 11 12 13 ...
##  $ EnvironmentSatisfaction : int  2 3 4 4 1 4 3 4 4 3 ...
##  $ Gender                  : chr  &quot;Female&quot; &quot;Male&quot; &quot;Male&quot; &quot;Female&quot; ...
##  $ HourlyRate              : int  94 61 92 56 40 79 81 67 44 94 ...
##  $ JobInvolvement          : int  3 2 2 3 3 3 4 3 2 3 ...
##  $ JobLevel                : int  2 2 1 1 1 1 1 1 3 2 ...
##  $ JobRole                 : chr  &quot;Sales Executive&quot; &quot;Research Scientist&quot; &quot;Laboratory Technician&quot; &quot;Research Scientist&quot; ...
##  $ JobSatisfaction         : int  4 2 3 3 2 4 1 3 3 3 ...
##  $ MaritalStatus           : chr  &quot;Single&quot; &quot;Married&quot; &quot;Single&quot; &quot;Married&quot; ...
##  $ MonthlyIncome           : int  5993 5130 2090 2909 3468 3068 2670 2693 9526 5237 ...
##  $ MonthlyRate             : int  19479 24907 2396 23159 16632 11864 9964 13335 8787 16577 ...
##  $ NumCompaniesWorked      : int  8 1 6 1 9 0 4 1 0 6 ...
##  $ Over18                  : chr  &quot;Y&quot; &quot;Y&quot; &quot;Y&quot; &quot;Y&quot; ...
##  $ OverTime                : chr  &quot;Yes&quot; &quot;No&quot; &quot;Yes&quot; &quot;Yes&quot; ...
##  $ PercentSalaryHike       : int  11 23 15 11 12 13 20 22 21 13 ...
##  $ PerformanceRating       : int  3 4 3 3 3 3 4 4 4 3 ...
##  $ RelationshipSatisfaction: int  1 4 2 3 4 3 1 2 2 2 ...
##  $ StandardHours           : int  80 80 80 80 80 80 80 80 80 80 ...
##  $ StockOptionLevel        : int  0 1 0 0 1 0 3 1 0 2 ...
##  $ TotalWorkingYears       : int  8 10 7 8 6 8 12 1 10 17 ...
##  $ TrainingTimesLastYear   : int  0 3 3 3 3 2 3 2 2 3 ...
##  $ WorkLifeBalance         : int  1 3 3 3 3 2 2 3 3 2 ...
##  $ YearsAtCompany          : int  6 10 0 8 2 7 1 1 9 7 ...
##  $ YearsInCurrentRole      : int  4 7 0 7 2 7 0 0 7 7 ...
##  $ YearsSinceLastPromotion : int  0 1 0 3 2 3 0 0 1 7 ...
##  $ YearsWithCurrManager    : int  5 7 0 0 2 6 0 0 8 7 ...</code></pre>
<p>Most columns have character and integer values.</p>
<ul>
<li>Shape of the data set</li>
</ul>
<div class="sourceCode" id="cb346"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb346-1"><a href="working-with-files.html#cb346-1" tabindex="-1"></a>attrition_df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/HR-Employee-Attrition.csv&quot;</span>)</span>
<span id="cb346-2"><a href="working-with-files.html#cb346-2" tabindex="-1"></a></span>
<span id="cb346-3"><a href="working-with-files.html#cb346-3" tabindex="-1"></a><span class="co"># Dimension of the data set </span></span>
<span id="cb346-4"><a href="working-with-files.html#cb346-4" tabindex="-1"></a><span class="fu">dim</span>(attrition_df)</span></code></pre></div>
<pre><code>## [1] 1470   35</code></pre>
<p>There are 1470 rows and 35 variables in the data set. Lets now show the first few rows of the data to get the idea of how the data looks like.</p>
<div class="sourceCode" id="cb348"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb348-1"><a href="working-with-files.html#cb348-1" tabindex="-1"></a>attrition_df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/HR-Employee-Attrition.csv&quot;</span>)</span>
<span id="cb348-2"><a href="working-with-files.html#cb348-2" tabindex="-1"></a></span>
<span id="cb348-3"><a href="working-with-files.html#cb348-3" tabindex="-1"></a><span class="co">#Show the first few rows</span></span>
<span id="cb348-4"><a href="working-with-files.html#cb348-4" tabindex="-1"></a><span class="fu">head</span>(attrition_df)</span></code></pre></div>
<pre><code>##   Age Attrition    BusinessTravel DailyRate             Department
## 1  41       Yes     Travel_Rarely      1102                  Sales
## 2  49        No Travel_Frequently       279 Research &amp; Development
## 3  37       Yes     Travel_Rarely      1373 Research &amp; Development
## 4  33        No Travel_Frequently      1392 Research &amp; Development
## 5  27        No     Travel_Rarely       591 Research &amp; Development
## 6  32        No Travel_Frequently      1005 Research &amp; Development
##   DistanceFromHome Education EducationField EmployeeCount EmployeeNumber
## 1                1         2  Life Sciences             1              1
## 2                8         1  Life Sciences             1              2
## 3                2         2          Other             1              4
## 4                3         4  Life Sciences             1              5
## 5                2         1        Medical             1              7
## 6                2         2  Life Sciences             1              8
##   EnvironmentSatisfaction Gender HourlyRate JobInvolvement JobLevel
## 1                       2 Female         94              3        2
## 2                       3   Male         61              2        2
## 3                       4   Male         92              2        1
## 4                       4 Female         56              3        1
## 5                       1   Male         40              3        1
## 6                       4   Male         79              3        1
##                 JobRole JobSatisfaction MaritalStatus MonthlyIncome MonthlyRate
## 1       Sales Executive               4        Single          5993       19479
## 2    Research Scientist               2       Married          5130       24907
## 3 Laboratory Technician               3        Single          2090        2396
## 4    Research Scientist               3       Married          2909       23159
## 5 Laboratory Technician               2       Married          3468       16632
## 6 Laboratory Technician               4        Single          3068       11864
##   NumCompaniesWorked Over18 OverTime PercentSalaryHike PerformanceRating
## 1                  8      Y      Yes                11                 3
## 2                  1      Y       No                23                 4
## 3                  6      Y      Yes                15                 3
## 4                  1      Y      Yes                11                 3
## 5                  9      Y       No                12                 3
## 6                  0      Y       No                13                 3
##   RelationshipSatisfaction StandardHours StockOptionLevel TotalWorkingYears
## 1                        1            80                0                 8
## 2                        4            80                1                10
## 3                        2            80                0                 7
## 4                        3            80                0                 8
## 5                        4            80                1                 6
## 6                        3            80                0                 8
##   TrainingTimesLastYear WorkLifeBalance YearsAtCompany YearsInCurrentRole
## 1                     0               1              6                  4
## 2                     3               3             10                  7
## 3                     3               3              0                  0
## 4                     3               3              8                  7
## 5                     3               3              2                  2
## 6                     2               2              7                  7
##   YearsSinceLastPromotion YearsWithCurrManager
## 1                       0                    5
## 2                       1                    7
## 3                       0                    0
## 4                       3                    0
## 5                       2                    2
## 6                       3                    6</code></pre>
<p>It is also important when analyzing data to find the existence of null fields and duplicated records.</p>
<ul>
<li>Are there any null values?</li>
</ul>
<div class="sourceCode" id="cb350"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb350-1"><a href="working-with-files.html#cb350-1" tabindex="-1"></a><span class="co"># Read the csv file </span></span>
<span id="cb350-2"><a href="working-with-files.html#cb350-2" tabindex="-1"></a>attrition_df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/HR-Employee-Attrition.csv&quot;</span>)</span>
<span id="cb350-3"><a href="working-with-files.html#cb350-3" tabindex="-1"></a></span>
<span id="cb350-4"><a href="working-with-files.html#cb350-4" tabindex="-1"></a><span class="co"># Count the null values in the data</span></span>
<span id="cb350-5"><a href="working-with-files.html#cb350-5" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">is.na</span>(attrition_df))</span></code></pre></div>
<pre><code>## [1] 0</code></pre>
<p>The data is complete with no null values.</p>
<ul>
<li>Are there any duplicated rows?</li>
</ul>
<div class="sourceCode" id="cb352"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb352-1"><a href="working-with-files.html#cb352-1" tabindex="-1"></a><span class="co"># Read the csv file </span></span>
<span id="cb352-2"><a href="working-with-files.html#cb352-2" tabindex="-1"></a>attrition_df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/HR-Employee-Attrition.csv&quot;</span>)</span>
<span id="cb352-3"><a href="working-with-files.html#cb352-3" tabindex="-1"></a></span>
<span id="cb352-4"><a href="working-with-files.html#cb352-4" tabindex="-1"></a><span class="co">#Count the duplicated observations in the data </span></span>
<span id="cb352-5"><a href="working-with-files.html#cb352-5" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">duplicated</span>(attrition_df))</span></code></pre></div>
<pre><code>## [1] 0</code></pre>
<p>Also, all the rows are unique and there is no duplicated row in the data.</p>
</div>
<div id="data-analysis-1" class="section level4 hasAnchor" number="4.2.2.3">
<h4><span class="header-section-number">4.2.2.3</span> Data Analysis<a href="working-with-files.html#data-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The following questions will enable us to get a brief overview of the company;</p>
<ul>
<li><strong>What are the unique departments in the company?</strong></li>
</ul>
<div class="sourceCode" id="cb354"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb354-1"><a href="working-with-files.html#cb354-1" tabindex="-1"></a><span class="co"># Read the csv file </span></span>
<span id="cb354-2"><a href="working-with-files.html#cb354-2" tabindex="-1"></a>attrition_df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/HR-Employee-Attrition.csv&quot;</span>)</span>
<span id="cb354-3"><a href="working-with-files.html#cb354-3" tabindex="-1"></a></span>
<span id="cb354-4"><a href="working-with-files.html#cb354-4" tabindex="-1"></a><span class="co"># Unique departments in the company</span></span>
<span id="cb354-5"><a href="working-with-files.html#cb354-5" tabindex="-1"></a><span class="fu">unique</span>(attrition_df<span class="sc">$</span>Department)</span></code></pre></div>
<pre><code>## [1] &quot;Sales&quot;                  &quot;Research &amp; Development&quot; &quot;Human Resources&quot;</code></pre>
<p>There are three distinct departments in the company data; Human Resources, Sales, and Research &amp; Development.</p>
<ul>
<li><strong>Is there a difference in attrition rates between genders?</strong></li>
</ul>
<div class="sourceCode" id="cb356"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb356-1"><a href="working-with-files.html#cb356-1" tabindex="-1"></a><span class="co"># Read the csv file </span></span>
<span id="cb356-2"><a href="working-with-files.html#cb356-2" tabindex="-1"></a>attrition_df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/HR-Employee-Attrition.csv&quot;</span>)</span>
<span id="cb356-3"><a href="working-with-files.html#cb356-3" tabindex="-1"></a></span>
<span id="cb356-4"><a href="working-with-files.html#cb356-4" tabindex="-1"></a><span class="co"># Table of attrition by gender</span></span>
<span id="cb356-5"><a href="working-with-files.html#cb356-5" tabindex="-1"></a><span class="fu">table</span>(attrition_df<span class="sc">$</span>Gender, attrition_df<span class="sc">$</span>Attrition)</span></code></pre></div>
<pre><code>##         
##           No Yes
##   Female 501  87
##   Male   732 150</code></pre>
<p>From the table, we can see the number of males and females who have left the company (Attrition=Yes) and those who are still employed (Attrition=No)</p>
<ul>
<li><strong>What is the attrition rate among employees in different job roles?</strong></li>
</ul>
<div class="sourceCode" id="cb358"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb358-1"><a href="working-with-files.html#cb358-1" tabindex="-1"></a><span class="co"># Read the csv file </span></span>
<span id="cb358-2"><a href="working-with-files.html#cb358-2" tabindex="-1"></a>attrition_df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/HR-Employee-Attrition.csv&quot;</span>)</span>
<span id="cb358-3"><a href="working-with-files.html#cb358-3" tabindex="-1"></a></span>
<span id="cb358-4"><a href="working-with-files.html#cb358-4" tabindex="-1"></a><span class="co"># Table of attrition by job role</span></span>
<span id="cb358-5"><a href="working-with-files.html#cb358-5" tabindex="-1"></a><span class="fu">table</span>(attrition_df<span class="sc">$</span>JobRole, attrition_df<span class="sc">$</span>Attrition)</span></code></pre></div>
<pre><code>##                            
##                              No Yes
##   Healthcare Representative 122   9
##   Human Resources            40  12
##   Laboratory Technician     197  62
##   Manager                    97   5
##   Manufacturing Director    135  10
##   Research Director          78   2
##   Research Scientist        245  47
##   Sales Executive           269  57
##   Sales Representative       50  33</code></pre>
<p>From the table above there is a high attrition rate among Laboratory technicians(62) and Sales Executives(57) while lowest attrition rate among Research directors(2) and Managers(5).</p>
<ul>
<li><strong>Does age impact attrition rate?</strong></li>
</ul>
<p>To find a clear answer to this, the ages will be summarized into different groups and create a table comparing the attrition and retention rates among different age groups.</p>
<div class="sourceCode" id="cb360"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb360-1"><a href="working-with-files.html#cb360-1" tabindex="-1"></a><span class="co"># Read the csv file </span></span>
<span id="cb360-2"><a href="working-with-files.html#cb360-2" tabindex="-1"></a>attrition_df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/HR-Employee-Attrition.csv&quot;</span>)</span>
<span id="cb360-3"><a href="working-with-files.html#cb360-3" tabindex="-1"></a></span>
<span id="cb360-4"><a href="working-with-files.html#cb360-4" tabindex="-1"></a><span class="co"># Table of attrition by age group</span></span>
<span id="cb360-5"><a href="working-with-files.html#cb360-5" tabindex="-1"></a>attrition_df<span class="sc">$</span>AgeGroup <span class="ot">&lt;-</span> <span class="fu">cut</span>(attrition_df<span class="sc">$</span>Age, <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">30</span>, <span class="dv">40</span>, <span class="dv">50</span>, <span class="dv">60</span>, <span class="dv">70</span>), <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;0-30&quot;</span>, <span class="st">&quot;31-40&quot;</span>, <span class="st">&quot;41-50&quot;</span>, <span class="st">&quot;51-60&quot;</span>, <span class="st">&quot;61-70&quot;</span>))</span>
<span id="cb360-6"><a href="working-with-files.html#cb360-6" tabindex="-1"></a></span>
<span id="cb360-7"><a href="working-with-files.html#cb360-7" tabindex="-1"></a><span class="co"># Create a table</span></span>
<span id="cb360-8"><a href="working-with-files.html#cb360-8" tabindex="-1"></a><span class="fu">table</span>(attrition_df<span class="sc">$</span>AgeGroup, attrition_df<span class="sc">$</span>Attrition)</span></code></pre></div>
<pre><code>##        
##          No Yes
##   0-30  286 100
##   31-40 534  85
##   41-50 288  34
##   51-60 125  18
##   61-70   0   0</code></pre>
<p>Young employees below 30 years tend to leave the company than elder employees of above 40 years who are mostly retained by the company.</p>
<p><span style="color: green;"><strong>Activity</strong></span></p>
<p>data link: <a href="https://raw.githubusercontent.com/insaid2018/Term-1/master/Data/Projects/car_sales.csv" class="uri">https://raw.githubusercontent.com/insaid2018/Term-1/master/Data/Projects/car_sales.csv</a></p>
<p>In this activity we shall be reading a csv file directly from an online repository and provide the answers to the questions below;</p>
<ol style="list-style-type: lower-roman">
<li>Read the csv file to a data frame <code>car_sales</code>. <strong>Hint</strong>:<em>Use the data link above as a filepath</em></li>
<li>What are the columns in the data frame? Use <code>colnames()</code> function.</li>
<li>How many records and columns are there in the data frame?</li>
<li>Print the last five records in the <code>car_sales</code> data frame</li>
</ol>
<div class="sourceCode" id="cb362"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb362-1"><a href="working-with-files.html#cb362-1" tabindex="-1"></a><span class="co"># i. Read the csv file to a data frame `car_sales`. </span></span>
<span id="cb362-2"><a href="working-with-files.html#cb362-2" tabindex="-1"></a><span class="co"># CODE HERE </span></span>
<span id="cb362-3"><a href="working-with-files.html#cb362-3" tabindex="-1"></a></span>
<span id="cb362-4"><a href="working-with-files.html#cb362-4" tabindex="-1"></a><span class="co"># ii. What are the columns in the data frame? Use `colnames()` function.</span></span>
<span id="cb362-5"><a href="working-with-files.html#cb362-5" tabindex="-1"></a><span class="co"># CODE HERE </span></span>
<span id="cb362-6"><a href="working-with-files.html#cb362-6" tabindex="-1"></a></span>
<span id="cb362-7"><a href="working-with-files.html#cb362-7" tabindex="-1"></a><span class="co"># iii. How many records and columns are there in the data frame?</span></span>
<span id="cb362-8"><a href="working-with-files.html#cb362-8" tabindex="-1"></a><span class="co"># CODE HERE </span></span>
<span id="cb362-9"><a href="working-with-files.html#cb362-9" tabindex="-1"></a></span>
<span id="cb362-10"><a href="working-with-files.html#cb362-10" tabindex="-1"></a><span class="co"># iv. Print the last five records in the `car_sales` data frame</span></span>
<span id="cb362-11"><a href="working-with-files.html#cb362-11" tabindex="-1"></a><span class="co"># CODE HERE </span></span></code></pre></div>
</div>
</div>
</div>
<div id="code-challenge-3" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> CODE CHALLENGE<a href="working-with-files.html#code-challenge-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>excel data - <a href="https://www.kaggle.com/datasets/suruchiarora/yahoo-finance-dataset-2018-2023" class="uri">https://www.kaggle.com/datasets/suruchiarora/yahoo-finance-dataset-2018-2023</a></p>
<p>csv data - <a href="https://www.kaggle.com/datasets/prasad22/vehicle-population-statistics" class="uri">https://www.kaggle.com/datasets/prasad22/vehicle-population-statistics</a></p>
<ol style="list-style-type: decimal">
<li><p>In this question you will be required to download kaggle data set, the Yahoo data 2018-2023 from <a href="https://www.kaggle.com/datasets/suruchiarora/yahoo-finance-dataset-2018-2023">here</a>. Follow the instructions below to work on the data set.</p>
<ol style="list-style-type: lower-alpha">
<li>Install and load the <code>readxl</code> library.</li>
<li>Read the excel file to a data frame using <code>read_excel()</code> function. Store the data frame in a variable, <code>yahoo_data</code>.</li>
<li>Use <code>str()</code> command to get a glimpse of the dataset.</li>
<li>Show the first 4 records and the last 10 records of the <code>yahoo_data</code> using the <code>head()</code> and <code>tail()</code> command respectively.</li>
<li>Calculate the total null values in the <code>yahoo_data</code> using the <code>sum()</code> along with <code>is.na()</code> functions.</li>
<li>Find the total duplicated observations in the <code>yahoo_data</code> with <code>sum()</code> and <code>duplicated()</code> commands.</li>
</ol></li>
<li><p>In this second part of the challenge, you will be required to download a kaggle data set - the vehicle population statistics csv file from <a href="https://www.kaggle.com/datasets/prasad22/vehicle-population-statistics">here</a>. You will use the data set to answer the following questions.</p>
<ol style="list-style-type: lower-alpha">
<li>Read the data set to a data frame using the <code>read.csv</code> inbuilt function. Store the data frame in a variable <code>vehicle_data</code>.</li>
<li>Display the first 6 rows and the last 3 rows using the <code>head()</code> and <code>tail()</code> functions respectively.</li>
<li>Show the first 10 observations in the <code>"Scooters"</code> column.</li>
<li>What are the number of Jeeps per Region. Use table functions. <strong>Hint</strong>:<em>Refer from “What is the survival rate by gender?” in the content above</em></li>
<li>What are the distinct categories in the <code>vehicle_data</code>.</li>
</ol></li>
</ol>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="loops.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="dpyr-library.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/balsaedi/RCourse/edit/main/04-read_dataframe_from_file.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/balsaedi/RCourse/blob/main/04-read_dataframe_from_file.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
